// @skbkontur/react-ui/cjs/components/DateInput/DateInput.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: default;\n      color: ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-size: ", ";\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-size: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-size: ", ";\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: pointer;\n      color: ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var jsStyles = {
  icon: function icon(t) {
    return (0, _Emotion.css)(_templateObject(),

    t.dateInputIconColor);

  },

  iconSmall: function iconSmall(t) {
    return (0, _Emotion.css)(_templateObject2(),
    t.inputFontSizeSmall);

  },

  iconMedium: function iconMedium(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.inputFontSizeMedium);

  },

  iconLarge: function iconLarge(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.inputFontSizeLarge);

  },

  iconDisabled: function iconDisabled(t) {
    return (0, _Emotion.css)(_templateObject5(),

    t.textColorDisabled);

  } };exports.jsStyles = jsStyles;
