// @skbkontur/react-ui/cjs/components/Textarea/Textarea.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      -ms-user-select: none;\n      color: #aaa;\n      left: 12px;\n      overflow: hidden;\n      position: absolute;\n      right: 12px;\n      top: 9px;\n      user-select: none;\n      z-index: 3;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 0;\n      left: 0;\n      position: absolute;\n      top: 0;\n      visibility: hidden;\n      width: 100%;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n      ", "&:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n      ", "&:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      -webkit-appearance: none;\n      background-clip: padding-box;\n      background: ", ";\n      border: 1px solid ", ";\n      border-top-color: ", ";\n      box-shadow: ", ";\n      box-sizing: border-box;\n      color: ", ";\n      font-family: inherit;\n      font-size: inherit;\n      line-height: inherit;\n      max-width: 100%;\n      min-height: ", ";\n      outline: none;\n      padding: 6px 10px;\n      transition: height 0.2s ease-out;\n      vertical-align: middle;\n      width: 100%;\n\n      &:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n        position: relative;\n        z-index: 2;\n\n        &::placeholder {\n          color: ", ";\n        }\n      }\n      &:disabled {\n        color: ", ";\n        background: ", ";\n        border-color: ", ";\n        box-shadow: none;\n      }\n\n      &::placeholder {\n        color: ", ";\n      }\n\n      &:-moz-placeholder {\n        color: inherit; // Need to be discussed\n      }\n\n      &::-moz-placeholder {\n        color: inherit;\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      font-size: ", ";\n      line-height: ", ";\n      position: relative;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),

    t.fontSizeMedium,
    t.textareaLineHeight);


  },

  textarea: function textarea(t) {
    return (0, _Emotion.css)(_templateObject2(),


    t.textareaBg,
    t.borderColorGrayLight,
    t.textareaBorderTopColor,
    t.textareaShadow,

    t.textareaColor,




    t.controlHeightSmall,







    t.borderColorFocus,
    t.borderColorFocus,




    t.placeholderColorLight,



    t.textColorDisabled,
    t.textareaDisabledBg,
    t.textareaDisabledBorderColor,




    t.placeholderColor);










  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject3(),
    (0, _Emotion.cssName)(styles.textarea(t)),
    t.borderColorError,
    t.borderColorError,

    (0, _Emotion.cssName)(styles.textarea(t)),
    t.borderColorError,
    t.borderColorError);


  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject4(),
    (0, _Emotion.cssName)(styles.textarea(t)),
    t.borderColorWarning,
    t.borderColorWarning,

    (0, _Emotion.cssName)(styles.textarea(t)),
    t.borderColorWarning,
    t.borderColorWarning);


  },

  fake: function fake() {
    return (0, _Emotion.css)(_templateObject5());







  },

  placeholder: function placeholder() {
    return (0, _Emotion.css)(_templateObject6());










  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
