// @skbkontur/react-ui/cjs/components/Kebab/Kebab.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      overflow: hidden;\n      border-radius: ", ";\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", " & {\n        margin-top: -5px;\n        margin-left: 0.5px;\n      }\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", " & {\n        margin-top: -2px;\n        margin-left: 0.5px;\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", " & {\n        margin-top: 1px;\n        margin-left: 0.8124px;\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        cursor: default;\n\n        &:hover {\n          background: none;\n        }\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        background: rgba(0, 0, 0, 0.09);\n        cursor: default;\n      }\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        background: rgba(0, 0, 0, 0.09);\n        border-color: ", ";\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      text-align: center;\n      border: 2px solid transparent;\n      box-sizing: border-box;\n      border-radius: 50%;\n      user-select: none;\n      cursor: pointer;\n      outline: 0;\n      transition: background 0.12s ease-out;\n      width: 26px;\n      height: 26px;\n\n      &:hover {\n        background: rgba(0, 0, 0, 0.09);\n      }\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  kebab: function kebab() {
    return (0, _Emotion.css)(_templateObject());
















  },

  focused: function focused(t) {
    return (0, _Emotion.css)(_templateObject2(),
    (0, _Emotion.cssName)(styles.kebab()),

    t.borderColorFocus);


  },

  opened: function opened() {
    return (0, _Emotion.css)(_templateObject3(),
    (0, _Emotion.cssName)(styles.kebab()));




  },

  disabled: function disabled() {
    return (0, _Emotion.css)(_templateObject4(),
    (0, _Emotion.cssName)(styles.kebab()));







  },

  iconsmall: function iconsmall() {
    return (0, _Emotion.css)(_templateObject5(),
    (0, _Emotion.cssName)(styles.kebab()));




  },

  iconmedium: function iconmedium() {
    return (0, _Emotion.css)(_templateObject6(),
    (0, _Emotion.cssName)(styles.kebab()));




  },

  iconlarge: function iconlarge() {
    return (0, _Emotion.css)(_templateObject7(),
    (0, _Emotion.cssName)(styles.kebab()));




  },

  menu: function menu(t) {
    return (0, _Emotion.css)(_templateObject8(),

    t.popupBorderRadius);

  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
