// @skbkontur/react-ui/cjs/components/Toast/ToastView.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      height: 8px;\n      line-height: 0;\n      margin: -12px -8px;\n      padding: 12px 8px;\n      text-align: center;\n      width: 8px;\n\n      &:hover {\n        color: ", ";\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      font-weight: 600;\n      margin: -10px 24px -11px 4px;\n      padding: 10px 10px 11px;\n\n      &:hover {\n        text-decoration: underline;\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      line-height: 40px;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: fixed;\n      top: 0;\n      left: 0;\n      right: 0;\n      height: 0;\n      text-align: center;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      border-radius: 2px;\n      color: ", ";\n      display: inline-block;\n      font-size: 14px;\n      opacity: 1;\n      padding: 10px 20px 11px;\n      position: relative;\n      top: 20px;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.toastBg,

    t.toastColor);







  },

  wrapper: function wrapper() {
    return (0, _Emotion.css)(_templateObject2());







  },

  closeWrapper: function closeWrapper() {
    return (0, _Emotion.css)(_templateObject3());






  },

  link: function link(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.toastLinkColor);










  },

  close: function close(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.toastCloseColor,










    t.toastCloseHoverColor);


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
