// @skbkontur/react-ui/components/Radio/Radio.styles/Radio.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #a0a0a0;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      display: ", ";\n      line-height: 20px;\n      margin-left: 9px;\n      white-space: normal;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", " !important;\n      border-color: transparent !important;\n      box-shadow: ", " !important;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      height: 0;\n      opacity: 0;\n      position: absolute;\n      width: 0;\n      z-index: -1;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      background-color: ", ";\n\n      &::before {\n        content: ' ';\n        position: absolute;\n        top: 0;\n        right: 0;\n        bottom: 0;\n        left: 0;\n        margin: auto;\n        height: 8px;\n        width: 8px;\n        border-radius: 50%;\n        background: ", " !important;\n      }\n\n      ", "&::before {\n        background: ", " !important;\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      background-image: ", ";\n      border-radius: 50%;\n      border: ", ";\n      box-shadow: ", ";\n      box-sizing: border-box;\n      display: inline-block;\n      height: ", ";\n      margin-bottom: 2px;\n      margin-top: 2px;\n      position: relative;\n      vertical-align: ", ";\n      width: ", ";\n\n      ", ":hover &:not(", ") {\n        background: ", ";\n        box-shadow: ", ";\n      }\n      ", ":hover &", ":not(", ") {\n        background: ", ";\n      }\n      ", ":active & {\n        background: ", ";\n        box-shadow: ", ";\n      }\n      ", ":focus + &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n      &::after {\n        content: ' ';\n        display: inline-block;\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      content: ' ';\n      position: absolute;\n      left: -2px;\n      top: -2px;\n      width: 20px;\n      height: 20px;\n      border-width: 2px;\n      border-style: solid;\n      border-radius: 50%;\n      box-sizing: border-box;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: pointer;\n      position: relative;\n      white-space: nowrap;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root() {
    return css(_templateObject());
  },
  after: function after() {
    return css(_templateObject2());
  },
  radio: function radio(t) {
    return css(_templateObject3(), t.radioBgImage, t.radioBorder, t.radioBoxShadow, t.radioSize, t.radioVerticalAlign, t.radioSize, cssName(styles.root()), cssName(styles.disabled(t)), t.radioHoverBg, t.radioHoverShadow, cssName(styles.root()), cssName(styles.checked(t)), cssName(styles.disabled(t)), t.radioCheckedHoverBgColor, cssName(styles.root()), t.radioActiveBg, t.radioActiveShadow, cssName(styles.input()), styles.after(), t.radioFocusShadow, t.borderColorFocus);
  },
  focus: function focus(t) {
    return css(_templateObject4(), styles.after(), t.radioFocusShadow, t.borderColorFocus);
  },
  warning: function warning(t) {
    return css(_templateObject5(), styles.after(), t.radioFocusShadow, t.borderColorWarning);
  },
  error: function error(t) {
    return css(_templateObject6(), styles.after(), t.radioFocusShadow, t.borderColorError);
  },
  checked: function checked(t) {
    return css(_templateObject7(), t.radioCheckedBgColor, t.radioCheckedBulletColor, cssName(styles.disabled(t)), t.gray);
  },
  input: function input() {
    return css(_templateObject8());
  },
  disabled: function disabled(t) {
    return css(_templateObject9(), t.bgDisabled, t.radioDisabledShadow);
  },
  label: function label(t) {
    return css(_templateObject10(), t.radioLabelDisplay);
  },
  labelDisabled: function labelDisabled() {
    return css(_templateObject11());
  },
  placeholder: function placeholder() {
    return css(_templateObject12());
  }
};
export var jsStyles = memoizeStyle(styles);
