// @skbkontur/react-ui/cjs/components/TopBar/TopBar.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject22() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 10px;\n    "]);_templateObject22 = function _templateObject22() {return data;};return data;}function _templateObject21() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-sizing: border-box;\n      height: 0;\n      overflow: hidden;\n      padding-right: 10px;\n      white-space: normal;\n    "]);_templateObject21 = function _templateObject21() {return data;};return data;}function _templateObject20() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      right: 0;\n      text-align: center;\n      width: 20px;\n    "]);_templateObject20 = function _templateObject20() {return data;};return data;}function _templateObject19() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: #aaa;\n      padding-left: 10px;\n      position: absolute;\n      right: 20px;\n    "]);_templateObject19 = function _templateObject19() {return data;};return data;}function _templateObject18() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-sizing: border-box;\n      left: 0;\n      overflow: hidden;\n      padding: 0 20px 0 10px;\n      position: absolute;\n      text-overflow: ellipsis;\n      width: 100%;\n    "]);_templateObject18 = function _templateObject18() {return data;};return data;}function _templateObject17() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: table-cell;\n      height: 50px;\n      white-space: nowrap;\n    "]);_templateObject17 = function _templateObject17() {return data;};return data;}function _templateObject16() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      box-sizing: border-box;\n      height: 50px;\n      line-height: 50px;\n      padding: 0 10px;\n      vertical-align: middle;\n      white-space: nowrap;\n    "]);_templateObject16 = function _templateObject16() {return data;};return data;}function _templateObject15() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: flex;\n      flex-wrap: nowrap;\n      justify-content: space-between;\n    "]);_templateObject15 = function _templateObject15() {return data;};return data;}function _templateObject14() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin: 0 -10px;\n    "]);_templateObject14 = function _templateObject14() {return data;};return data;}function _templateObject13() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin: auto;\n    "]);_templateObject13 = function _templateObject13() {return data;};return data;}function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-bottom: 0;\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &,\n      &:hover,\n      ", " {\n        background: #ffe3e3;\n      }\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &,\n      &:hover,\n      ", " {\n        background: #ffeca9;\n      }\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-right: 0;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-right: 4px;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: none;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: pointer;\n      height: 50px;\n      line-height: 50px;\n      outline: 0;\n      padding: 0 10px;\n      position: relative;\n      vertical-align: middle;\n\n      &:hover,\n      &:focus {\n        background: rgba(0, 0, 0, 0.06);\n      }\n\n      .kontur-projects_opened & {\n        background: rgba(0, 0, 0, 0.06);\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: rgba(0, 0, 0, 0.06);\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      padding-right: 60px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      align-items: stretch;\n      display: flex;\n      flex-wrap: nowrap;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      align-self: center;\n      background-color: ", ";\n      height: 30px;\n      width: 1px;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      box-shadow: ", ";\n      height: 50px;\n      margin-bottom: 20px;\n      padding: 0 15px;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.tbBg,
    t.tbShadow);




  },

  divider: function divider(t) {
    return (0, _Emotion.css)(_templateObject2(),

    t.tdDividerBg);



  },

  endItems: function endItems() {
    return (0, _Emotion.css)(_templateObject3());




  },

  startItems: function startItems() {
    return (0, _Emotion.css)(_templateObject4(),
    styles.endItems());



  },

  buttonActive: function buttonActive() {
    return (0, _Emotion.css)(_templateObject5());


  },

  button: function button() {
    return (0, _Emotion.css)(_templateObject6());

















  },

  noShadow: function noShadow() {
    return (0, _Emotion.css)(_templateObject7());


  },

  icon: function icon() {
    return (0, _Emotion.css)(_templateObject8());


  },

  iconOnly: function iconOnly() {
    return (0, _Emotion.css)(_templateObject9());


  },

  usePay: function usePay() {
    return (0, _Emotion.css)(_templateObject10(),


    (0, _Emotion.cssName)(styles.buttonActive()));



  },

  useDanger: function useDanger() {
    return (0, _Emotion.css)(_templateObject11(),


    (0, _Emotion.cssName)(styles.buttonActive()));



  },

  noMargin: function noMargin() {
    return (0, _Emotion.css)(_templateObject12());


  },

  center: function center() {
    return (0, _Emotion.css)(_templateObject13());


  },

  containerWrap: function containerWrap() {
    return (0, _Emotion.css)(_templateObject14());


  },

  container: function container() {
    return (0, _Emotion.css)(_templateObject15());




  },

  item: function item() {
    return (0, _Emotion.css)(_templateObject16());








  },

  spwDropdown: function spwDropdown() {
    return (0, _Emotion.css)(_templateObject17());




  },

  organizationsTitle: function organizationsTitle() {
    return (0, _Emotion.css)(_templateObject18());








  },

  organizationsComment: function organizationsComment() {
    return (0, _Emotion.css)(_templateObject19());





  },

  organizationsArrow: function organizationsArrow() {
    return (0, _Emotion.css)(_templateObject20());





  },

  organizationsTitleDummy: function organizationsTitleDummy() {
    return (0, _Emotion.css)(_templateObject21());






  },

  organizationsCommentDummy: function organizationsCommentDummy() {
    return (0, _Emotion.css)(_templateObject22());


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
