// @skbkontur/react-ui/cjs/internal/InternalMenu/InternalMenu.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1);\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: 0 1px 0 rgba(0, 0, 0, 0.1);\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      bottom: -5px;\n      position: relative;\n      z-index: 1;\n      width: 100%;\n      overflow: hidden;\n      line-height: 18px;\n      box-sizing: border-box;\n      padding: 6px 18px 7px 8px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      top: -5px;\n      position: relative;\n      z-index: 1;\n      width: 100%;\n      overflow: hidden;\n      line-height: 18px;\n      box-sizing: border-box;\n      padding: 6px 18px 7px 8px;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 1px solid #d5d5d5;\n      box-shadow: 0 2px 6px rgba(0, 0, 0, 0.2);\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      overflow: auto;\n      padding: 5px 0;\n      outline: none;\n      background: ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),



    t.bgDefault);

  },

  shadow: function shadow() {
    return (0, _Emotion.css)(_templateObject2());



  },

  header: function header() {
    return (0, _Emotion.css)(_templateObject3());









  },

  footer: function footer() {
    return (0, _Emotion.css)(_templateObject4());









  },

  fixedHeader: function fixedHeader() {
    return (0, _Emotion.css)(_templateObject5());


  },

  fixedFooter: function fixedFooter() {
    return (0, _Emotion.css)(_templateObject6());


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
