// @skbkontur/react-ui/components/SidePage/SidePageHeader/SidePageHeader.js
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import cn from 'classnames';
import { Sticky } from "../../Sticky";
import { CrossIcon } from "../../../internal/icons/CrossIcon";
import { isFunction } from "../../../lib/utils";
import { ThemeContext } from "../../../lib/theming/ThemeContext";
import { jsStyles } from "../SidePage.styles";
import { SidePageContext } from "../SidePageContext";
var REGULAR_HEADER_PADDING_TOP = 25;
var FIXED_HEADER_PADDING_TOP = 13;
var FONT_FAMILY_CORRECTION = 1;
var CLOSE_ELEMENT_OFFSET = REGULAR_HEADER_PADDING_TOP - FIXED_HEADER_PADDING_TOP - FONT_FAMILY_CORRECTION;
var FIXED_HEADER_HEIGHT = 50;
/**
                               * Шапка сайдпейджа
                               *
                               * @visibleName SidePage.Header
                               */

export var SidePageHeader =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(SidePageHeader, _React$Component);

  function SidePageHeader() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.state = {
      isReadyToFix: false
    };
    _this.theme = void 0;
    _this.wrapper = null;
    _this.lastRegularHeight = 0;

    _this.componentDidMount = function () {
      window.addEventListener('scroll', _this.update, true);
    };

    _this.componentWillUnmount = function () {
      window.removeEventListener('scroll', _this.update, true);
    };

    _this.update = function () {
      _this.updateReadyToFix();
    };

    _this.renderHeader = function (fixed) {
      var _cn, _cn2;

      if (fixed === void 0) {
        fixed = false;
      }

      return React.createElement("div", {
        className: cn(jsStyles.header(), (_cn = {}, _cn[jsStyles.headerFixed(_this.theme)] = fixed, _cn))
      }, _this.renderClose(), React.createElement("div", {
        className: cn(jsStyles.title(), (_cn2 = {}, _cn2[jsStyles.titleFixed()] = fixed, _cn2))
      }, isFunction(_this.props.children) ? _this.props.children(fixed) : _this.props.children));
    };

    _this.renderCloseContent = function (fixed) {
      return React.createElement(SidePageContext.Consumer, null, function (_ref) {
        var _cn3;

        var requestClose = _ref.requestClose;
        return React.createElement("a", {
          className: cn(jsStyles.close(_this.theme), (_cn3 = {}, _cn3[jsStyles.fixed()] = fixed, _cn3)),
          onClick: requestClose,
          "data-tid": "SidePage__close"
        }, React.createElement("span", {
          className: jsStyles.closeIcon()
        }, React.createElement(CrossIcon, null)));
      });
    };

    _this.renderClose = function () {
      return React.createElement(Sticky, {
        side: "top",
        offset: CLOSE_ELEMENT_OFFSET
      }, _this.renderCloseContent);
    };

    _this.updateReadyToFix = function () {
      if (_this.wrapper) {
        var wrapperScrolledUp = _this.wrapper.getBoundingClientRect().top;

        var isReadyToFix = _this.regularHeight + wrapperScrolledUp <= FIXED_HEADER_HEIGHT;

        _this.setState(function (state) {
          return state.isReadyToFix !== isReadyToFix ? {
            isReadyToFix: isReadyToFix
          } : state;
        });
      }
    };

    _this.wrapperRef = function (el) {
      _this.wrapper = el;
    };

    return _this;
  }

  var _proto = SidePageHeader.prototype;

  _proto.render = function render() {
    var _this2 = this;

    return React.createElement(ThemeContext.Consumer, null, function (theme) {
      _this2.theme = theme;
      return _this2.renderMain();
    });
  };

  _proto.renderMain = function renderMain() {
    var isReadyToFix = this.state.isReadyToFix;
    return React.createElement("div", {
      ref: this.wrapperRef
    }, isReadyToFix ? React.createElement(Sticky, {
      side: "top"
    }, this.renderHeader) : this.renderHeader());
  };

  _createClass(SidePageHeader, [{
    key: "regularHeight",
    get: function get() {
      var isReadyToFix = this.state.isReadyToFix;

      if (!this.wrapper) {
        return 0;
      }

      if (!isReadyToFix) {
        this.lastRegularHeight = this.wrapper.getBoundingClientRect().height;
      }

      return this.lastRegularHeight;
    }
  }]);

  return SidePageHeader;
}(React.Component);
SidePageHeader.__KONTUR_REACT_UI__ = 'SidePageHeader';
