// @skbkontur/react-ui/internal/ThemePlayground/CheckboxPlayground/CheckboxPlayground.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { Gapped } from "../../../components/Gapped";
import { Checkbox } from "../../../components/Checkbox";
import { getComponentsFromPropsList } from "../helpers";
var propsList = [{
  children: 'Default'
}, {
  children: 'Checked',
  checked: true
}, {
  children: 'Disabled',
  checked: true,
  disabled: true
}, {
  children: 'Semichecked',
  initialIndeterminate: true
}, {
  children: 'Focused',
  focused: true
}, {
  children: 'Error',
  error: true
}, {
  children: 'Warning',
  warning: true
}];
export var CheckboxPlayground =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(CheckboxPlayground, _React$Component);

  function CheckboxPlayground() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = CheckboxPlayground.prototype;

  _proto.render = function render() {
    return React.createElement(Gapped, {
      vertical: true
    }, getComponentsFromPropsList(React.createElement(Checkbox, null), propsList));
  };

  return CheckboxPlayground;
}(React.Component);
