// @skbkontur/react-ui/cjs/components/ScrollContainer/ScrollContainer.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &::after {\n        background: #ccc;\n        background: rgba(255, 255, 255, 0.5);\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      width: 10px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      right: 2px;\n      transition: width 0.2s;\n      width: 4px;\n      z-index: 200;\n\n      &::after {\n        background: #b7b7b7;\n        border-radius: 5px;\n        bottom: 1px;\n        content: '';\n        display: block;\n        left: 0;\n        position: absolute;\n        right: 0;\n        top: 1px;\n      }\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-bottom: -1px;\n      max-height: 100%;\n\n      /* IE sometimes enabled scroll: http://codepen.io/anon/pen/RRrLNX */\n      padding-bottom: 1px;\n      overflow-y: scroll;\n\n      &::-webkit-scrollbar {\n        visibility: hidden;\n        width: 0;\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      overflow: hidden;\n      position: relative;\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject());




  },

  inner: function inner() {
    return (0, _Emotion.css)(_templateObject2());












  },

  scroll: function scroll() {
    return (0, _Emotion.css)(_templateObject3());


















  },

  scrollHover: function scrollHover() {
    return (0, _Emotion.css)(_templateObject4());


  },

  scrollInvert: function scrollInvert() {
    return (0, _Emotion.css)(_templateObject5());





  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
