// @skbkontur/react-ui/internal/SpinnerOld/SpinnerOldFallback/SpinnerOldFallback.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import * as PropTypes from 'prop-types';
import { SPINNER_CLOUD_SIZE } from "../../icons/SpinnerOldIcon";
import fallbackImage_mini from "../fallback_circle.png";
import fallbackImage_mini_dimmed from "../fallback_circle_dimmed.png";
import fallbackImage_big from "../fallback_cloud_big.png";
import fallbackImage_normal from "../fallback_cloud_normal.png";
import { jsStyles } from "../SpinnerOld.styles";
export var types = {
  big: 'big',
  mini: 'mini',
  normal: 'normal'
};
export var SpinnerOldFallback =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(SpinnerOldFallback, _React$Component);

  function SpinnerOldFallback() {
    var _this$_framesCount, _this$imageUrls;

    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.state = {
      frame: 0
    };
    _this.mounted = false;
    _this._framesCount = (_this$_framesCount = {}, _this$_framesCount[types.mini] = 180, _this$_framesCount[types.normal] = 60, _this$_framesCount[types.big] = 60, _this$_framesCount.dimmed = 60, _this$_framesCount);
    _this.imageUrls = (_this$imageUrls = {}, _this$imageUrls[types.mini] = fallbackImage_mini, _this$imageUrls[types.normal] = fallbackImage_normal, _this$imageUrls[types.big] = fallbackImage_big, _this$imageUrls.dimmed = fallbackImage_mini_dimmed, _this$imageUrls);

    _this.animate = function () {
      if (!_this.mounted) {
        return;
      }

      var frame = _this.state.frame;

      var framesCount = _this._framesCount[_this.getSpriteSettingsKey()];

      var nextFrame = frame < framesCount ? frame + 1 : 0;

      _this.setState({
        frame: nextFrame
      });

      setTimeout(_this.animate, 1000 / 25);
    };

    _this.getSpriteSettingsKey = function () {
      return _this.props.type === types.mini && _this.props.dimmed ? 'dimmed' : _this.props.type;
    };

    return _this;
  }

  var _proto = SpinnerOldFallback.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.mounted = true;
    this.animate();
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    this.mounted = false;
  };

  _proto.render = function render() {
    return this.props.type === 'mini' ? this.renderCircle() : this.renderCloud();
  };

  _proto.renderCircle = function renderCircle() {
    var dimmed = this.props.dimmed;
    var frame = this.state.frame;
    var cssSet = {
      backgroundImage: "url('" + this.imageUrls[dimmed ? 'dimmed' : 'mini'] + "')",
      height: 16,
      width: 16,
      marginBottom: -3,
      marginLeft: -1,
      marginRight: -1
    };

    if (!process.env.enableReactTesting) {
      cssSet.backgroundPosition = "0 -" + frame * 16 + "px";
    }

    return React.createElement("span", {
      className: jsStyles.fallback(),
      style: cssSet
    });
  };

  _proto.renderCloud = function renderCloud() {
    var type = this.props.type;
    var frame = this.state.frame;
    var multiply = type === 'big' ? 2 : 1;
    var cssSet = {
      backgroundImage: "url('" + this.imageUrls[type] + "')",
      height: SPINNER_CLOUD_SIZE.height * multiply,
      top: 0,
      width: SPINNER_CLOUD_SIZE.width * multiply
    };

    if (!process.env.enableReactTesting) {
      cssSet.backgroundPosition = "0 -" + frame * SPINNER_CLOUD_SIZE.height * multiply + "px";
    }

    return React.createElement("span", {
      className: jsStyles.fallback(),
      style: cssSet
    });
  };

  return SpinnerOldFallback;
}(React.Component);
SpinnerOldFallback.__KONTUR_REACT_UI__ = 'SpinnerOldFallback';
SpinnerOldFallback.propTypes = {
  type: PropTypes.oneOf(Object.keys(types)),
  dimmed: PropTypes.bool
};
