// @skbkontur/react-ui/components/TokenInput/TokenInputMenu/TokenInputMenu.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { Popup } from "../../../internal/Popup";
import { ComboBoxMenu } from "../../../internal/CustomComboBox";
export var TokenInputMenu =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(TokenInputMenu, _React$Component);

  function TokenInputMenu() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.menu = null;

    _this.getMenuRef = function () {
      return _this.menu;
    };

    _this.menuRef = function (node) {
      return _this.menu = node;
    };

    return _this;
  }

  var _proto = TokenInputMenu.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        loading = _this$props.loading,
        maxMenuHeight = _this$props.maxMenuHeight,
        renderTotalCount = _this$props.renderTotalCount,
        totalCount = _this$props.totalCount,
        opened = _this$props.opened,
        items = _this$props.items,
        renderNotFound = _this$props.renderNotFound,
        renderItem = _this$props.renderItem,
        onValueChange = _this$props.onValueChange,
        renderAddButton = _this$props.renderAddButton;
    return React.createElement(Popup, {
      opened: opened,
      positions: ['bottom left'],
      anchorElement: this.props.anchorElement,
      margin: 6,
      popupOffset: 5
    }, React.createElement(ComboBoxMenu, {
      items: items,
      loading: loading,
      maxMenuHeight: maxMenuHeight,
      onValueChange: onValueChange,
      opened: opened,
      refMenu: this.menuRef,
      renderTotalCount: renderTotalCount,
      renderItem: renderItem,
      renderNotFound: renderNotFound,
      totalCount: totalCount,
      renderAddButton: renderAddButton
    }));
  };

  return TokenInputMenu;
}(React.Component);
TokenInputMenu.__KONTUR_REACT_UI__ = 'TokenInputMenu';
