// @skbkontur/react-ui/cjs/internal/ThemePlayground/Playground.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject18() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n    "]);_templateObject18 = function _templateObject18() {return data;};return data;}function _templateObject17() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n    "]);_templateObject17 = function _templateObject17() {return data;};return data;}function _templateObject16() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject16 = function _templateObject16() {return data;};return data;}function _templateObject15() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n    "]);_templateObject15 = function _templateObject15() {return data;};return data;}function _templateObject14() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:after {\n        box-shadow: 0 2px 10px 0px ", ";\n      }\n    "]);_templateObject14 = function _templateObject14() {return data;};return data;}function _templateObject13() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      width: 210px;\n    "]);_templateObject13 = function _templateObject13() {return data;};return data;}function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin: 0 -32px;\n      padding: 11px 32px;\n      position: relative;\n\n      &::after {\n        content: '';\n        position: absolute;\n        bottom: 0;\n        left: 0;\n        height: 1px;\n        width: 100%;\n        z-index: -1;\n      }\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 300px;\n      margin: auto;\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      width: 500px;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: relative;\n      z-index: 1;\n      padding-left: 10px;\n      width: 240px;\n      display: flex;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: relative;\n      z-index: 1;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-left: auto;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 16px;\n      width: 16px;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-bottom: 5px;\n      white-space: nowrap;\n      overflow: hidden;\n      text-overflow: ellipsis;\n      width: 250px;\n      position: relative;\n      z-index: 1;\n      line-height: 34px;\n      color: ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 30px;\n      padding-bottom: 30px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      max-width: 700px;\n      color: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      width: 110px;\n      color: ", ";\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin: -16px;\n      padding: 32px;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  playground: function playground() {
    return (0, _Emotion.css)(_templateObject());



  },

  title: function title(t) {
    return (0, _Emotion.css)(_templateObject2(),

    t.textColorDefault);

  },

  componentsGroup: function componentsGroup(t) {
    return (0, _Emotion.css)(_templateObject3(),

    t.textColorDefault);

  },

  sidePageBody: function sidePageBody() {
    return (0, _Emotion.css)(_templateObject4());



  },

  variableName: function variableName(t) {
    return (0, _Emotion.css)(_templateObject5(),








    t.textColorDefault);

  },

  colorIcon: function colorIcon() {
    return (0, _Emotion.css)(_templateObject6());



  },

  baseLinkWrapper: function baseLinkWrapper() {
    return (0, _Emotion.css)(_templateObject7());


  },

  linkRoot: function linkRoot() {
    return (0, _Emotion.css)(_templateObject8());



  },

  baseVariableRoot: function baseVariableRoot() {
    return (0, _Emotion.css)(_templateObject9());






  },

  loaderWrapper: function loaderWrapper() {
    return (0, _Emotion.css)(_templateObject10());


  },

  loader: function loader() {
    return (0, _Emotion.css)(_templateObject11());



  },

  tabsWrapper: function tabsWrapper() {
    return (0, _Emotion.css)(_templateObject12());














  },

  tooltipContent: function tooltipContent() {
    return (0, _Emotion.css)(_templateObject13());


  },

  stickyTabsWrapper: function stickyTabsWrapper(t) {
    return (0, _Emotion.css)(_templateObject14(),

    t.textColorDefault);


  },
  tabsInnerWrapper: function tabsInnerWrapper(t) {
    return (0, _Emotion.css)(_templateObject15(),
    t.textColorDefault);

  },
  playgroundWrapper: function playgroundWrapper(t) {
    return (0, _Emotion.css)(_templateObject16(),
    t.bgDefault);

  },
  editorHeaderWrapper: function editorHeaderWrapper(t) {
    return (0, _Emotion.css)(_templateObject17(),
    t.textColorDefault);

  },
  editorGroupHeader: function editorGroupHeader(t) {
    return (0, _Emotion.css)(_templateObject18(),
    t.textColorDefault);

  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
