// @skbkontur/react-ui/cjs/internal/ThemePlayground/Playground.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.Playground = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Search = _interopRequireDefault(require("@skbkontur/react-icons/Search"));
var _Card = _interopRequireDefault(require("@skbkontur/react-icons/Card"));
var _Link = _interopRequireDefault(require("@skbkontur/react-icons/Link"));
var _Ok = _interopRequireDefault(require("@skbkontur/react-icons/Ok"));
var _Error = _interopRequireDefault(require("@skbkontur/react-icons/Error"));
var _Trash = _interopRequireDefault(require("@skbkontur/react-icons/Trash"));
var _HelpDot = _interopRequireDefault(require("@skbkontur/react-icons/HelpDot"));

var _Button = require("../../components/Button");
var _Tabs = require("../../components/Tabs");
var _Gapped = require("../../components/Gapped");
var _Link2 = require("../../components/Link");
var _Input = require("../../components/Input");
var _ThemeContext = require("../../lib/theming/ThemeContext");
var _Tooltip = require("../../components/Tooltip");
var _Sticky = require("../../components/Sticky");


var _constants = require("./constants");
var _TokenInputPlayground = require("./TokenInputPlayground");
var _AnotherInputsPlayground = require("./AnotherInputsPlayground");
var _TogglePlayground = require("./TogglePlayground");
var _SwitcherPlayground = require("./SwitcherPlayground");
var _FxInputPlayground = require("./FxInputPlayground");
var _CurrencyInputPlayground = require("./CurrencyInputPlayground");
var _SelectPlayground = require("./SelectPlayground");
var _helpers = require("./helpers");
var _CheckboxPlayground = require("./CheckboxPlayground");
var _RadioPlayground = require("./RadioPlayground");
var _PagingPlayground = require("./PagingPlayground");
var _HintPlayground = require("./HintPlayground");
var _ComponentsGroup = require("./ComponentsGroup");
var _Playground = require("./Playground.styles");

var useSticky = process.env.enableReactTesting !== 'true';var







Playground = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(Playground, _React$Component);function Playground() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.
    theme = void 0;_this.
    stopEl = _react.default.createRef();_this.




































    renderTabsGroup = function () {
      return useSticky ?
      _react.default.createElement(_Sticky.Sticky, { side: 'top', getStop: _this.getStickyStop },
      _this.renderTabs()) :


      _this.renderTabs();

    };_this.

























    renderSizesGroup = function () {
      var Group = function Group(_ref) {var size = _ref.size;return (
          _react.default.createElement(_Gapped.Gapped, { wrap: true, verticalAlign: "middle", gap: 10 },
          _react.default.createElement(_SelectPlayground.SelectPlayground, { width: 120, size: size }),
          _react.default.createElement(_Input.Input, { rightIcon: _react.default.createElement(_Card.default, null), placeholder: 'Text value', size: size }),
          _react.default.createElement(_Button.Button, { width: 120, size: size }, "Button"),


          _react.default.createElement(_Button.Button, { icon: _react.default.createElement(_Link.default, null), use: 'link', size: size }, "Button like a link")));};





      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Размеры', theme: _this.theme },
        _react.default.createElement(Group, { size: 'small' }),
        _react.default.createElement(Group, { size: 'medium' }),
        _react.default.createElement(Group, { size: 'large' })));


    };_this.

    renderLinksGroup = function () {
      var propsList = [
      { icon: _react.default.createElement(_Link.default, null), children: 'Enabled' },
      { icon: _react.default.createElement(_Ok.default, null), use: 'success', children: 'Success' },
      { icon: _react.default.createElement(_Error.default, null), use: 'danger', children: 'Danger' },
      { icon: _react.default.createElement(_Trash.default, null), use: 'grayed', children: 'Grayed' },
      { icon: _react.default.createElement(_Trash.default, null), children: 'Disabled', disabled: true }];

      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Ссылки', theme: _this.theme },
        _react.default.createElement(_Gapped.Gapped, { wrap: true, verticalAlign: "middle", gap: 10 },
        (0, _helpers.getComponentsFromPropsList)(_react.default.createElement(_Link2.Link, null), propsList))));



    };_this.

    renderButtonsGroup = function () {
      var propsList = [
      { children: 'Default' },
      { children: 'Primary', use: 'primary' },
      { children: 'Danger', use: 'danger' },
      { children: 'Pay', use: 'pay' },
      { children: 'Disabled', disabled: true },
      { children: 'Back', arrow: 'left', size: 'medium', width: 110 },
      { children: 'Forward', arrow: true, size: 'medium', use: 'primary', width: 110 },
      { children: 'Loading', size: 'medium', loading: true }];


      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Кнопки', theme: _this.theme },
        (0, _helpers.getComponentsFromPropsList)(_react.default.createElement(_Button.Button, { width: 120, size: 'small' }), propsList)));


    };_this.

    renderInputsGroup = function () {
      var propsList = [
      { placeholder: 'Enabled' },
      { placeholder: 'Error', error: true },
      { placeholder: 'Warning', warning: true },
      { placeholder: 'Disabled', disabled: true }];

      var fromProps = (0, _helpers.getComponentsFromPropsList)(_react.default.createElement(_Input.Input, { width: 120 }), propsList);
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Поле ввода', theme: _this.theme },
        _react.default.createElement(_Input.Input, { width: 380, prefix: "https://kontur.ru/search?query=", rightIcon: _react.default.createElement(_Search.default, null) }),
        _react.default.createElement("div", null,
        _react.default.createElement(_Gapped.Gapped, { gap: 10 }, fromProps))));



    };_this.

    renderTokenInputsGroup = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Поле с токеном', theme: _this.theme },
        _react.default.createElement(_TokenInputPlayground.TokenInputPlayground, null)));


    };_this.

    renderOtherInputsGroup = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Прочие поля', theme: _this.theme },
        _react.default.createElement(_CurrencyInputPlayground.CurrencyInputPlayground, null),
        _react.default.createElement(_FxInputPlayground.FxInputPlayground, null),
        _react.default.createElement(_AnotherInputsPlayground.DatePickerPlayground, null)));


    };_this.

    renderSwitchersGroup = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Переключатели', theme: _this.theme },
        _react.default.createElement(_SwitcherPlayground.SwitcherPlayground, null)));


    };_this.

    renderControlsGroup = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Радио, чекбоксы', theme: _this.theme },
        _react.default.createElement(_Gapped.Gapped, { verticalAlign: 'top', gap: 60 },
        _react.default.createElement(_CheckboxPlayground.CheckboxPlayground, null),
        _react.default.createElement(_RadioPlayground.RadioPlayground, null),
        _react.default.createElement(_TogglePlayground.TogglePlayground, null))));



    };_this.

    renderHintsGroup = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Тултип', theme: _this.theme },
        _react.default.createElement(_HintPlayground.HintPlayground, null)));


    };_this.

    renderTooltip = function () {
      var tooltipContent = function tooltipContent() {return (
          _react.default.createElement("div", { className: _Playground.jsStyles.tooltipContent() },
          'Информация об ошибке. Короткий объясняющий текст и ссылка, если нужно'));};


      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Тултип', theme: _this.theme },
        _react.default.createElement(_Tooltip.Tooltip, { render: tooltipContent, pos: "right middle", trigger: 'opened', disableAnimations: true },
        _react.default.createElement(_Link2.Link, { icon: _react.default.createElement(_HelpDot.default, null) }))));



    };_this.

    renderPaging = function () {
      return (
        _react.default.createElement(_ComponentsGroup.ComponentsGroup, { title: 'Пейджинг', theme: _this.theme },
        _react.default.createElement(_PagingPlayground.PagingPlayground, null)));


    };_this.

    renderStickyStopElement = function () {
      return _react.default.createElement("div", { ref: _this.stopEl, style: { height: 50 } });
    };_this.

    getStickyStop = function () {return _this.stopEl.current;};return _this;}var _proto = Playground.prototype;_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var wrapperClassName = (0, _classnames.default)(_Playground.jsStyles.playground(), _Playground.jsStyles.playgroundWrapper(this.theme));return _react.default.createElement("div", { className: wrapperClassName }, _react.default.createElement(_Gapped.Gapped, { vertical: true, gap: 50 }, this.renderTabsGroup(), this.renderSizesGroup(), this.renderLinksGroup(), this.renderButtonsGroup(), this.renderInputsGroup(), this.renderOtherInputsGroup(), this.renderTokenInputsGroup(), this.renderSwitchersGroup(), this.renderControlsGroup()), this.renderStickyStopElement(), _react.default.createElement(_Gapped.Gapped, { vertical: true, gap: 50 }, this.renderHintsGroup(), this.renderTooltip(), this.renderPaging()));};_proto.renderTabs = function renderTabs() {var _cn;var _this$props = this.props,currentThemeType = _this$props.currentThemeType,onThemeChange = _this$props.onThemeChange,onEditLinkClick = _this$props.onEditLinkClick;var tabsOuterWrapperStyle = { background: this.theme.bgDefault };var tabsOuterWrapperClass = (0, _classnames.default)((_cn = {}, _cn[_Playground.jsStyles.tabsWrapper()] = true, _cn[_Playground.jsStyles.stickyTabsWrapper(this.theme)] = useSticky, _cn));return _react.default.createElement("div", { style: tabsOuterWrapperStyle, className: tabsOuterWrapperClass }, _react.default.createElement(_Gapped.Gapped, { gap: 40 }, _react.default.createElement(_Tabs.Tabs, { value: currentThemeType, onValueChange: onThemeChange, vertical: false }, _react.default.createElement("div", { className: _Playground.jsStyles.tabsInnerWrapper(this.theme) }, _react.default.createElement(_Tabs.Tabs.Tab, { id: _constants.ThemeType.Default }, "\u0414\u0435\u0444\u043E\u043B\u0442\u043D\u0430\u044F"), _react.default.createElement(_Tabs.Tabs.Tab, { id: _constants.ThemeType.Flat }, "\u041F\u043B\u043E\u0441\u043A\u0430\u044F"), _react.default.createElement(_Tabs.Tabs.Tab, { id: _constants.ThemeType.Dark }, "\u0422\u0435\u043C\u043D\u0430\u044F"))), _react.default.createElement(_Link2.Link, { onClick: onEditLinkClick }, "\u041D\u0430\u0441\u0442\u0440\u043E\u0438\u0442\u044C \u0442\u0435\u043C\u0443")));};return Playground;}(_react.default.Component);exports.Playground = Playground;
