// @skbkontur/react-ui/cjs/internal/ThemePlayground/ThemeEditor.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.ThemeEditor = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));

var _ThemeFactory = require("../../lib/theming/ThemeFactory");

var _Gapped = require("../../components/Gapped");
var _Loader = require("../../components/Loader");

var _VariableValue = require("./VariableValue");
var _constants = require("./constants");

var _Playground = require("./Playground.styles");var
















ThemeEditor = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(ThemeEditor, _React$Component);function ThemeEditor() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.
    state = {
      groups: [],
      isLoading: true };_this.

    updateTimeout = void 0;_this.





















    renderGroups = function () {var _this$props =
      _this.props,editingTheme = _this$props.editingTheme,currentTheme = _this$props.currentTheme,currentErrors = _this$props.currentErrors,onValueChange = _this$props.onValueChange;
      var keys = _ThemeFactory.ThemeFactory.getKeys(editingTheme);

      return (
        _react.default.createElement(_Gapped.Gapped, { wrap: true, verticalAlign: "middle" },
        _this.state.groups.map(function (i) {return (
            _react.default.createElement(Group, {
              editingTheme: editingTheme,
              currentTheme: currentTheme,
              currentErrors: currentErrors,
              onValueChange: onValueChange,
              title: i.title,
              variables: keys.filter(
              i.isCommon ?
              isCommonVariable.bind(null, _this.state.groups.reduce(prefixesReducer, [])) :
              isGroupVariable.bind(null, i.prefix)),

              key: i.title }));})));




    };return _this;}var _proto = ThemeEditor.prototype;_proto.render = function render() {return this.state.isLoading ? _react.default.createElement("div", { className: _Playground.jsStyles.loaderWrapper() }, _react.default.createElement(_Loader.Loader, { type: "big", active: true, className: _Playground.jsStyles.loader() })) : this.renderGroups();};_proto.componentDidMount = function componentDidMount() {var _this2 = this;this.updateTimeout = window.setTimeout(function () {_this2.setState({ groups: _constants.VARIABLES_GROUPS, isLoading: false });}, 500);};_proto.componentWillUnmount = function componentWillUnmount() {clearTimeout(this.updateTimeout);};return ThemeEditor;}(_react.default.Component);exports.ThemeEditor = ThemeEditor;










var Group = function Group(props) {var
  editingTheme = props.editingTheme,currentTheme = props.currentTheme,currentErrors = props.currentErrors,onValueChange = props.onValueChange,title = props.title,variables = props.variables;

  return variables.length > 0 ?
  _react.default.createElement(_react.default.Fragment, null,
  _react.default.createElement("h2", { className: _Playground.jsStyles.editorGroupHeader(currentTheme) }, title),
  _react.default.createElement(_Gapped.Gapped, { gap: 16, wrap: true, verticalAlign: "middle" },
  variables.map(function (variable) {
    var value = editingTheme[variable];
    var isError = currentErrors[variable];
    return (
      _react.default.createElement(_VariableValue.VariableValue, {
        theme: currentTheme,
        onChange: onValueChange,
        value: value,
        isError: isError || false,
        variable: variable,
        key: variable,
        baseVariables: getBaseVariables(editingTheme, variable) }));


  }))) :


  null;
};

var isGroupVariable = function isGroupVariable(prefix, name) {
  var splitPrefix = prefix.split(' ') || [];

  for (var _iterator = splitPrefix, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {var _ref;if (_isArray) {if (_i >= _iterator.length) break;_ref = _iterator[_i++];} else {_i = _iterator.next();if (_i.done) break;_ref = _i.value;}var item = _ref;
    if (name.startsWith(item.trim())) {
      return true;
    }
  }
  return false;
};
var isCommonVariable = function isCommonVariable(prefixes, name) {
  for (var _iterator2 = prefixes, _isArray2 = Array.isArray(_iterator2), _i2 = 0, _iterator2 = _isArray2 ? _iterator2 : _iterator2[Symbol.iterator]();;) {var _ref2;if (_isArray2) {if (_i2 >= _iterator2.length) break;_ref2 = _iterator2[_i2++];} else {_i2 = _iterator2.next();if (_i2.done) break;_ref2 = _i2.value;}var item = _ref2;
    if (name.startsWith(item.trim())) {
      return false;
    }
  }
  return true;
};
var prefixesReducer = function prefixesReducer(acc, current) {
  var splitPrefix = current.prefix.split(' ');
  return [].concat(acc, splitPrefix);
};
var getBaseVariables = function getBaseVariables(theme, variable) {
  for (; theme != null; theme = Object.getPrototypeOf(theme)) {
    if (Object.prototype.hasOwnProperty.call(theme, variable)) {
      var descriptor = Object.getOwnPropertyDescriptor(theme, variable);

      if (descriptor && typeof descriptor.get !== 'undefined') {
        var getterBody = descriptor.get.toString();
        var variableNameMatchArray = getterBody.match(/this\.(\w+)\b/gm) || [];
        return (variableNameMatchArray || []).map(function (v) {return v.replace(/this\./g, '');});
      }
      break;
    }
  }
  return [];
};
