// @skbkontur/react-ui/components/Input/Input.styles/Input.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject27() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 2px;\n      flex-shrink: 0;\n      cursor: text;\n      z-index: 2;\n    "]);

  _templateObject27 = function _templateObject27() {
    return data;
  };

  return data;
}

function _templateObject26() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-right: 2px;\n      flex-shrink: 0;\n      cursor: text;\n      z-index: 2;\n    "]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _taggedTemplateLiteralLoose(["\n      justify-self: flex-end;\n      margin: 0 0 0 auto;\n      padding-left: 0;\n      padding-right: 10px;\n    "]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _taggedTemplateLiteralLoose(["\n      align-items: center;\n      display: flex;\n      flex-shrink: 0;\n      height: 100%;\n      min-width: 10px;\n      padding-left: 10px;\n\n      &::before {\n        content: '\\a0';\n        display: inline-block;\n        width: 0;\n      }\n    "], ["\n      align-items: center;\n      display: flex;\n      flex-shrink: 0;\n      height: 100%;\n      min-width: 10px;\n      padding-left: 10px;\n\n      &::before {\n        content: '\\\\a0';\n        display: inline-block;\n        width: 0;\n      }\n    "]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-top: ", ";\n      padding-bottom: ", ";\n      line-height: normal !important;\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: ", ";\n      line-height: ", " !important;\n      height: ", ";\n      padding-top: ", ";\n      padding-bottom: ", ";\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-top: ", ";\n      padding-bottom: ", ";\n      line-height: normal !important;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: ", ";\n      line-height: ", " !important;\n      padding-top: ", ";\n      padding-bottom: ", ";\n      height: ", ";\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-top: ", ";\n      padding-bottom: ", ";\n      line-height: normal !important;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      & {\n        font-size: ", ";\n        line-height: ", " !important;\n        padding-top: ", ";\n        padding-bottom: ", ";\n        height: ", ";\n      }\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      animation: ", " 0.15s ease-in;\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n    0% {\n      background-color: ", ";\n    }\n  "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      border-color: ", " !important;\n      box-shadow: none;\n\n      ", ",\n      ", " {\n        cursor: default;\n      }\n      ", " {\n        color: ", ";\n        pointer-events: none;\n        /* fix text color in safari */\n        -webkit-text-fill-color: currentcolor;\n      }\n      ", ":-moz-placeholder {\n        -webkit-text-fill-color: ", ";\n      }\n      ", "::-moz-placeholder {\n        -webkit-text-fill-color: ", ";\n      }\n      ", "::placeholder {\n        -webkit-text-fill-color: ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: none !important;\n      outline: 1px solid ", " !important;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      border-color: ", " !important;\n      box-shadow: 0 0 0 1px ", " !important;\n      z-index: 2;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: none !important;\n      outline: 1px solid ", " !important;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      & {\n        border-color: ", " !important;\n        box-shadow: 0 0 0 1px ", " !important;\n        z-index: 2;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      -webkit-appearance: none;\n      background: transparent;\n      border: 0 none;\n      color: ", ";\n      font: inherit;\n      line-height: inherit;\n      margin: 0;\n      outline: none;\n      padding: 0;\n      text-overflow: clip;\n      white-space: nowrap;\n      width: 100%;\n\n      &:-moz-placeholder {\n        opacity: 1;\n      }\n      &::-moz-placeholder {\n        opacity: 1;\n      }\n      &::-ms-clear {\n        display: none;\n      }\n      &:-moz-placeholder {\n        color: ", ";\n      }\n      &::-moz-placeholder {\n        color: ", ";\n      }\n      &::placeholder {\n        color: ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      -ms-user-select: none;\n      color: ", ";\n      cursor: text;\n      font-size: inherit;\n      height: 100%;\n      left: 0;\n      overflow: hidden;\n      pointer-events: none;\n      position: absolute;\n      top: 0;\n      user-select: none;\n      white-space: nowrap;\n      width: 100%;\n\n      ", " & {\n        color: ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: none;\n      outline: 1px solid ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      border-color: ", ";\n      box-shadow: ", ";\n      outline: none;\n      z-index: 2;\n\n      ", ":-moz-placeholder {\n        color: ", ";\n      }\n      ", "::-moz-placeholder {\n        color: ", ";\n      }\n      ", "::placeholder {\n        color: ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: none;\n      border-color: transparent;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      align-items: center;\n      background-clip: padding-box;\n      background-color: ", ";\n      border: ", " solid ", ";\n      border-top-color: ", ";\n      box-shadow: ", ";\n      box-sizing: border-box;\n      color: ", ";\n      cursor: text;\n      display: inline-flex;\n      outline: none;\n      position: relative;\n      width: 250px;\n\n      & * {\n        box-sizing: border-box;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      align-items: center;\n      display: flex;\n      margin: 0;\n      min-width: 0;\n      overflow: hidden;\n      position: relative;\n      text-overflow: clip;\n      white-space: nowrap;\n      width: 100%;\n\n      &::before {\n        content: '\\A0';\n        display: inline-block;\n        width: 0;\n      }\n    "], ["\n      align-items: center;\n      display: flex;\n      margin: 0;\n      min-width: 0;\n      overflow: hidden;\n      position: relative;\n      text-overflow: clip;\n      white-space: nowrap;\n      width: 100%;\n\n      &::before {\n        content: '\\\\A0';\n        display: inline-block;\n        width: 0;\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, keyframes, memoizeStyle } from "../../../lib/theming/Emotion";
import { shift } from "../../../lib/styles/DimensionFunctions";
import { resetText } from "../../../lib/styles/Mixins";
var styles = {
  wrapper: function wrapper() {
    return css(_templateObject());
  },
  root: function root(t) {
    return css(_templateObject2(), resetText(), t.inputBg, t.inputBorderWidth, t.borderColorGrayLight, t.inputBorderTopColor, t.inputShadow, t.inputColor);
  },
  borderless: function borderless() {
    return css(_templateObject3());
  },
  useDefaultColor: function useDefaultColor(t) {
    return css(_templateObject4(), t.inputIconColor);
  },
  focus: function focus(t) {
    return css(_templateObject5(), t.borderColorFocus, t.inputFocusShadow, cssName(styles.input(t)), t.placeholderColorLight, cssName(styles.input(t)), t.placeholderColorLight, cssName(styles.input(t)), t.placeholderColorLight);
  },
  focusFallback: function focusFallback(t) {
    return css(_templateObject6(), t.inputFocusOutline);
  },
  placeholder: function placeholder(t) {
    return css(_templateObject7(), t.placeholderColor, cssName(styles.focus(t)), t.placeholderColorLight);
  },
  input: function input(t) {
    return css(_templateObject8(), t.inputTextColor, t.placeholderColor, t.placeholderColor, t.placeholderColor);
  },
  warning: function warning(t) {
    return css(_templateObject9(), t.borderColorWarning, t.borderColorWarning);
  },
  warningFallback: function warningFallback(t) {
    return css(_templateObject10(), t.borderColorWarning);
  },
  error: function error(t) {
    return css(_templateObject11(), t.borderColorError, t.borderColorError);
  },
  errorFallback: function errorFallback(t) {
    return css(_templateObject12(), t.borderColorError);
  },
  disabled: function disabled(t) {
    return css(_templateObject13(), t.inputDisabledBg, t.inputDisabledBorderColor, cssName(styles.leftIcon()), cssName(styles.rightIcon()), cssName(styles.input(t)), t.textColorDisabled, cssName(styles.input(t)), t.placeholderColor, cssName(styles.input(t)), t.placeholderColor, cssName(styles.input(t)), t.placeholderColor);
  },
  blink: function blink(t) {
    var blinkAnimation = keyframes(_templateObject14(), t.blinkColor);
    return css(_templateObject15(), blinkAnimation);
  },
  sizeSmall: function sizeSmall(t) {
    return css(_templateObject16(), t.inputFontSizeSmall, t.controlLineHeightSmall, t.controlPaddingYSmall, t.controlPaddingYSmall, t.controlHeightSmall);
  },
  sizeSmallFallback: function sizeSmallFallback(t) {
    return css(_templateObject17(), shift(t.controlPaddingYSmall, '-1'), shift(t.controlPaddingYSmall, '1'));
  },
  sizeMedium: function sizeMedium(t) {
    return css(_templateObject18(), t.inputFontSizeMedium, t.controlLineHeightMedium, t.controlPaddingYMedium, t.controlPaddingYMedium, t.controlHeightMedium);
  },
  sizeMediumFallback: function sizeMediumFallback(t) {
    return css(_templateObject19(), shift(t.controlPaddingYMedium, '-1'), shift(t.controlPaddingYMedium, '1'));
  },
  sizeLarge: function sizeLarge(t) {
    return css(_templateObject20(), t.inputFontSizeLarge, t.controlLineHeightLarge, t.controlHeightLarge, shift(t.controlPaddingYLarge, '-1'), shift(t.controlPaddingYLarge, '1'));
  },
  sizeLargeFallback: function sizeLargeFallback(t) {
    return css(_templateObject21(), shift(t.controlPaddingYLarge, '-2'), shift(t.controlPaddingYLarge, '2'));
  },
  prefix: function prefix(t) {
    return css(_templateObject22(), t.placeholderColor);
  },
  suffix: function suffix(t) {
    return css(_templateObject23(), t.placeholderColor);
  },
  sideContainer: function sideContainer() {
    return css(_templateObject24());
  },
  rightContainer: function rightContainer() {
    return css(_templateObject25());
  },
  leftIcon: function leftIcon() {
    return css(_templateObject26());
  },
  rightIcon: function rightIcon() {
    return css(_templateObject27());
  }
};
export var jsStyles = memoizeStyle(styles);
