// @skbkontur/react-ui/internal/ThemeShowcase/ThemeShowcase.styles/ThemeShowcase.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #b77daa;\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #ac4142;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #7f9a44;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #d28445;\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 16px;\n      margin: 20px 0;\n      background-color: #fff0bc;\n      border-radius: 3px;\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      text-decoration: line-through;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      box-sizing: border-box;\n      width: 16px;\n      height: 16px;\n      border: 1px solid transparent;\n      margin-right: 4px;\n      vertical-align: -3px;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #b77daa;\n      font-weight: 400;\n      cursor: pointer;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      white-space: pre;\n      color: #333;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #7f9a44;\n      font-weight: 400;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: #fff0bc;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 0 !important;\n      height: 0 !important;\n      font-size: 0 !important;\n      line-height: 0 !important;\n      border: 0 none !important;\n      border-top: 1px solid #e8e8e8 !important;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: red;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 0 !important;\n      height: 0 !important;\n\n      ", " tbody tr:not(&) > td {\n        border-top: 0 none;\n      }\n\n      ", " tbody tr&:hover ~ tr:not(&) {\n        background-color: #f8f8f8;\n      }\n\n      ", " tbody tr&:hover ~ & ~ tr {\n        background-color: transparent;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      width: 880px;\n      border-collapse: collapse;\n\n      & tbody tr:hover {\n        background-color: #f8f8f8;\n      }\n\n      & th {\n        text-align: left;\n        padding-left: 5px;\n      }\n\n      & td {\n        border-top: 1px solid #e8e8e8;\n        border-bottom: 1px solid #e8e8e8;\n        vertical-align: top;\n        padding: 10px 5px;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      border-bottom: 1px solid #e8e8e8;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 24px 0 16px 5px;\n      margin: 0;\n      background-color: #fff;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      box-sizing: border-box;\n      height: 40px;\n      padding: 5px;\n      background-color: #fff;\n\n      &::after {\n        content: attr(data-perf-info);\n        position: absolute;\n        top: 11px;\n        right: 20px;\n        color: #808080;\n        height: 20px;\n        line-height: 20px;\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  searchBar: function searchBar() {
    return css(_templateObject());
  },
  heading: function heading() {
    return css(_templateObject2());
  },
  headingSticky: function headingSticky() {
    return css(_templateObject3());
  },
  table: function table() {
    return css(_templateObject4());
  },
  invisibleRow: function invisibleRow() {
    return css(_templateObject5(), cssName(styles.table()), cssName(styles.table()), cssName(styles.table()));
  },
  suspiciousRow: function suspiciousRow() {
    return css(_templateObject6());
  },
  //noinspection CssReplaceWithShorthandSafely
  invisibleCell: function invisibleCell() {
    return css(_templateObject7());
  },
  suspiciousCell: function suspiciousCell() {
    return css(_templateObject8());
  },
  elementName: function elementName() {
    return css(_templateObject9());
  },
  relativeCss: function relativeCss() {
    return css(_templateObject10());
  },
  variableName: function variableName() {
    return css(_templateObject11());
  },
  colorExample: function colorExample() {
    return css(_templateObject12());
  },
  undefined: function undefined() {
    return css(_templateObject13());
  },
  unusedVariablesWarning: function unusedVariablesWarning() {
    return css(_templateObject14());
  },
  cssClassObject: function cssClassObject() {
    return css(_templateObject15());
  },
  cssClassName: function cssClassName() {
    return css(_templateObject16());
  },
  cssThemeObject: function cssThemeObject() {
    return css(_templateObject17());
  },
  cssThemeProp: function cssThemeProp() {
    return css(_templateObject18());
  }
};
export var jsStyles = memoizeStyle(styles);
