// @skbkontur/react-ui/cjs/internal/DateSelect/DateSelect.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");exports.__esModule = true;exports.DateSelect = void 0;var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _classnames = _interopRequireDefault(require("classnames"));

var _identifiers = require("../../lib/events/keyboard/identifiers");
var _locale = require("../../components/DatePicker/locale");
var _decorators = require("../../lib/locale/decorators");
var _RenderLayer = require("../RenderLayer");
var _DropdownContainer = require("../DropdownContainer");
var LayoutEvents = _interopRequireWildcard(require("../../lib/LayoutEvents"));


var _ThemeContext = require("../../lib/theming/ThemeContext");
var _px = require("../icons/16px");

var _DateSelect = require("./DateSelect.styles");var _dec, _class, _class2, _temp;

var itemHeight = 24;
var visibleYearsCount = 11;
var itemsToMoveCount = -5;
var monthsCount = 12;
var defaultMinYear = 1900;
var defaultMaxYear = 2100;var























DateSelect = (_dec = (0, _decorators.locale)('DatePicker', _locale.DatePickerLocaleHelper), _dec(_class = (_temp = _class2 = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(DateSelect, _React$Component);function DateSelect() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.

























    state = {
      botCapped: false,
      current: 0,
      height: 0,
      opened: false,
      pos: 0,
      top: 0,
      topCapped: false,
      nodeTop: Infinity };_this.


    theme = void 0;_this.
    locale = void 0;_this.
    root = null;_this.
    itemsContainer = null;_this.
    listener = void 0;_this.
    timeout = void 0;_this.
    longClickTimer = 0;_this.
    setPositionRepeatTimer = 0;_this.
    yearStep = 3;_this.






























    open = function () {
      if (_this.props.disabled) {
        return;
      }

      if (_this.state.opened) {
        return;
      }

      _this.setPosition(0);
      _this.setState({
        opened: true,
        current: 0 });

    };_this.




    close = function () {
      if (!_this.state.opened) {
        return;
      }

      _this.setState({ opened: false });
    };_this.








































    refRoot = function (element) {
      _this.root = element;
    };_this.

    setNodeTop = function () {
      var root = _this.root;
      if (!root) {
        return;
      }
      if (_this.timeout) {
        clearTimeout(_this.timeout);
      }
      _this.timeout = setTimeout(function () {return (
          _this.setState({
            nodeTop: root.getBoundingClientRect().top }));});


    };_this.































































































































    refItemsContainer = function (element) {
      if (!_this.itemsContainer && element) {
        element.addEventListener('wheel', _this.handleWheel, { passive: false });
      }
      if (_this.itemsContainer && !element) {
        _this.itemsContainer.removeEventListener('wheel', _this.handleWheel);
      }
      _this.itemsContainer = element;
    };_this.

    handleLongClickUp = function (event) {
      event.preventDefault();
      _this.longClickTimer = window.setTimeout(function () {
        _this.setPositionRepeatTimer = window.setInterval(function () {return _this.setPosition(_this.state.pos - itemHeight);}, 100);
      }, 200);
    };_this.

    handleLongClickDown = function (event) {
      event.preventDefault();
      _this.longClickTimer = window.setTimeout(function () {
        _this.setPositionRepeatTimer = window.setInterval(function () {return _this.setPosition(_this.state.pos + itemHeight);}, 100);
      }, 200);
    };_this.

    handleLongClickStop = function () {
      clearTimeout(_this.longClickTimer);
      clearTimeout(_this.setPositionRepeatTimer);
    };_this.

    getAnchor = function () {return _this.root;};_this.

    handleWheel = function (event) {
      if (!(event instanceof WheelEvent)) {
        return;
      }
      event.preventDefault();
      event.stopPropagation();

      var deltaY = event.deltaY;
      if (event.deltaMode === 1) {
        deltaY *= itemHeight;
      } else if (event.deltaMode === 2) {
        deltaY *= itemHeight * 4;
      }
      var pos = _this.state.pos + deltaY;
      _this.setPosition(pos);
    };_this.

    handleItemClick = function (shift) {
      return function (e) {
        var value = _this.props.value + shift;
        if (_this.props.onValueChange) {
          _this.props.onValueChange(value);
        }
        _this.setState({ opened: false });
      };
    };_this.

    handleKey = function (e) {
      if (_this.state.opened && (0, _identifiers.isKeyEscape)(e)) {
        e.preventDefault();
        _this.close();
        e.stopPropagation();
      }
    };_this.

    handleUp = function (event) {
      event.preventDefault();
      _this.setPosition(_this.state.pos - itemHeight * _this.yearStep);
    };_this.

    handleDown = function (event) {
      event.preventDefault();
      _this.setPosition(_this.state.pos + itemHeight * _this.yearStep);
    };return _this;}var _proto = DateSelect.prototype;_proto.UNSAFE_componentWillReceiveProps = function UNSAFE_componentWillReceiveProps() {this.setNodeTop();};_proto.componentDidMount = function componentDidMount() {this.listener = LayoutEvents.addListener(this.setNodeTop);this.setNodeTop();window.addEventListener('keydown', this.handleKey);};_proto.componentWillUnmount = function componentWillUnmount() {if (this.listener) {this.listener.remove();}if (this.timeout) {clearTimeout(this.timeout);}if (this.longClickTimer) {clearTimeout(this.longClickTimer);}if (this.setPositionRepeatTimer) {clearTimeout(this.setPositionRepeatTimer);}window.removeEventListener('keydown', this.handleKey);} /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        * @public
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        */;_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var _cn, _cn2;var _this$props = this.props,width = _this$props.width,disabled = _this$props.disabled;var rootProps = { className: (0, _classnames.default)((_cn = {}, _cn[_DateSelect.jsStyles.root(this.theme)] = true, _cn[_DateSelect.jsStyles.disabled()] = Boolean(disabled), _cn)), style: { width: width }, ref: this.refRoot };return _react.default.createElement("span", rootProps, _react.default.createElement("div", { "data-tid": "DateSelect__caption", className: _DateSelect.jsStyles.caption(), onClick: this.open }, this.getItem(0), _react.default.createElement("div", { className: (0, _classnames.default)((_cn2 = {}, _cn2[_DateSelect.jsStyles.arrow(this.theme)] = true, _cn2[_DateSelect.jsStyles.arrowDisabled()] = Boolean(disabled), _cn2)) }, _react.default.createElement(_px.ArrowTriangleUpDownIcon, { size: 12 }))), this.state.opened && this.renderMenu());};_proto.disableItems = function disableItems(index) {var value = this.props.value + index;if (this.props.maxValue != null && this.props.minValue != null) {return value > this.props.maxValue || value < this.props.minValue;}if (this.props.minValue != null) {return value < this.props.minValue;}if (this.props.maxValue != null) {return value > this.props.maxValue;}};_proto.renderMenu = function renderMenu() {var _this3 = this,_cn3;var _this$state = this.state,top = _this$state.top,height = _this$state.height,nodeTop = _this$state.nodeTop;var shift = this.state.pos % itemHeight;if (shift < 0) {shift += itemHeight;}var from = (this.state.pos - shift + top) / itemHeight;var to = from + Math.ceil((height + shift) / itemHeight);var items = [];var _loop = function _loop(i) {var _cn4;var disableItems = _this3.disableItems(i) || false;var className = (0, _classnames.default)((_cn4 = {}, _cn4[_DateSelect.jsStyles.menuItem(_this3.theme)] = true, _cn4[_DateSelect.jsStyles.menuItemSelected(_this3.theme)] = i === 0, _cn4[_DateSelect.jsStyles.menuItemActive(_this3.theme)] = i === _this3.state.current, _cn4[_DateSelect.jsStyles.menuItemDisabled(_this3.theme)] = disableItems, _cn4));var clickHandler = { onMouseDown: preventDefault, onClick: _this3.handleItemClick(i) };items.push(_react.default.createElement("div", (0, _extends2.default)({ "data-tid": "DateSelect__menuItem", "data-prop-disabled": disableItems, key: i, className: className, onMouseEnter: function onMouseEnter() {return _this3.setState({ current: i });}, onMouseLeave: function onMouseLeave() {return _this3.setState({ current: null });} }, clickHandler), _this3.getItem(i)));};for (var i = from; i < to; ++i) {_loop(i);}var style = { top: top - 5, left: 0, right: 0 };var shiftStyle = { position: 'relative', top: -shift };var holderClass = (0, _classnames.default)((_cn3 = {}, _cn3[_DateSelect.jsStyles.menuHolder(this.theme)] = true, _cn3[_DateSelect.jsStyles.isTopCapped()] = this.state.topCapped, _cn3[_DateSelect.jsStyles.isBotCapped()] = this.state.botCapped, _cn3));var dropdownOffset = -itemHeight;if (nodeTop < -top) {var overflowOffsetDelta = this.state.topCapped ? 6 : 17;dropdownOffset -= nodeTop + top - overflowOffsetDelta;}return _react.default.createElement(_RenderLayer.RenderLayer, { onClickOutside: this.close, onFocusOutside: this.close, active: true }, _react.default.createElement("div", null, _react.default.createElement(_DropdownContainer.DropdownContainer, { getParent: this.getAnchor, offsetY: dropdownOffset, offsetX: -10 }, _react.default.createElement("div", { className: holderClass, style: style }, !this.state.topCapped && _react.default.createElement("div", { className: _DateSelect.jsStyles.menuUp(this.theme), onClick: this.handleUp, onMouseDown: this.handleLongClickUp, onMouseUp: this.handleLongClickStop, onMouseLeave: this.handleLongClickStop, onTouchStart: this.handleLongClickUp, onTouchEnd: this.handleLongClickStop }, _react.default.createElement("span", null, _react.default.createElement(_px.ArrowTriangleUpIcon, null))), _react.default.createElement("div", { className: _DateSelect.jsStyles.itemsHolder(), style: { height: height } }, _react.default.createElement("div", { ref: this.refItemsContainer, style: shiftStyle }, items)), !this.state.botCapped && _react.default.createElement("div", { className: _DateSelect.jsStyles.menuDown(this.theme), onClick: this.handleDown, onMouseDown: this.handleLongClickDown, onMouseUp: this.handleLongClickStop, onMouseLeave: this.handleLongClickStop, onTouchStart: this.handleLongClickDown, onTouchEnd: this.handleLongClickStop }, _react.default.createElement("span", null, _react.default.createElement(_px.ArrowTriangleDownIcon, null)))))));};_proto.getItem = function getItem(index) {
    var value = this.props.value + index;
    if (this.props.type === 'month') {
      return this.locale.months[value];
    }
    return value;
  };_proto.

  setPosition = function setPosition(pos) {
    var top = itemsToMoveCount * itemHeight;
    var height = visibleYearsCount * itemHeight;
    if (this.props.type === 'month') {
      top = -this.props.value * itemHeight;
      height = monthsCount * itemHeight;
    }

    var minPos = this.getMinPos() - top;
    var maxPos = this.getMaxPos() - top - height + itemHeight;
    if (minPos >= pos) {
      pos = minPos;
    }
    if (maxPos <= pos) {
      pos = maxPos;
    }
    var topCapped = pos <= minPos;
    var botCapped = pos >= maxPos;

    this.setState({ pos: pos, top: top, height: height, topCapped: topCapped, botCapped: botCapped });
  };_proto.

  getMinPos = function getMinPos() {
    if (this.props.type === 'month') {
      return -this.props.value * itemHeight;
    } else if (this.props.type === 'year') {
      return ((this.props.minValue || defaultMinYear) - this.props.value) * itemHeight;
    }
    return -Infinity; // Be defensive.
  };_proto.

  getMaxPos = function getMaxPos() {
    if (this.props.type === 'month') {
      return (visibleYearsCount - this.props.value) * itemHeight;
    } else if (this.props.type === 'year') {
      return ((this.props.maxValue || defaultMaxYear) - this.props.value) * itemHeight;
    }
    return Infinity; // Be defensive.
  };return DateSelect;}(_react.default.Component), _class2.__KONTUR_REACT_UI__ = 'DateSelect', _class2.propTypes = { disabled: _propTypes.default.bool, type: _propTypes.default.string, value: _propTypes.default.number.isRequired, width: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]), onValueChange: _propTypes.default.func, minValue: _propTypes.default.number, maxValue: _propTypes.default.number }, _class2.defaultProps = { type: 'year', minMonth: 0, maxMonth: 11, width: 'auto' }, _temp)) || _class);exports.DateSelect = DateSelect;


function preventDefault(e) {
  e.preventDefault();
}
