// @skbkontur/react-ui/cjs/components/Token/Token.styles.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsTokenColors = exports.jsStyles = void 0;var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");

var ColorFunctions = _interopRequireWildcard(require("../../lib/styles/ColorFunctions"));function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        box-shadow: 0 0 0 2px ", ", inset 0 0 0 1px ", ";\n      "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        box-shadow: 0 0 0 2px ", ", inset 0 0 0 1px ", ";\n      "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        background-color: ", ";\n        color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n\n        & ", " {\n          fill: ", ";\n          opacity: 1;\n        }\n      "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        background-color: ", ";\n        color: ", ";\n        box-shadow: 0 0 0 1px ", ", inset 0 0 0 1px ", ";\n\n        ", "\n\n        & ", ":hover {\n          color: ", ";\n        }\n      "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        box-shadow: 0 0 0 2px ", ", inset 0 0 0 1px ", ";\n      "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        box-shadow: 0 0 0 2px ", ", inset 0 0 0 1px ", ";\n      "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 1em;\n      width: 1em;\n      flex-shrink: 0;\n      padding: 2px;\n      box-sizing: border-box;\n      margin-left: 4px;\n      transition: none;\n      fill: currentColor;\n      opacity: 0.5;\n      line-height: 0;\n\n      &:hover {\n        opacity: 1;\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      padding-bottom: 1px;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: none !important;\n      margin: 2px;\n      padding: 1px 4px;\n      user-select: text;\n      cursor: text;\n      color: ", " !important;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-flex;\n      align-items: center;\n      border-radius: 1px;\n      padding: 0 4px;\n      line-height: 1.5;\n      font-size: 14px;\n      margin: 3px;\n      min-width: 0;\n      word-break: break-word;\n      user-select: none;\n\n      &:hover {\n        cursor: pointer;\n      }\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  token: function token() {
    return (0, _Emotion.css)(_templateObject());















  },

  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject2(),





    t.textColorDisabled);

  },

  text: function text() {
    return (0, _Emotion.css)(_templateObject3());



  },

  removeIcon: function removeIcon() {
    return (0, _Emotion.css)(_templateObject4());















  } };






















var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;

var jsTokenColors = [
{ name: 'defaultIdle', color: function color(t) {return t.grayXLight;} },
{ name: 'defaultActive', color: function color(t) {return t.brand;} },
{ name: 'grayIdle', color: function color(t) {return t.grayXLight;} },
{ name: 'grayActive', color: function color(t) {return t.grayDark;} },
{ name: 'blueIdle', color: function color(t) {return t.blueLight;} },
{ name: 'blueActive', color: function color(t) {return t.blueDark;} },
{ name: 'greenIdle', color: function color(t) {return t.greenXxLight;} },
{ name: 'greenActive', color: function color(t) {return t.greenDark;} },
{ name: 'yellowIdle', color: function color(t) {return t.yellowXxLight;} },
{ name: 'yellowActive', color: function color(t) {return t.yellowDark;} },
{ name: 'redIdle', color: function color(t) {return t.redXxLight;} },
{ name: 'redActive', color: function color(t) {return t.redDark;} },
{ name: 'white', color: function color(t) {return t.white;} },
{ name: 'black', color: function color(t) {return t.black;} }].
reduce(
function (colors, _ref) {var _extends2;var name = _ref.name,color = _ref.color;return (0, _extends3.default)({},
  colors, (_extends2 = {}, _extends2[
  name] = function (t, v) {
    var warning = (0, _Emotion.css)(_templateObject5(),
    t.borderColorWarning, color(t));

    var error = (0, _Emotion.css)(_templateObject6(),
    t.borderColorError, color(t));

    var vStyle = v === 'error' ? error : v === 'warning' ? warning : '';

    return (0, _Emotion.css)(_templateObject7(),
    color(t),
    ColorFunctions.contrast(color(t)),
    ColorFunctions.darken(color(t), '5%'), color(t),

    vStyle,

    (0, _Emotion.cssName)(jsStyles.removeIcon()),
    ColorFunctions.contrast(color(t)));


  }, _extends2));},

{
  defaultDisabled: function defaultDisabled(t) {
    return (0, _Emotion.css)(_templateObject8(),
    t.tokenDisabledBg,
    ColorFunctions.contrast(t.tokenDisabledBg),
    t.tokenDisabledBg,

    (0, _Emotion.cssName)(jsStyles.removeIcon()),
    t.textColorDisabled);



  },
  defaultDisabledWarning: function defaultDisabledWarning(t) {
    return (0, _Emotion.css)(_templateObject9(),
    t.borderColorWarning, t.tokenDisabledBg);

  },
  defaultDisabledError: function defaultDisabledError(t) {
    return (0, _Emotion.css)(_templateObject10(),
    t.borderColorError, t.tokenDisabledBg);

  } });exports.jsTokenColors = jsTokenColors;
