// @skbkontur/react-ui/cjs/components/Center/Center.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      text-align: left;\n      vertical-align: middle;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      height: 100%;\n      vertical-align: middle;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      text-align: right;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      text-align: left;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      text-align: center;\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject());



  },

  rootAlignLeft: function rootAlignLeft() {
    return (0, _Emotion.css)(_templateObject2());


  },

  rootAlignRight: function rootAlignRight() {
    return (0, _Emotion.css)(_templateObject3());


  },

  spring: function spring() {
    return (0, _Emotion.css)(_templateObject4());




  },

  container: function container() {
    return (0, _Emotion.css)(_templateObject5());




  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
