// @skbkontur/react-ui/cjs/components/Fias/Fias.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.Fias = void 0;var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _lodash = _interopRequireDefault(require("lodash.isequal"));
var _warning = _interopRequireDefault(require("warning"));
var _classnames = _interopRequireDefault(require("classnames"));

var _Link = require("../Link");
var _decorators = require("../../lib/locale/decorators");
var _ThemeContext = require("../../lib/theming/ThemeContext");

var _px = require("../../internal/icons/16px");
var _locale = require("../../lib/locale");

var _locale2 = require("./locale");
var _types = require("./types");







var _FiasModal = require("./FiasModal");
var _FiasForm = require("./Form/FiasForm");
var _FiasAPI = require("./api/FiasAPI");
var _FiasAddress = require("./models/FiasAddress");
var _FiasLogger = require("./logger/FiasLogger");
var _Fias = require("./Fias.styles");var _dec, _class, _class2, _temp;



























































































function deepMerge(dst) {for (var _len = arguments.length, src = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {src[_key - 1] = arguments[_key];}
  src.forEach(function (obj) {
    for (var k in obj) {
      if (dst[k] != null && typeof obj[k] === 'object') {
        dst[k] = deepMerge(dst[k], obj[k]);
      } else {
        dst[k] = obj[k];
      }
    }
  });
  return dst;
}

/**
   * @deprecated Контур-специфичный компонент, будет удален в 3.0.0, перенесен в `@skbkontur/react-ui-addons` смотри [миграцию](https://github.com/skbkontur/retail-ui/blob/master/packages/react-ui/MIGRATION.md)
   */var


Fias = (_dec = (0, _decorators.locale)('Fias', _locale2.FiasLocaleHelper), _dec(_class = (_temp = _class2 = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(Fias, _React$Component);



























  function Fias(props) {var _this;
    _this = _React$Component.call(this, props) || this;_this.state = { opened: false, address: new _FiasAddress.FiasAddress(), locale: _locale2.FiasLocaleHelper.get(), fieldsSettings: _this.fieldsSettings };_this.theme = void 0;_this.api = _this.props.api || new _FiasAPI.FiasAPI(_this.props.baseUrl, _this.props.version);_this.form = null;_this.locale = void 0;_this.
































    componentDidMount = function () {
      _this.updateLocale();
      _this.init();
    };_this.

    componentDidUpdate = function (prevProps, prevState) {
      if (!(0, _lodash.default)(prevProps.value, _this.props.value)) {
        _this.updateAddress();
      }
      if (!(0, _lodash.default)(prevState.locale, _this.locale)) {
        _this.updateLocale(_this.locale);
      }
      if (!(0, _lodash.default)(prevProps.fieldsSettings, _this.props.fieldsSettings)) {
        _this.updateFieldsSettings();
      }
    };_this.







































































    init = /*#__PURE__*/(0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee() {var address;return _regenerator.default.wrap(function _callee$(_context) {while (1) {switch (_context.prev = _context.next) {case 0:_context.next = 2;return (
                _this.updateAddress());case 2:address = _context.sent;
              if (_this.props.onInit) {
                _this.props.onInit(address.getValue(_this.isFieldVisible(_types.FiasExtraFields.postalcode)));
              }case 4:case "end":return _context.stop();}}}, _callee);}));_this.


    updateAddress = /*#__PURE__*/(0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee2() {var address;return _regenerator.default.wrap(function _callee2$(_context2) {while (1) {switch (_context2.prev = _context2.next) {case 0:_context2.next = 2;return (
                _FiasAddress.FiasAddress.getAddress(_this.api, _this.props.value, _this.state.fieldsSettings));case 2:address = _context2.sent;
              _this.setState({
                address: address });return _context2.abrupt("return",

              address);case 5:case "end":return _context2.stop();}}}, _callee2);}));_this.


    updateLocale = function (nextLocale) {if (nextLocale === void 0) {nextLocale = _locale2.FiasLocaleHelper.get();}
      _this.setState({ locale: nextLocale });
    };_this.

    updateFieldsSettings = function () {
      _this.setState({
        fieldsSettings: _this.fieldsSettings });

    };_this.

    handleOpen = function () {
      _this.setState({ opened: true });
    };_this.

    handleClose = function () {
      _this.setState({ opened: false });
      var onClose = _this.props.onClose;
      if (onClose) {
        onClose();
      }
    };_this.

    handleSave = /*#__PURE__*/(0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee3() {var address;return _regenerator.default.wrap(function _callee3$(_context3) {while (1) {switch (_context3.prev = _context3.next) {case 0:if (!
              _this.form) {_context3.next = 7;break;}_context3.next = 3;return (
                _this.form.submit());case 3:address = _context3.sent;if (!(
              address.hasErrors && _this.props.allowNotVerified === false)) {_context3.next = 6;break;}return _context3.abrupt("return");case 6:


              _this.handleChange(address);case 7:

              _this.handleClose();case 8:case "end":return _context3.stop();}}}, _callee3);}));_this.


    handleChange = function (address) {
      if (_this.props.onValueChange) {
        _this.props.onValueChange(address.getValue(_this.isFieldVisible(_types.FiasExtraFields.postalcode)));
      }
    };_this.

    refForm = function (element) {
      _this.form = element;
    };(0, _warning.default)(true, "Fias has been deprecated, use Fias from @skbkontur/react-ui-addons instead, see [migration](https://github.com/skbkontur/retail-ui/blob/master/packages/react-ui/MIGRATION.md)");if (!props.baseUrl && !props.api) {_FiasLogger.FiasLogger.log(_FiasLogger.FiasLogger.warnings.baseUrlOrApiIsRequired);}return _this;}var _proto = Fias.prototype;_proto.isFieldVisible = function isFieldVisible(field) {return _FiasAddress.FiasAddress.isFieldVisible(field, this.state.fieldsSettings);};_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var _cn;var _this$props = this.props,showAddressText = _this$props.showAddressText,label = _this$props.label,icon = _this$props.icon,error = _this$props.error,warning = _this$props.warning,feedback = _this$props.feedback;var _this$state = this.state,opened = _this$state.opened,address = _this$state.address;var linkText = label || (address.isEmpty ? this.state.locale.addressFill : this.state.locale.addressEdit);var validation = (error || warning) && feedback ? _react.default.createElement("span", { className: (0, _classnames.default)((_cn = {}, _cn[_Fias.jsStyles.error(this.theme)] = Boolean(error), _cn[_Fias.jsStyles.warning(this.theme)] = Boolean(warning), _cn)) }, feedback) : null;return _react.default.createElement(_locale.LocaleContext.Provider, { value: { locale: { Fias: this.state.locale } } }, _react.default.createElement("div", null, showAddressText && _react.default.createElement("span", null, address.getFullText(this.isFieldVisible(_types.FiasExtraFields.postalcode))), !this.props.readonly && _react.default.createElement("div", null, _react.default.createElement(_Link.Link, { icon: icon, onClick: this.handleOpen }, linkText)), validation, opened && this.renderModal()));};_proto.renderModal = function renderModal() {var _this$state2 = this.state,address = _this$state2.address,fieldsSettings = _this$state2.fieldsSettings;var _this$props2 = this.props,search = _this$props2.search,limit = _this$props2.limit,formValidation = _this$props2.formValidation,countrySelector = _this$props2.countrySelector;return _react.default.createElement(_FiasModal.FiasModal, { onClose: this.handleClose, onSave: this.handleSave }, _react.default.createElement(_FiasForm.FiasForm, { ref: this.refForm, address: address, api: this.api, search: search, limit: limit, validationLevel: formValidation, fieldsSettings: fieldsSettings, countrySelector: countrySelector }));};(0, _createClass2.default)(Fias, [{ key: "fieldsSettings", get: function get() {var _deepMerge;var _this$props3 = this.props,userSettings = _this$props3.fieldsSettings,countrySelector = _this$props3.countrySelector; // TODO: implement deepMerge with clone
      var defaultSettings = _FiasAddress.FiasAddress.ALL_FIELDS.reduce(function (settings, field) {var _extends2;return (0, _extends3.default)({}, settings, (_extends2 = {}, _extends2[field] = { visible: true }, _extends2));}, {});return deepMerge(defaultSettings, (_deepMerge = {}, _deepMerge[_types.FiasExtraFields.postalcode] = { visible: Boolean(countrySelector) }, _deepMerge), userSettings);} }]);return Fias;}(_react.default.Component), _class2.__KONTUR_REACT_UI__ = 'Fias', _class2.defaultProps = { showAddressText: true, error: false, warning: false, readonly: false, search: false, icon: _react.default.createElement(_px.EditIcon, null), allowNotVerified: true, fieldsSettings: {}, countrySelector: false }, _temp)) || _class);exports.Fias = Fias;
