// @skbkontur/react-ui/cjs/components/Fias/types.js
"use strict";exports.__esModule = true;exports.FiasFormValidation = exports.FiasExtraFields = exports.FiasFields = exports.FiasLiveStatuses = exports.FiasStructureStatuses = exports.FiasEstateStatuses = void 0;var




































FiasEstateStatuses;exports.FiasEstateStatuses = FiasEstateStatuses;(function (FiasEstateStatuses) {FiasEstateStatuses["Hold"] = "Hold";FiasEstateStatuses["House"] = "House";FiasEstateStatuses["HouseHold"] = "HouseHold";FiasEstateStatuses["None"] = "None";})(FiasEstateStatuses || (exports.FiasEstateStatuses = FiasEstateStatuses = {}));var






FiasStructureStatuses;exports.FiasStructureStatuses = FiasStructureStatuses;(function (FiasStructureStatuses) {FiasStructureStatuses["Structure"] = "Structure";FiasStructureStatuses["Construction"] = "Construction";FiasStructureStatuses["Liter"] = "Liter";FiasStructureStatuses["None"] = "None";})(FiasStructureStatuses || (exports.FiasStructureStatuses = FiasStructureStatuses = {}));var






FiasLiveStatuses;exports.FiasLiveStatuses = FiasLiveStatuses;(function (FiasLiveStatuses) {FiasLiveStatuses["active"] = "active";FiasLiveStatuses["inactive"] = "inactive";})(FiasLiveStatuses || (exports.FiasLiveStatuses = FiasLiveStatuses = {}));var






FiasFields;exports.FiasFields = FiasFields;(function (FiasFields) {FiasFields["region"] = "region";FiasFields["district"] = "district";FiasFields["city"] = "city";FiasFields["intracityarea"] = "intracityarea";FiasFields["settlement"] = "settlement";FiasFields["planningstructure"] = "planningstructure";FiasFields["street"] = "street";FiasFields["stead"] = "stead";FiasFields["house"] = "house";FiasFields["room"] = "room";})(FiasFields || (exports.FiasFields = FiasFields = {}));var












FiasExtraFields;exports.FiasExtraFields = FiasExtraFields;(function (FiasExtraFields) {FiasExtraFields["postalcode"] = "postalcode";})(FiasExtraFields || (exports.FiasExtraFields = FiasExtraFields = {}));var




































FiasFormValidation;exports.FiasFormValidation = FiasFormValidation;(function (FiasFormValidation) {FiasFormValidation["Error"] = "Error";FiasFormValidation["Warning"] = "Warning";FiasFormValidation["None"] = "None";})(FiasFormValidation || (exports.FiasFormValidation = FiasFormValidation = {}));
