// @skbkontur/react-ui/components/Loader/Loader.styles/Loader.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n      ", ";\n\n      &::before {\n        content: ' ';\n        display: inline-block;\n        height: 100%;\n        min-height: 100%;\n        vertical-align: middle;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      position: fixed;\n\n      &::before {\n        content: ' ';\n        display: inline-block;\n        height: 100%;\n        min-height: 100%;\n        vertical-align: middle;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      display: block;\n      margin: auto;\n      text-align: center;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      box-sizing: border-box;\n      display: inline-block;\n      position: relative;\n      width: 100%;\n      z-index: inherit;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      bottom: 0;\n      left: 0;\n      position: absolute;\n      right: 0;\n      top: 0;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      background: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  active: function active(t) {
    return css(_templateObject(), styles.fillContainerPosition(), t.loaderBg);
  },
  fillContainerPosition: function fillContainerPosition() {
    return css(_templateObject2());
  },
  loader: function loader() {
    return css(_templateObject3());
  },
  spinnerContainer: function spinnerContainer() {
    return css(_templateObject4());
  },
  spinnerContainerSticky: function spinnerContainerSticky() {
    return css(_templateObject5(), styles.spinnerContainer());
  },
  spinnerContainerCenter: function spinnerContainerCenter() {
    return css(_templateObject6(), styles.spinnerContainer(), styles.fillContainerPosition());
  }
};
export var jsStyles = memoizeStyle(styles);
