// @skbkontur/react-ui/cjs/components/Spinner/Spinner.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");

var _AnimationKeyframes = require("../../lib/theming/AnimationKeyframes");function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: relative;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      text-align: center;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: block;\n      font-size: 16px;\n      line-height: 1.375;\n      margin-top: -6px;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: block;\n      font-size: 18px;\n      line-height: 1.33;\n      margin-top: -8px;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-left: 8px;\n      font-size: 14px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite,\n        ", " 6s ease-in-out infinite;\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  circle: function circle(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.red,
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleLength(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleRotate(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerColor(t));

  },
  circleDimmed: function circleDimmed(t) {
    return (0, _Emotion.css)(_templateObject2(),
    t.spinnerDimmedColor,
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleLength(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleRotate(t));

  },

  captionColor: function captionColor(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.spinnerCaptionColor);

  },

  mini: function mini() {
    return (0, _Emotion.css)(_templateObject4());



  },

  big: function big() {
    return (0, _Emotion.css)(_templateObject5());





  },

  normal: function normal() {
    return (0, _Emotion.css)(_templateObject6());





  },

  spinner: function spinner() {
    return (0, _Emotion.css)(_templateObject7());



  },

  inner: function inner() {
    return (0, _Emotion.css)(_templateObject8());


  },

  fallback: function fallback() {
    return (0, _Emotion.css)(_templateObject9());



  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
