// @skbkontur/react-ui/cjs/components/Spinner/SpinnerFallback.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.SpinnerFallback = exports.types = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));

var _fallback_circle_mini = _interopRequireDefault(require("./fallback_circle_mini.png"));
var _fallback_circle_dimmed = _interopRequireDefault(require("./fallback_circle_dimmed.png"));
var _fallback_circle_big = _interopRequireDefault(require("./fallback_circle_big.png"));
var _fallback_circle_normal = _interopRequireDefault(require("./fallback_circle_normal.png"));
var _Spinner = require("./Spinner.styles");


var types =

{
  big: 'big',
  mini: 'mini',
  normal: 'normal' };exports.types = types;var







SpinnerFallback = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(SpinnerFallback, _React$Component);function SpinnerFallback() {var _this$_framesCount, _this$imageUrls;var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.








    state = {
      frame: 0 };_this.


    mounted = false;_this.


    _framesCount = (_this$_framesCount = {}, _this$_framesCount[
    types.mini] = 151, _this$_framesCount[
    types.normal] = 151, _this$_framesCount[
    types.big] = 151, _this$_framesCount.
    dimmed = 60, _this$_framesCount);_this.


    imageUrls = (_this$imageUrls = {}, _this$imageUrls[
    types.mini] = _fallback_circle_mini.default, _this$imageUrls[
    types.normal] = _fallback_circle_normal.default, _this$imageUrls[
    types.big] = _fallback_circle_big.default, _this$imageUrls.
    dimmed = _fallback_circle_dimmed.default, _this$imageUrls);_this.






































    animate = function () {
      if (!_this.mounted) {
        return;
      }var

      frame = _this.state.frame;
      var framesCount = _this._framesCount[_this.getSpriteSettingsKey()];
      var nextFrame = frame < framesCount ? frame + 1 : 0;
      _this.setState({ frame: nextFrame });

      setTimeout(_this.animate, 1000 / 25);
    };_this.

    getSpriteSettingsKey = function () {return (
        _this.props.type === types.mini && _this.props.dimmed ? 'dimmed' : _this.props.type);};return _this;}var _proto = SpinnerFallback.prototype;_proto.componentDidMount = function componentDidMount() {this.mounted = true;this.animate();};_proto.componentWillUnmount = function componentWillUnmount() {this.mounted = false;};_proto.render = function render() {return this.renderCircle();};_proto.renderCircle = function renderCircle() {var dimmed = this.props.dimmed;var type = this.props.type;var frame = this.state.frame;var size = type === 'big' ? 96 : type === 'normal' ? 48 : 16;var backgroundImage = dimmed ? this.imageUrls.dimmed : this.imageUrls[type];var cssSet = { backgroundImage: "url('" + backgroundImage + "')", height: size, width: size, marginBottom: -3, marginLeft: -1, marginRight: -1 };if (!process.env.enableReactTesting) {cssSet.backgroundPosition = "0 -" + frame * size + "px";}return _react.default.createElement("span", { className: _Spinner.jsStyles.fallback(), style: cssSet });};return SpinnerFallback;}(_react.default.Component);exports.SpinnerFallback = SpinnerFallback;SpinnerFallback.__KONTUR_REACT_UI__ = 'SpinnerFallback';SpinnerFallback.propTypes = { type: _propTypes.default.oneOf(Object.keys(types)), dimmed: _propTypes.default.bool };
