// @skbkontur/react-ui/components/Kebab/Kebab.styles/Kebab.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      overflow: hidden;\n      border-radius: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      ", " & {\n        margin-top: -5px;\n        margin-left: 0.5px;\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", " & {\n        margin-top: -2px;\n        margin-left: 0.5px;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      ", " & {\n        margin-top: 1px;\n        margin-left: 0.8124px;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        cursor: default;\n\n        &:hover {\n          background: none;\n        }\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        background: rgba(0, 0, 0, 0.09);\n        cursor: default;\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        background: rgba(0, 0, 0, 0.09);\n        border-color: ", ";\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      text-align: center;\n      border: 2px solid transparent;\n      box-sizing: border-box;\n      border-radius: 50%;\n      user-select: none;\n      cursor: pointer;\n      outline: 0;\n      transition: background 0.12s ease-out;\n      width: 26px;\n      height: 26px;\n\n      &:hover {\n        background: rgba(0, 0, 0, 0.09);\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  kebab: function kebab() {
    return css(_templateObject());
  },
  focused: function focused(t) {
    return css(_templateObject2(), cssName(styles.kebab()), t.borderColorFocus);
  },
  opened: function opened() {
    return css(_templateObject3(), cssName(styles.kebab()));
  },
  disabled: function disabled() {
    return css(_templateObject4(), cssName(styles.kebab()));
  },
  iconsmall: function iconsmall() {
    return css(_templateObject5(), cssName(styles.kebab()));
  },
  iconmedium: function iconmedium() {
    return css(_templateObject6(), cssName(styles.kebab()));
  },
  iconlarge: function iconlarge() {
    return css(_templateObject7(), cssName(styles.kebab()));
  },
  menu: function menu(t) {
    return css(_templateObject8(), t.popupBorderRadius);
  }
};
export var jsStyles = memoizeStyle(styles);
