// @skbkontur/react-ui/internal/ThemePlayground/FxInputPlayground/FxInputPlayground.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { FxInput } from "../../../components/FxInput";
export var FxInputPlayground =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(FxInputPlayground, _React$Component);

  function FxInputPlayground(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;

    _this.onValueChange = function (value) {
      _this.setState({
        value: value,
        auto: false
      });
    };

    _this.handleRestore = function () {
      _this.setState({
        value: 'auto',
        auto: true
      });
    };

    _this.state = {
      auto: true,
      value: 'auto'
    };
    return _this;
  }

  var _proto = FxInputPlayground.prototype;

  _proto.render = function render() {
    return React.createElement(FxInput, {
      auto: this.state.auto,
      type: 'text',
      value: this.state.value,
      onRestore: this.handleRestore,
      onValueChange: this.onValueChange,
      width: 150
    });
  };

  return FxInputPlayground;
}(React.Component);
