// @skbkontur/react-ui/cjs/internal/IgnoreLayerClick/IgnoreLayerClick.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.IgnoreLayerClick = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _reactDom = require("react-dom");var






IgnoreLayerClick = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(IgnoreLayerClick, _React$Component);function IgnoreLayerClick() {return _React$Component.apply(this, arguments) || this;}var _proto = IgnoreLayerClick.prototype;_proto.
  render = function render() {
    var child = _react.default.Children.only(this.props.children);
    return this.props.active ? _react.default.createElement(IgnoreLayerClickWrapper, null, child) : child;
  };return IgnoreLayerClick;}(_react.default.Component);exports.IgnoreLayerClick = IgnoreLayerClick;






// NOTE Используется только в команде Контур.Бухгалтерия
var IgnoreLayerClickWrapper = /*#__PURE__*/function (_React$Component2) {(0, _inheritsLoose2.default)(IgnoreLayerClickWrapper, _React$Component2);function IgnoreLayerClickWrapper() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component2.call.apply(_React$Component2, [this].concat(args)) || this;_this.


    element = null;_this.




















    handleMouseDown = function (event) {
      event.stopPropagation();
    };return _this;}var _proto2 = IgnoreLayerClickWrapper.prototype;_proto2.componentDidMount = function componentDidMount() {var element = (0, _reactDom.findDOMNode)(this);if (element && element instanceof Element) {element.addEventListener('mousedown', this.handleMouseDown);this.element = element;}};_proto2.componentWillUnmount = function componentWillUnmount() {if (this.element) {this.element.removeEventListener('mousedown', this.handleMouseDown);this.element = null;}};_proto2.render = function render() {return _react.default.Children.only(this.props.children);};return IgnoreLayerClickWrapper;}(_react.default.Component);IgnoreLayerClickWrapper.__KONTUR_REACT_UI__ = 'IgnoreLayerClick';
