// @skbkontur/react-ui/cjs/components/Group/Group.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-left: 0;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-left: -1px;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      flex-basis: 100%;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      flex-grow: 1;\n      flex-shrink: 1;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      flex-shrink: 0;\n      display: inline-block;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-flex;\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject());


  },

  fixed: function fixed() {
    return (0, _Emotion.css)(_templateObject2());



  },

  stretch: function stretch() {
    return (0, _Emotion.css)(_templateObject3());



  },

  stretchFallback: function stretchFallback() {
    return (0, _Emotion.css)(_templateObject4());


  },

  item: function item() {
    return (0, _Emotion.css)(_templateObject5());


  },

  itemFirst: function itemFirst() {
    return (0, _Emotion.css)(_templateObject6());


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
