// @skbkontur/react-ui/components/Center/Center.styles/Center.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      text-align: left;\n      vertical-align: middle;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      height: 100%;\n      vertical-align: middle;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      text-align: right;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      text-align: left;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      text-align: center;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root() {
    return css(_templateObject());
  },
  rootAlignLeft: function rootAlignLeft() {
    return css(_templateObject2());
  },
  rootAlignRight: function rootAlignRight() {
    return css(_templateObject3());
  },
  spring: function spring() {
    return css(_templateObject4());
  },
  container: function container() {
    return css(_templateObject5());
  }
};
export var jsStyles = memoizeStyle(styles);
