// @skbkontur/react-ui/components/Tabs/Indicator.styles/Indicator.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      height: 3px;\n      position: absolute;\n      transition: all 0.2s ease-out;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.tabColorFocus);
  },
  primary: function primary(t) {
    return css(_templateObject2(), t.tabColorPrimary);
  },
  success: function success(t) {
    return css(_templateObject3(), t.tabColorSuccess);
  },
  warning: function warning(t) {
    return css(_templateObject4(), t.tabColorWarning);
  },
  error: function error(t) {
    return css(_templateObject5(), t.tabColorError);
  }
};
export var jsStyles = memoizeStyle(styles);
