// @skbkontur/react-ui/cjs/components/Fias/FiasSearch/FiasSearch.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.FiasSearch = void 0;var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _warning = _interopRequireDefault(require("warning"));

var _locale = require("../locale");
var _FiasComboBox = require("../Form/FiasComboBox");
var _types = require("../types");
var _decorators = require("../../../lib/locale/decorators");
var _filterProps = require("../../../lib/filterProps");

var _ = require("..");var _dec, _class, _class2, _temp;

var COMBOBOX_PASS_PROPS = {
  limit: true,
  align: true,
  searchOnFocus: true,
  drawArrow: true,
  autoFocus: true,
  borderless: true,
  disablePortal: true,
  disabled: true,
  error: true,
  maxLength: true,
  menuAlign: true,
  onBlur: true,
  onFocus: true,
  onInputValueChange: true,
  placeholder: true,
  size: true,
  warning: true,
  width: true,
  maxMenuHeight: true,
  onMouseEnter: true,
  onMouseOver: true,
  onMouseLeave: true };








/**
                         * @deprecated Контур-специфичный компонент, будет удален в 3.0.0, перенесен в `@skbkontur/react-ui-addons` смотри [миграцию](https://github.com/skbkontur/retail-ui/blob/master/packages/react-ui/MIGRATION.md)
                         */var


FiasSearch = (_dec = (0, _decorators.locale)('Fias', _locale.FiasLocaleHelper), _dec(_class = (_temp = _class2 = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(FiasSearch, _React$Component);











  function FiasSearch(props) {var _this;
    _this = _React$Component.call(this, props) || this;_this.locale = void 0;_this.























    renderItem = function (address) {
      return address.getText();
    };_this.

    renderValue = function (address) {
      return address.getText(_types.FiasFields.room);
    };_this.

    renderNotFound = function () {
      return _this.locale.searchNotFound;
    };_this.

    valueToString = function (address) {
      return address.getText(_types.FiasFields.room);
    };_this.

    onValueChange = function (address) {var
      onValueChange = _this.props.onValueChange;
      if (onValueChange) {
        onValueChange(address);
      }
    };_this.

    onUnexpectedInput = function (query) {
      if (!query) {
        return new _.FiasAddress();
      }
    };_this.

    getItems = /*#__PURE__*/function () {var _ref = (0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee(searchText) {var _this$props, api, limit, options;return _regenerator.default.wrap(function _callee$(_context) {while (1) {switch (_context.prev = _context.next) {case 0:_this$props =
                _this.props, api = _this$props.api, limit = _this$props.limit;

                options = {
                  searchText: searchText,
                  fullAddress: true,
                  directParent: false,
                  limit: (limit || FiasSearch.defaultProps.limit) + 1 // +1 to detect if there are more items
                };return _context.abrupt("return",

                api.search(options).then(function (result) {var
                  success = result.success,data = result.data,error = result.error;
                  return success && data ?
                  Promise.resolve(data.map(function (item) {return _.FiasAddress.createFromResponse(item);})) :
                  Promise.reject(error);
                }));case 3:case "end":return _context.stop();}}}, _callee);}));return function (_x) {return _ref.apply(this, arguments);};}();(0, _warning.default)(false, "FiasSearch has been deprecated, use FiasSearch from @skbkontur/react-ui-addons instead, see [migration](https://github.com/skbkontur/retail-ui/blob/master/packages/react-ui/MIGRATION.md)");return _this;}var _proto = FiasSearch.prototype;_proto.render = function render() {var restComboBoxProps = (0, _filterProps.filterProps)(this.props, COMBOBOX_PASS_PROPS);return _react.default.createElement(_FiasComboBox.FiasComboBox, (0, _extends2.default)({ getItems: this.getItems, value: this.props.address, renderItem: this.renderItem, renderValue: this.renderValue, valueToString: this.valueToString, onValueChange: this.onValueChange, onUnexpectedInput: this.onUnexpectedInput, renderNotFound: this.renderNotFound }, restComboBoxProps));};return FiasSearch;}(_react.default.Component), _class2.__KONTUR_REACT_UI__ = 'FiasSearch', _class2.defaultProps = { width: '100%', limit: 5, drawArrow: false, searchOnFocus: false }, _temp)) || _class);exports.FiasSearch = FiasSearch;
