// @skbkontur/react-ui/cjs/lib/theming/AnimationKeyframes.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.AnimationKeyframes = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("./Emotion");function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        100%, 0% { stroke: ", "; }\n        40% { stroke: ", "; }\n        66% { stroke: ", "; }\n        80%, 90% { stroke: ", "; }\n      "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        100% { transform: rotate(360deg); }\n      "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        0% { stroke-dasharray: 62.5%, 168.75%; }\n        50% {  stroke-dasharray: 187.5%, 43.75%; }\n        100% { stroke-dasharray: 62.5%, 168.75%; }\n      "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        0% { stroke-dashoffset: 0; }\n        100% { stroke-dashoffset: -231.25%; }\n      "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        0% { stroke-dasharray: 10, 96; }\n        50% { stroke-dasharray: 50, 56; }\n        100% { stroke-dasharray: 10, 96;}\n      "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n        0% { stroke-dashoffset: 10; }\n        100% { stroke-dashoffset: 116; }\n      "]);_templateObject = function _templateObject() {return data;};return data;}


var AnimationKeyframes = {
  spinnerCloudOffset: function spinnerCloudOffset(t) {
    return (0, _Emotion.keyframes)(_templateObject());



  },

  spinnerCloudLength: function spinnerCloudLength(t) {
    return (0, _Emotion.keyframes)(_templateObject2());




  },
  spinnerCircleOffset: function spinnerCircleOffset(t) {
    return (0, _Emotion.keyframes)(_templateObject3());



  },
  spinnerCircleLength: function spinnerCircleLength(t) {
    return (0, _Emotion.keyframes)(_templateObject4());




  },
  spinnerCircleRotate: function spinnerCircleRotate(t) {
    return (0, _Emotion.keyframes)(_templateObject5());


  },
  spinnerColor: function spinnerColor(t) {
    return (0, _Emotion.keyframes)(_templateObject6(),
    t.red,
    t.yellow,
    t.green,
    t.brand);

  } };exports.AnimationKeyframes = AnimationKeyframes;
