// @skbkontur/react-ui/internal/ThemePlayground/Playground.styles/Playground.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      &:after {\n        box-shadow: 0 2px 10px 0px ", ";\n      }\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      width: 210px;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: 0 -32px;\n      padding: 11px 32px;\n      position: relative;\n\n      &::after {\n        content: '';\n        position: absolute;\n        bottom: 0;\n        left: 0;\n        height: 1px;\n        width: 100%;\n        z-index: -1;\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 300px;\n      margin: auto;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      width: 500px;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      z-index: 1;\n      padding-left: 10px;\n      width: 240px;\n      display: flex;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      z-index: 1;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-left: auto;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 16px;\n      width: 16px;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-bottom: 5px;\n      white-space: nowrap;\n      overflow: hidden;\n      text-overflow: ellipsis;\n      width: 250px;\n      position: relative;\n      z-index: 1;\n      line-height: 34px;\n      color: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 30px;\n      padding-bottom: 30px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      max-width: 700px;\n      color: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      width: 110px;\n      color: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: -16px;\n      padding: 32px;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  playground: function playground() {
    return css(_templateObject());
  },
  title: function title(t) {
    return css(_templateObject2(), t.textColorDefault);
  },
  componentsGroup: function componentsGroup(t) {
    return css(_templateObject3(), t.textColorDefault);
  },
  sidePageBody: function sidePageBody() {
    return css(_templateObject4());
  },
  variableName: function variableName(t) {
    return css(_templateObject5(), t.textColorDefault);
  },
  colorIcon: function colorIcon() {
    return css(_templateObject6());
  },
  baseLinkWrapper: function baseLinkWrapper() {
    return css(_templateObject7());
  },
  linkRoot: function linkRoot() {
    return css(_templateObject8());
  },
  baseVariableRoot: function baseVariableRoot() {
    return css(_templateObject9());
  },
  loaderWrapper: function loaderWrapper() {
    return css(_templateObject10());
  },
  loader: function loader() {
    return css(_templateObject11());
  },
  tabsWrapper: function tabsWrapper() {
    return css(_templateObject12());
  },
  tooltipContent: function tooltipContent() {
    return css(_templateObject13());
  },
  stickyTabsWrapper: function stickyTabsWrapper(t) {
    return css(_templateObject14(), t.textColorDefault);
  },
  tabsInnerWrapper: function tabsInnerWrapper(t) {
    return css(_templateObject15(), t.textColorDefault);
  },
  playgroundWrapper: function playgroundWrapper(t) {
    return css(_templateObject16(), t.bgDefault);
  },
  editorHeaderWrapper: function editorHeaderWrapper(t) {
    return css(_templateObject17(), t.textColorDefault);
  },
  editorGroupHeader: function editorGroupHeader(t) {
    return css(_templateObject18(), t.textColorDefault);
  }
};
export var jsStyles = memoizeStyle(styles);
