// @skbkontur/react-ui/cjs/components/TokenInput/TextWidthHelper.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.TextWidthHelper = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));

var _RenderContainer = require("../../internal/RenderContainer");





var styles = {
  position: 'absolute',
  left: -1000,
  top: -1000,
  fontSize: 14 };var


TextWidthHelper = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(TextWidthHelper, _React$Component);function TextWidthHelper() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.
    element = null;_this.















    elementRef = function (node) {return _this.element = node;};return _this;}var _proto = TextWidthHelper.prototype;_proto.render = function render() {return _react.default.createElement(_RenderContainer.RenderContainer, null, _react.default.createElement("div", { style: styles, ref: this.elementRef }, this.props.text));};_proto.getTextWidth = function getTextWidth() {return this.element.getBoundingClientRect().width;};return TextWidthHelper;}(_react.default.Component);exports.TextWidthHelper = TextWidthHelper;
