// @skbkontur/react-ui/cjs/components/TokenInput/TokenInput.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      pointer-events: none;\n      /* fix text color in safari */\n      -webkit-text-fill-color: currentcolor;\n      color: ", ";\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      min-width: 0;\n      max-width: 100%;\n      width: 50px;\n      background: transparent;\n      border: none;\n      box-shadow: none;\n      outline: none;\n      margin: 2px 0;\n      font-family: inherit;\n      font-size: 14px;\n      padding: 0 0 0 5px;\n      height: 24px;\n      line-height: 24px;\n      -webkit-appearance: none;\n      white-space: nowrap;\n      text-overflow: clip;\n      background-clip: padding-box;\n      transition: background-color 0.15s ease-in;\n\n      &::-ms-clear {\n        display: none;\n      }\n      &::placeholder {\n        color: ", ";\n        -webkit-text-fill-color: ", ";\n      }\n      &:focus::placeholder {\n        color: ", ";\n        -webkit-text-fill-color: ", ";\n      }\n      /* fix firefox placeholder opacity */\n      &:-moz-placeholder {\n        opacity: 1;\n      }\n      &::-moz-placeholder {\n        opacity: 1;\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      border-color: ", ";\n      box-shadow: none;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 1px solid ", ";\n      box-shadow: 0 0 0 1px ", ";\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 1px solid ", ";\n      box-shadow: 0 0 0 1px ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 1px solid ", ";\n      box-shadow: 0 0 0 1px ", ";\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background-color: ", ";\n      box-shadow: ", ";\n      border: ", " solid ", ";\n      border-top-color: ", ";\n      box-sizing: border-box;\n      cursor: text;\n      padding: 2px 4px;\n      display: flex;\n      flex-wrap: wrap;\n      outline: none;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  label: function label(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.inputBg,
    t.inputShadow,
    t.inputBorderWidth, t.borderColorGrayLight,
    t.inputBorderTopColor);







  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject2(),
    t.borderColorWarning,
    t.borderColorWarning);

  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.borderColorError,
    t.borderColorError);

  },

  labelFocused: function labelFocused(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.borderColorFocus,
    t.borderColorFocus);

  },

  labelDisabled: function labelDisabled(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.inputDisabledBg,
    t.inputDisabledBorderColor);


  },

  input: function input(t) {
    return (0, _Emotion.css)(_templateObject6(),























    t.placeholderColor,
    t.placeholderColor,


    t.placeholderColorLight,
    t.placeholderColorLight);









  },

  inputDisabled: function inputDisabled(t) {
    return (0, _Emotion.css)(_templateObject7(),



    t.textColorDisabled);

  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
