// @skbkontur/react-ui/lib/ModalStack/ModalStack.js
import EventEmitter from 'eventemitter3';
export var ModalStack =
/*#__PURE__*/
function () {
  function ModalStack() {}

  ModalStack.add = function add(component, onChange) {
    var _ModalStack$getStackI = ModalStack.getStackInfo(),
        emitter = _ModalStack$getStackI.emitter,
        mounted = _ModalStack$getStackI.mounted;

    mounted.unshift(component);

    var changeHandler = function changeHandler() {
      return onChange([].concat(mounted));
    };

    emitter.addListener('change', changeHandler);
    emitter.emit('change');
    return {
      remove: function remove() {
        emitter.removeListener('change', changeHandler);
      }
    };
  };

  ModalStack.remove = function remove(component) {
    var _ModalStack$getStackI2 = ModalStack.getStackInfo(),
        emitter = _ModalStack$getStackI2.emitter,
        mounted = _ModalStack$getStackI2.mounted;

    var index = mounted.indexOf(component);

    if (index !== -1) {
      mounted.splice(index, 1);
    }

    emitter.emit('change');
  };

  ModalStack.getStackInfo = function getStackInfo() {
    var globalWithStack = global;
    return globalWithStack.__ReactUIStackInfo || (globalWithStack.__ReactUIStackInfo = {
      emitter: new EventEmitter(),
      mounted: []
    });
  };

  return ModalStack;
}();
