// @skbkontur/react-ui/cjs/internal/Calendar/CalendarDateShape.js
"use strict";exports.__esModule = true;exports.isBetween = exports.isGreaterOrEqual = exports.isGreater = exports.isLessOrEqual = exports.isLess = exports.create = exports.comparator = exports.isEqual = void 0;







var isEqual = function isEqual(a, b) {return (
    !a && !b || a && b && a.year === b.year && a.month === b.month && a.date === b.date);};exports.isEqual = isEqual;

var comparator = function comparator(a, b) {
  if (a.year < b.year) {
    return -1;
  } else if (a.year > b.year) {
    return 1;
  } else if (a.month < b.month) {
    return -1;
  } else if (a.month > b.month) {
    return 1;
  } else if (a.date < b.date) {
    return -1;
  } else if (a.date > b.date) {
    return 1;
  }
  return 0;
};exports.comparator = comparator;

var create = function create(date, month, year) {return { date: date, month: month, year: year };};exports.create = create;

var isLess = function isLess(left, right) {return comparator(left, right) === -1;};exports.isLess = isLess;

var isLessOrEqual = function isLessOrEqual(left, right) {return (
    isLess(left, right) || isEqual(left, right));};exports.isLessOrEqual = isLessOrEqual;

var isGreater = function isGreater(left, right) {return !isLessOrEqual(left, right);};exports.isGreater = isGreater;

var isGreaterOrEqual = function isGreaterOrEqual(left, right) {return !isLess(left, right);};exports.isGreaterOrEqual = isGreaterOrEqual;

var isBetween = function isBetween(
date,
left,
right)
{
  if (left && isLess(date, left)) {
    return false;
  }
  if (right && isGreater(date, right)) {
    return false;
  }
  return true;
};exports.isBetween = isBetween;
