// @skbkontur/react-ui/cjs/internal/Calendar/MonthView.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-bottom: 1px solid ", ";\n      font-weight: bold;\n      margin-bottom: 10px;\n      position: relative;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background-color: ", ";\n      z-index: 1;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      width: 210px;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: absolute;\n      right: 0;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  headerMonth: function headerMonth() {
    return (0, _Emotion.css)(_templateObject());


  },

  headerYear: function headerYear() {
    return (0, _Emotion.css)(_templateObject2());




  },

  month: function month() {
    return (0, _Emotion.css)(_templateObject3());



  },

  headerSticky: function headerSticky(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.calendarMonthHeaderStickedBgColor);


  },

  monthTitle: function monthTitle(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.calendarMonthTitleBorderBottomColor);




  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
