// @skbkontur/react-ui/components/DateInput/DateInput.styles/DateInput.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: default;\n      color: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: pointer;\n      color: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css } from "../../../lib/theming/Emotion";
export var jsStyles = {
  icon: function icon(t) {
    return css(_templateObject(), t.dateInputIconColor);
  },
  iconSmall: function iconSmall(t) {
    return css(_templateObject2(), t.inputFontSizeSmall);
  },
  iconMedium: function iconMedium(t) {
    return css(_templateObject3(), t.inputFontSizeMedium);
  },
  iconLarge: function iconLarge(t) {
    return css(_templateObject4(), t.inputFontSizeLarge);
  },
  iconDisabled: function iconDisabled(t) {
    return css(_templateObject5(), t.textColorDisabled);
  }
};
