// @skbkontur/react-ui/internal/InternalMenu/InternalMenu.styles/InternalMenu.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1);\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: 0 1px 0 rgba(0, 0, 0, 0.1);\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      bottom: -5px;\n      position: relative;\n      z-index: 1;\n      width: 100%;\n      overflow: hidden;\n      line-height: 18px;\n      box-sizing: border-box;\n      padding: 6px 18px 7px 8px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      top: -5px;\n      position: relative;\n      z-index: 1;\n      width: 100%;\n      overflow: hidden;\n      line-height: 18px;\n      box-sizing: border-box;\n      padding: 6px 18px 7px 8px;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      border: 1px solid #d5d5d5;\n      box-shadow: 0 2px 6px rgba(0, 0, 0, 0.2);\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      overflow: auto;\n      padding: 5px 0;\n      outline: none;\n      background: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.bgDefault);
  },
  shadow: function shadow() {
    return css(_templateObject2());
  },
  header: function header() {
    return css(_templateObject3());
  },
  footer: function footer() {
    return css(_templateObject4());
  },
  fixedHeader: function fixedHeader() {
    return css(_templateObject5());
  },
  fixedFooter: function fixedFooter() {
    return css(_templateObject6());
  }
};
export var jsStyles = memoizeStyle(styles);
