// @skbkontur/react-ui/cjs/components/Fias/models/FiasData.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.FiasData = void 0;var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));var










FiasData = /*#__PURE__*/function () {



































  function FiasData(data) {this.data = data;}(0, _createClass2.default)(FiasData, [{ key: "name", get: function get()

    {var
      data = this.data;
      if (FiasData.isAddressObject(data)) {
        return data.name;
      } else if (FiasData.isStead(data)) {
        return data.number;
      } else if (FiasData.isHouse(data)) {
        return data.number || data.structureNumber || data.buildingNumber || '';
      } else if (FiasData.isRoom(data)) {
        return data.flatNumber;
      }
      return '';
    } }, { key: "abbreviation", get: function get()

    {
      return FiasData.isAddressObject(this.data) ? this.data.abbreviation : '';
    } }, { key: "code", get: function get()

    {
      return FiasData.isAddressObject(this.data) && this.data.code || '';
    } }, { key: "number", get: function get()

    {
      return (FiasData.isStead(this.data) || FiasData.isHouse(this.data)) && this.data.number ? this.data.number : '';
    } }, { key: "structureNumber", get: function get()

    {
      return FiasData.isHouse(this.data) && this.data.structureNumber ? this.data.structureNumber : '';
    } }, { key: "buildingNumber", get: function get()

    {
      return FiasData.isHouse(this.data) && this.data.buildingNumber ? this.data.buildingNumber : '';
    } }, { key: "fiasId", get: function get()

    {
      return this.data.fiasId;
    } }, { key: "parentFiasId", get: function get()

    {
      return this.data && this.data.parentFiasId || '';
    } }, { key: "postalCode", get: function get()

    {
      return this.data && this.data.postalCode || '';
    } }, { key: "estateStatus", get: function get()

    {
      if (FiasData.isHouse(this.data)) {
        return this.data.estateStatus;
      }
    } }, { key: "structureStatus", get: function get()

    {
      if (FiasData.isHouse(this.data)) {
        return this.data.structureStatus;
      }
    } }]);return FiasData;}();exports.FiasData = FiasData;FiasData.isAddressObject = function (data) {return data && Object.prototype.hasOwnProperty.call(data, 'name') && Object.prototype.hasOwnProperty.call(data, 'abbreviation');};FiasData.isStead = function (data) {return data && Object.prototype.hasOwnProperty.call(data, 'number') && Object.prototype.hasOwnProperty.call(data, 'liveStatus') && !Object.prototype.hasOwnProperty.call(data, 'estateStatus');};FiasData.isHouse = function (data) {return data && Object.prototype.hasOwnProperty.call(data, 'structureStatus') && (Object.prototype.hasOwnProperty.call(data, 'number') || Object.prototype.hasOwnProperty.call(data, 'structureNumber'));};FiasData.isRoom = function (data) {return data && Object.prototype.hasOwnProperty.call(data, 'flatType') && Object.prototype.hasOwnProperty.call(data, 'flatNumber') && Object.prototype.hasOwnProperty.call(data, 'liveStatus');};
