// @skbkontur/react-ui/components/LocaleProvider/LocaleProvider/LocaleProvider.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import warning from 'warning';
import React from 'react';
import { defaultLangCode } from "../../../lib/locale/constants";
import { LocaleContext } from "../../../lib/locale";
export var LocaleProvider =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(LocaleProvider, _React$Component);

  function LocaleProvider() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = LocaleProvider.prototype;

  _proto.componentDidMount = function componentDidMount() {
    warning(true, "LocaleProvider was deprecated please use 'LocaleContext' instead");
  };

  _proto.render = function render() {
    var _this$props = this.props,
        locale = _this$props.locale,
        langCode = _this$props.langCode;
    return React.createElement(LocaleContext.Provider, {
      value: {
        locale: locale,
        langCode: langCode
      }
    }, this.props.children);
  };

  return LocaleProvider;
}(React.Component);
LocaleProvider.__KONTUR_REACT_UI__ = 'LocaleProvider';
LocaleProvider.defaultProps = {
  locale: {},
  langCode: defaultLangCode
};
export var LocaleConsumer =
/*#__PURE__*/
function (_React$Component2) {
  _inheritsLoose(LocaleConsumer, _React$Component2);

  function LocaleConsumer() {
    return _React$Component2.apply(this, arguments) || this;
  }

  var _proto2 = LocaleConsumer.prototype;

  _proto2.componentDidMount = function componentDidMount() {
    warning(true, "LocaleProvider was deprecated please use 'LocaleContext' instead");
  };

  _proto2.render = function render() {
    return React.createElement(LocaleContext.Consumer, null, this.props.children);
  };

  return LocaleConsumer;
}(React.Component);
