// @skbkontur/react-ui/cjs/internal/Popup/Popup.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject13() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      opacity: 0.01;\n      transition: opacity 0.15s ease-out;\n      transform: translate(0, 0);\n    "]);_templateObject13 = function _templateObject13() {return data;};return data;}function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transition: transform 0.18s cubic-bezier(0.22, 0.61, 0.36, 1), opacity 0.18s cubic-bezier(0.22, 0.61, 0.36, 1);\n      opacity: 1;\n      transform: translate(0, 0);\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateX(-10px);\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateX(10px);\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateY(-10px);\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateY(10px);\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      opacity: 0.01;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: ", ";\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      filter: ", ";\n      -webkit-filter: ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n      background: ", ";\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      overflow: hidden;\n      border-radius: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      pointer-events: none;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      min-width: 18px;\n      border-radius: ", ";\n      border: ", " ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  popup: function popup(t) {
    return (0, _Emotion.css)(_templateObject(),


    t.popupBorderRadius,
    t.popupBorder, t.popupBorderColor);

  },

  popupIgnoreHover: function popupIgnoreHover() {
    return (0, _Emotion.css)(_templateObject2());


  },

  content: function content(t) {
    return (0, _Emotion.css)(_templateObject3(),

    t.popupBorderRadius);

  },

  contentInner: function contentInner(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.popupTextColor,
    t.popupBackground);

  },

  shadow: function shadow(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.popupDropShadow,
    t.popupDropShadow);

  },

  shadowFallback: function shadowFallback(t) {
    return (0, _Emotion.css)(_templateObject6(),
    t.popupBoxShadow);

  },

  transitionEnter: function transitionEnter() {
    return (0, _Emotion.css)(_templateObject7());


  },
  'transition-enter-top': function transitionEnterTop() {
    return (0, _Emotion.css)(_templateObject8());


  },
  'transition-enter-bottom': function transitionEnterBottom() {
    return (0, _Emotion.css)(_templateObject9());


  },
  'transition-enter-left': function transitionEnterLeft() {
    return (0, _Emotion.css)(_templateObject10());


  },
  'transition-enter-right': function transitionEnterRight() {
    return (0, _Emotion.css)(_templateObject11());


  },
  transitionEnterActive: function transitionEnterActive() {
    return (0, _Emotion.css)(_templateObject12());




  },
  transitionExit: function transitionExit() {
    return (0, _Emotion.css)(_templateObject13());




  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
