// @skbkontur/react-ui/components/Modal/Modal.styles/Modal.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-bottom: 30px;\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-top: 30px;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-bottom: 22px;\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-top: 20px;\n      margin-top: 10px;\n      background: ", ";\n\n      &:before {\n        top: 11px;\n        content: '';\n        position: absolute;\n        height: 1px;\n        width: 100%;\n        left: 0;\n        z-index: -1;\n        box-shadow: ", ";\n      }\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-bottom: 10px;\n      background: ", ";\n\n      &:after {\n        bottom: 11px;\n        content: '';\n        position: absolute;\n        height: 1px;\n        width: 100%;\n        left: 0;\n        z-index: -1;\n        box-shadow: ", ";\n      }\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        padding-top: 20px;\n        padding-bottom: 20px;\n        background: ", ";\n      }\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 0 35px 30px 30px;\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 0 35px 25px 30px;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: 22px;\n      line-height: 30px;\n      padding: 24px 110px 11px 30px;\n      overflow-wrap: break-word;\n      word-wrap: break-word;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      right: 0;\n      top: 0;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      ", " & {\n        display: none;\n      }\n\n      ", ":focus & {\n        border: 2px solid ", ";\n        position: absolute;\n        display: block;\n        top: 24px;\n        right: 26px;\n        width: 26px;\n        height: 26px;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        pointer-events: none;\n        cursor: default;\n        color: ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      background: none;\n      margin: 2px 2px 0 0;\n      width: 76px;\n      height: 65px;\n      position: relative;\n      float: right;\n      cursor: pointer;\n\n      &:after,\n      &:before {\n        content: '';\n        position: absolute;\n        top: 30px;\n        right: 40px;\n        height: 18px;\n        width: 2px;\n        background: ", ";\n      }\n\n      &:after {\n        transform: rotate(45deg);\n      }\n\n      &:before {\n        transform: rotate(-45deg);\n      }\n\n      &:focus::before,\n      &:focus::after,\n      &:hover::before,\n      &:hover::after {\n        background: ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        vertical-align: top;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      display: inline-block;\n      text-align: left;\n      vertical-align: middle;\n      box-sizing: border-box;\n      z-index: 10;\n      margin: 40px 20px;\n\n      @media screen and (max-width: ", ") {\n        margin: 0;\n        width: 100%;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      white-space: normal;\n      margin: auto;\n      box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);\n      background: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      white-space: nowrap;\n      text-align: center;\n      height: 100%;\n      overflow-y: auto;\n      outline: none;\n\n      &::before {\n        content: '';\n        display: inline-block;\n        vertical-align: middle;\n        height: 80%; /* to vertical align modal 40%/60% of screen height */\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      bottom: 0;\n      left: 0;\n      position: absolute;\n      right: 0;\n      top: 0;\n      background: ", ";\n      opacity: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      left: 0;\n      position: fixed;\n      top: 0;\n      width: 100%;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
import { resetButton } from "../../../lib/styles/Mixins";
var styles = {
  root: function root() {
    return css(_templateObject());
  },
  bg: function bg(t) {
    return css(_templateObject2(), t.modalBackBg, t.modalBackOpacity);
  },
  container: function container() {
    return css(_templateObject3());
  },
  window: function window(t) {
    return css(_templateObject4(), t.bgDefault);
  },
  centerContainer: function centerContainer(t) {
    return css(_templateObject5(), t.modalAdaptiveThreshold);
  },
  alignTop: function alignTop(t) {
    return css(_templateObject6(), cssName(styles.centerContainer(t)));
  },
  close: function close(t) {
    return css(_templateObject7(), resetButton(), t.modalCloseButtonColor, t.modalCloseButtonHoverColor);
  },
  disabled: function disabled(t) {
    return css(_templateObject8(), cssName(styles.close(t)), t.modalCloseButtonDisabledColor);
  },
  closeOutline: function closeOutline(t) {
    return css(_templateObject9(), cssName(styles.close(t)), cssName(styles.close(t)), t.borderColorFocus);
  },
  absoluteClose: function absoluteClose() {
    return css(_templateObject10());
  },
  closeWrapper: function closeWrapper() {
    return css(_templateObject11());
  },
  header: function header() {
    return css(_templateObject12());
  },
  body: function body() {
    return css(_templateObject13());
  },
  footer: function footer() {
    return css(_templateObject14());
  },
  panel: function panel(t) {
    return css(_templateObject15(), cssName(styles.footer()), t.modalFooterBg);
  },
  fixedHeader: function fixedHeader(t) {
    return css(_templateObject16(), t.modalFixedHeaderBg, t.modalFixedHeaderShadow);
  },
  fixedFooter: function fixedFooter(t) {
    return css(_templateObject17(), t.modalFixedHeaderBg, t.modalFixedFooterShadow);
  },
  headerAddPadding: function headerAddPadding() {
    return css(_templateObject18());
  },
  bodyWithoutHeader: function bodyWithoutHeader() {
    return css(_templateObject19());
  },
  bodyAddPadding: function bodyAddPadding() {
    return css(_templateObject20());
  }
};
export var jsStyles = memoizeStyle(styles);
