// @skbkontur/react-ui/components/Paging/Paging.styles/Paging.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      margin: 0 -20px;\n      font-size: 11px;\n\n      ", " & {\n        color: ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 15px;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      color: transparent;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: 0 -1px !important;\n      border: solid 2px ", ";\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: default;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      border-radius: 16px;\n      color: ", ";\n      cursor: pointer;\n      display: block;\n      font-size: 16px;\n      margin: 2px 1px;\n      outline: none;\n      padding: 2px 10px 5px;\n      text-decoration: none;\n\n      &:not(", "):hover {\n        background: ", ";\n      }\n\n      ", "& {\n        background: ", ";\n        color: ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      font-size: 0;\n      text-align: center;\n      user-select: none;\n      vertical-align: top;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        color: ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      vertical-align: -2px;\n      position: absolute;\n      right: 0;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      text-decoration: underline;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      font-size: 16px;\n      margin: 4px 0 2px 10px;\n      outline: none;\n      padding-right: 22px;\n      position: relative;\n      text-decoration: none;\n      user-select: none;\n      vertical-align: top;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      display: inline-block;\n      font-size: 16px;\n      padding: 6px 10px 0;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      user-select: none;\n      outline: 0;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  paging: function paging() {
    return css(_templateObject());
  },
  dots: function dots(t) {
    return css(_templateObject2(), t.pagingDotsColor);
  },
  forwardLink: function forwardLink(t) {
    return css(_templateObject3(), t.pagingForwardLinkColor);
  },
  forwardLinkFocused: function forwardLinkFocused() {
    return css(_templateObject4());
  },
  forwardIcon: function forwardIcon() {
    return css(_templateObject5());
  },
  disabled: function disabled(t) {
    return css(_templateObject6(), cssName(styles.forwardLink(t)), t.pagingForwardLinkDisabledColor);
  },
  pageLinkWrapper: function pageLinkWrapper() {
    return css(_templateObject7());
  },
  pageLink: function pageLink(t) {
    return css(_templateObject8(), t.pagingForwardLinkColor, cssName(styles.active()), t.pagingPageLinkHoverBg, cssName(styles.active()), t.pagingPageLinkActiveBg, t.pagingPageLinkActiveColor);
  },
  active: function active() {
    return css(_templateObject9());
  },
  pageLinkFocused: function pageLinkFocused(t) {
    return css(_templateObject10(), t.borderColorFocus);
  },
  transparent: function transparent() {
    return css(_templateObject11());
  },
  pageLinkHintPlaceHolder: function pageLinkHintPlaceHolder() {
    return css(_templateObject12());
  },
  pageLinkHint: function pageLinkHint(t) {
    return css(_templateObject13(), cssName(styles.pageLinkWrapper()), t.pagingPageLinkHintColor);
  }
};
export var jsStyles = memoizeStyle(styles);
