// @skbkontur/react-ui/components/Fias/FiasModal/FiasModal.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";

var _dec, _class, _class2, _temp;

import React from 'react';
import { Button } from "../../Button";
import { Gapped } from "../../Gapped";
import { Modal } from "../../Modal";
import { locale } from "../../../lib/locale/decorators";
import { FiasLocaleHelper } from "../locale";
export var FiasModal = (_dec = locale('Fias', FiasLocaleHelper), _dec(_class = (_temp = _class2 =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(FiasModal, _React$Component);

  function FiasModal() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.locale = void 0;
    return _this;
  }

  var _proto = FiasModal.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        onClose = _this$props.onClose,
        onSave = _this$props.onSave,
        children = _this$props.children;
    return React.createElement(Modal, {
      width: 500,
      onClose: onClose
    }, React.createElement(Modal.Header, null, this.locale.modalTitle), React.createElement(Modal.Body, null, children), React.createElement(Modal.Footer, {
      panel: true
    }, React.createElement(Gapped, null, React.createElement(Button, {
      size: "medium",
      use: "primary",
      onClick: onSave
    }, this.locale.modalButtonOk), React.createElement(Button, {
      size: "medium",
      onClick: onClose
    }, this.locale.modalButtonCancel))));
  };

  return FiasModal;
}(React.Component), _class2.__KONTUR_REACT_UI__ = 'FiasModal', _class2.defaultProps = {
  onClose: function onClose() {
    return null;
  },
  onSave: function onSave() {
    return null;
  }
}, _temp)) || _class);
