// @skbkontur/react-ui/cjs/components/MenuItem/MenuItem.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));
var _Emotion = require("../../lib/theming/Emotion");
var _Mixins = require("../../lib/styles/Mixins");function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: absolute;\n      left: 15px;\n      top: 5px;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: #fff;\n      opacity: 0.6;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: #a0a0a0;\n      white-space: normal;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      & {\n        padding-left: ", ";\n      }\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 15px;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: transparent;\n      color: ", ";\n      cursor: default;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      color: ", " !important;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      cursor: pointer;\n      display: block;\n      line-height: 18px;\n      padding: 6px 18px 7px 8px;\n      position: relative;\n      text-decoration: none;\n\n      button& {\n        min-width: 100%;\n      }\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject(),
    (0, _Mixins.resetButton)());












  },
  hover: function hover(t) {
    // Color with !important in purpose to override `a:hover`
    return (0, _Emotion.css)(_templateObject2(),
    t.dropdownMenuHoverBg,
    t.textColorInvert);

  },
  selected: function selected(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.dropdownMenuSelectedBg);

  },
  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject4(),

    t.textColorDisabled);


  },
  link: function link(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.linkColor);

  },
  loose: function loose() {
    return (0, _Emotion.css)(_templateObject6());


  },
  withIcon: function withIcon(t) {
    return (0, _Emotion.css)(_templateObject7(),

    t.menuItemPaddingForIcon);


  },
  comment: function comment() {
    return (0, _Emotion.css)(_templateObject8());



  },
  commentHover: function commentHover() {
    return (0, _Emotion.css)(_templateObject9());



  },
  icon: function icon() {
    return (0, _Emotion.css)(_templateObject10());





  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
