// @skbkontur/react-ui/internal/Calendar/MonthView.styles/MonthView.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      border-bottom: 1px solid ", ";\n      font-weight: bold;\n      margin-bottom: 10px;\n      position: relative;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: ", ";\n      z-index: 1;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      width: 210px;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: absolute;\n      right: 0;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  headerMonth: function headerMonth() {
    return css(_templateObject());
  },
  headerYear: function headerYear() {
    return css(_templateObject2());
  },
  month: function month() {
    return css(_templateObject3());
  },
  headerSticky: function headerSticky(t) {
    return css(_templateObject4(), t.calendarMonthHeaderStickedBgColor);
  },
  monthTitle: function monthTitle(t) {
    return css(_templateObject5(), t.calendarMonthTitleBorderBottomColor);
  }
};
export var jsStyles = memoizeStyle(styles);
