// @skbkontur/react-ui/components/Textarea/Textarea.styles/Textarea.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      -ms-user-select: none;\n      color: #aaa;\n      left: 12px;\n      overflow: hidden;\n      position: absolute;\n      right: 12px;\n      top: 9px;\n      user-select: none;\n      z-index: 3;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 0;\n      left: 0;\n      position: absolute;\n      top: 0;\n      visibility: hidden;\n      width: 100%;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n      ", "&:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n      ", "&:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      -webkit-appearance: none;\n      background-clip: padding-box;\n      background: ", ";\n      border: 1px solid ", ";\n      border-top-color: ", ";\n      box-shadow: ", ";\n      box-sizing: border-box;\n      color: ", ";\n      font-family: inherit;\n      font-size: inherit;\n      line-height: inherit;\n      max-width: 100%;\n      min-height: ", ";\n      outline: none;\n      padding: 6px 10px;\n      transition: height 0.2s ease-out;\n      vertical-align: middle;\n      width: 100%;\n\n      &:focus {\n        border-color: ", ";\n        box-shadow: 0 0 0 1px ", ";\n        position: relative;\n        z-index: 2;\n\n        &::placeholder {\n          color: ", ";\n        }\n      }\n      &:disabled {\n        color: ", ";\n        background: ", ";\n        border-color: ", ";\n        box-shadow: none;\n      }\n\n      &::placeholder {\n        color: ", ";\n      }\n\n      &:-moz-placeholder {\n        color: inherit; // Need to be discussed\n      }\n\n      &::-moz-placeholder {\n        color: inherit;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      font-size: ", ";\n      line-height: ", ";\n      position: relative;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.fontSizeMedium, t.textareaLineHeight);
  },
  textarea: function textarea(t) {
    return css(_templateObject2(), t.textareaBg, t.borderColorGrayLight, t.textareaBorderTopColor, t.textareaShadow, t.textareaColor, t.controlHeightSmall, t.borderColorFocus, t.borderColorFocus, t.placeholderColorLight, t.textColorDisabled, t.textareaDisabledBg, t.textareaDisabledBorderColor, t.placeholderColor);
  },
  error: function error(t) {
    return css(_templateObject3(), cssName(styles.textarea(t)), t.borderColorError, t.borderColorError, cssName(styles.textarea(t)), t.borderColorError, t.borderColorError);
  },
  warning: function warning(t) {
    return css(_templateObject4(), cssName(styles.textarea(t)), t.borderColorWarning, t.borderColorWarning, cssName(styles.textarea(t)), t.borderColorWarning, t.borderColorWarning);
  },
  fake: function fake() {
    return css(_templateObject5());
  },
  placeholder: function placeholder() {
    return css(_templateObject6());
  }
};
export var jsStyles = memoizeStyle(styles);
