// @skbkontur/react-ui/components/Select/Select.styles/Select.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: 1px 4px 4px;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 0;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      max-width: 100%;\n      overflow: hidden;\n      text-overflow: ellipsis;\n      vertical-align: top;\n      white-space: nowrap;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      box-sizing: border-box;\n      display: inline-block;\n      padding: 0 28px 0 9px;\n      max-width: 100%;\n      width: 100%;\n      position: relative;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      border-top-color: inherit;\n      color: inherit;\n      opacity: 0.7;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      right: 0;\n      padding: 0 10px;\n\n      &::before {\n        content: '';\n        display: inline-block;\n        height: 100%;\n        vertical-align: middle;\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      border: 4px solid transparent;\n      border-bottom-width: 0;\n      border-top-color: ", ";\n\n      ", " & {\n        display: inline-block;\n        margin-bottom: 3px;\n        vertical-align: middle;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      color: inherit;\n      opacity: 0.7;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      text-overflow: ellipsis;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: relative;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject());
  },
  placeholder: function placeholder(t) {
    return css(_templateObject2(), t.sltPlaceholderColor);
  },
  customUsePlaceholder: function customUsePlaceholder() {
    return css(_templateObject3());
  },
  arrow: function arrow(t) {
    return css(_templateObject4(), t.btnMenuArrowColor, cssName(styles.arrowWrap()));
  },
  arrowWrap: function arrowWrap() {
    return css(_templateObject5());
  },
  customUseArrow: function customUseArrow() {
    return css(_templateObject6());
  },
  label: function label() {
    return css(_templateObject7());
  },
  labelText: function labelText() {
    return css(_templateObject8());
  },
  labelWithLeftIcon: function labelWithLeftIcon() {
    return css(_templateObject9());
  },
  search: function search() {
    return css(_templateObject10());
  }
};
export var jsStyles = memoizeStyle(styles);
