// @skbkontur/react-ui/cjs/internal/SpinnerOld/SpinnerOldFallback.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.SpinnerOldFallback = exports.types = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var PropTypes = _interopRequireWildcard(require("prop-types"));

var _SpinnerOldIcon = require("../icons/SpinnerOldIcon");

var _fallback_circle = _interopRequireDefault(require("./fallback_circle.png"));
var _fallback_circle_dimmed = _interopRequireDefault(require("./fallback_circle_dimmed.png"));
var _fallback_cloud_big = _interopRequireDefault(require("./fallback_cloud_big.png"));
var _fallback_cloud_normal = _interopRequireDefault(require("./fallback_cloud_normal.png"));
var _SpinnerOld = require("./SpinnerOld.styles");


var types =

{
  big: 'big',
  mini: 'mini',
  normal: 'normal' };exports.types = types;var







SpinnerOldFallback = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(SpinnerOldFallback, _React$Component);function SpinnerOldFallback() {var _this$_framesCount, _this$imageUrls;var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.








    state = {
      frame: 0 };_this.


    mounted = false;_this.

    _framesCount = (_this$_framesCount = {}, _this$_framesCount[
    types.mini] = 180, _this$_framesCount[
    types.normal] = 60, _this$_framesCount[
    types.big] = 60, _this$_framesCount.
    dimmed = 60, _this$_framesCount);_this.


    imageUrls = (_this$imageUrls = {}, _this$imageUrls[
    types.mini] = _fallback_circle.default, _this$imageUrls[
    types.normal] = _fallback_cloud_normal.default, _this$imageUrls[
    types.big] = _fallback_cloud_big.default, _this$imageUrls.
    dimmed = _fallback_circle_dimmed.default, _this$imageUrls);_this.





















































    animate = function () {
      if (!_this.mounted) {
        return;
      }var

      frame = _this.state.frame;
      var framesCount = _this._framesCount[_this.getSpriteSettingsKey()];
      var nextFrame = frame < framesCount ? frame + 1 : 0;
      _this.setState({ frame: nextFrame });

      setTimeout(_this.animate, 1000 / 25);
    };_this.

    getSpriteSettingsKey = function () {return (
        _this.props.type === types.mini && _this.props.dimmed ? 'dimmed' : _this.props.type);};return _this;}var _proto = SpinnerOldFallback.prototype;_proto.componentDidMount = function componentDidMount() {this.mounted = true;this.animate();};_proto.componentWillUnmount = function componentWillUnmount() {this.mounted = false;};_proto.render = function render() {return this.props.type === 'mini' ? this.renderCircle() : this.renderCloud();};_proto.renderCircle = function renderCircle() {var dimmed = this.props.dimmed;var frame = this.state.frame;var cssSet = { backgroundImage: "url('" + this.imageUrls[dimmed ? 'dimmed' : 'mini'] + "')", height: 16, width: 16, marginBottom: -3, marginLeft: -1, marginRight: -1 };if (!process.env.enableReactTesting) {cssSet.backgroundPosition = "0 -" + frame * 16 + "px";}return _react.default.createElement("span", { className: _SpinnerOld.jsStyles.fallback(), style: cssSet });};_proto.renderCloud = function renderCloud() {var type = this.props.type;var frame = this.state.frame;var multiply = type === 'big' ? 2 : 1;var cssSet = { backgroundImage: "url('" + this.imageUrls[type] + "')", height: _SpinnerOldIcon.SPINNER_CLOUD_SIZE.height * multiply, top: 0, width: _SpinnerOldIcon.SPINNER_CLOUD_SIZE.width * multiply };if (!process.env.enableReactTesting) {cssSet.backgroundPosition = "0 -" + frame * _SpinnerOldIcon.SPINNER_CLOUD_SIZE.height * multiply + "px";}return _react.default.createElement("span", { className: _SpinnerOld.jsStyles.fallback(), style: cssSet });};return SpinnerOldFallback;}(_react.default.Component);exports.SpinnerOldFallback = SpinnerOldFallback;SpinnerOldFallback.__KONTUR_REACT_UI__ = 'SpinnerOldFallback';SpinnerOldFallback.propTypes = { type: PropTypes.oneOf(Object.keys(types)), dimmed: PropTypes.bool };
