// @skbkontur/react-ui/cjs/internal/SpinnerOld/SpinnerOld.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");

var _AnimationKeyframes = require("../../lib/theming/AnimationKeyframes");function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: relative;\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: block;\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-left: 5px;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      text-align: center;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-size: 14px;\n      color: ", ";\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite,\n        ", " 6s ease-in-out infinite;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke-dasharray: 20, 86;\n      stroke-dashoffset: 15;\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke-dasharray: 20, 86;\n      stroke-dashoffset: 15;\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 6s ease-in-out infinite;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      fill: none;\n      stroke-linecap: round;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      stroke: ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  cloudStroke: function cloudStroke(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.spinnerBgColor);

  },
  cloudWrapper: function cloudWrapper() {
    return (0, _Emotion.css)(_templateObject2());



  },
  cloud: function cloud(t) {
    return (0, _Emotion.css)(_templateObject3(),


    t.red,
    _AnimationKeyframes.AnimationKeyframes.spinnerCloudOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCloudLength(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerColor(t));

  },
  cloudDimmed: function cloudDimmed(t) {
    return (0, _Emotion.css)(_templateObject4(),


    t.spinnerDimmedColor,
    _AnimationKeyframes.AnimationKeyframes.spinnerCloudOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCloudLength(t));

  },
  circle: function circle(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.red,
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleLength(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleRotate(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerColor(t));

  },
  circleDimmed: function circleDimmed(t) {
    return (0, _Emotion.css)(_templateObject6(),
    t.spinnerDimmedColor,
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleOffset(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleLength(t),
    _AnimationKeyframes.AnimationKeyframes.spinnerCircleRotate(t));

  },

  caption: function caption(t) {
    return (0, _Emotion.css)(_templateObject7(),

    t.spinnerCaptionColor);

  },

  spinner: function spinner() {
    return (0, _Emotion.css)(_templateObject8());



  },

  inner: function inner() {
    return (0, _Emotion.css)(_templateObject9());


  },

  captionRight: function captionRight() {
    return (0, _Emotion.css)(_templateObject10());


  },

  captionBottom: function captionBottom() {
    return (0, _Emotion.css)(_templateObject11());


  },

  fallback: function fallback() {
    return (0, _Emotion.css)(_templateObject12());



  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
