// @skbkontur/react-ui/components/Spinner/Spinner.styles/Spinner.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: relative;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      text-align: center;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      display: block;\n      font-size: 16px;\n      line-height: 1.375;\n      margin-top: -6px;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      display: block;\n      font-size: 18px;\n      line-height: 1.33;\n      margin-top: -8px;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-left: 8px;\n      font-size: 14px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite,\n        ", " 6s ease-in-out infinite;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
import { AnimationKeyframes } from "../../../lib/theming/AnimationKeyframes";
var styles = {
  circle: function circle(t) {
    return css(_templateObject(), t.red, AnimationKeyframes.spinnerCircleOffset(t), AnimationKeyframes.spinnerCircleLength(t), AnimationKeyframes.spinnerCircleRotate(t), AnimationKeyframes.spinnerColor(t));
  },
  circleDimmed: function circleDimmed(t) {
    return css(_templateObject2(), t.spinnerDimmedColor, AnimationKeyframes.spinnerCircleOffset(t), AnimationKeyframes.spinnerCircleLength(t), AnimationKeyframes.spinnerCircleRotate(t));
  },
  captionColor: function captionColor(t) {
    return css(_templateObject3(), t.spinnerCaptionColor);
  },
  mini: function mini() {
    return css(_templateObject4());
  },
  big: function big() {
    return css(_templateObject5());
  },
  normal: function normal() {
    return css(_templateObject6());
  },
  spinner: function spinner() {
    return css(_templateObject7());
  },
  inner: function inner() {
    return css(_templateObject8());
  },
  fallback: function fallback() {
    return css(_templateObject9());
  }
};
export var jsStyles = memoizeStyle(styles);
