// @skbkontur/react-ui/internal/DateSelect/DateSelect.styles/DateSelect.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n\n      span {\n        position: relative;\n        top: -1px;\n      }\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n\n      span {\n        position: relative;\n        top: -0.5px;\n      }\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      box-sizing: border-box;\n      cursor: pointer;\n      height: 16px;\n      line-height: 0;\n      padding-left: 18px;\n      text-align: left;\n      user-select: none;\n\n      background: ", ";\n      color: ", ";\n\n      &:hover {\n        background: ", ";\n        color: ", ";\n      }\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      color: ", ";\n      pointer-events: none;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      color: ", ";\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      color: ", ";\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      cursor: pointer;\n      height: 24px;\n      line-height: 24px;\n      padding-left: 10px;\n      user-select: none;\n      white-space: nowrap;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      overflow: hidden;\n      position: relative;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-bottom: 5px;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-top: 0;\n      padding-top: 5px;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      box-shadow: ", ";\n      box-sizing: content-box;\n      color: ", ";\n      font-size: 14px;\n      margin-top: -11px;\n      overflow: hidden;\n      position: absolute;\n      z-index: 1000;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      opacity: 0;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      float: right;\n      line-height: 24px;\n      opacity: 1;\n      position: absolute;\n      right: 4px;\n      top: 0;\n      transform: scaleX(0.7);\n      transition: opacity 0.2s ease-out;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      color: inherit;\n      cursor: default;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      font-size: 14px;\n      line-height: 24px;\n      outline: 0;\n      padding-right: 2px;\n      position: relative;\n      text-align: left;\n\n      &:hover {\n        color: ", ";\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.textColorDefault, t.linkColor);
  },
  disabled: function disabled() {
    return css(_templateObject2());
  },
  caption: function caption() {
    return css(_templateObject3());
  },
  arrow: function arrow(t) {
    return css(_templateObject4(), t.textColorDisabled);
  },
  arrowDisabled: function arrowDisabled() {
    return css(_templateObject5());
  },
  menuHolder: function menuHolder(t) {
    return css(_templateObject6(), t.dateSelectMenuBg, t.popupBoxShadow, t.textColorDefault);
  },
  isTopCapped: function isTopCapped() {
    return css(_templateObject7());
  },
  isBotCapped: function isBotCapped() {
    return css(_templateObject8());
  },
  itemsHolder: function itemsHolder() {
    return css(_templateObject9());
  },
  menuItem: function menuItem(t) {
    return css(_templateObject10(), t.textColorDefault);
  },
  menuItemSelected: function menuItemSelected(t) {
    return css(_templateObject11(), t.dateSelectMenuItemBgSelected, t.dateSelectMenuItemFontSelected);
  },
  menuItemActive: function menuItemActive(t) {
    return css(_templateObject12(), t.dateSelectMenuItemBgActive, t.dateSelectMenuItemFontActive);
  },
  menuItemDisabled: function menuItemDisabled(t) {
    return css(_templateObject13(), t.dateSelectMenuItemBgDisabled, t.textColorDisabled);
  },
  menu: function menu(t) {
    return css(_templateObject14(), t.dateSelectMenuBg, t.textColorDefault, t.dateSelectMenuItemBgActive, t.textColorInvert);
  },
  menuUp: function menuUp(t) {
    return css(_templateObject15(), styles.menu(t));
  },
  menuDown: function menuDown(t) {
    return css(_templateObject16(), styles.menu(t));
  }
};
export var jsStyles = memoizeStyle(styles);
