// @skbkontur/react-ui/cjs/components/Radio/Radio.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: #a0a0a0;\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: ", ";\n      line-height: 20px;\n      margin-left: 9px;\n      white-space: normal;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", " !important;\n      border-color: transparent !important;\n      box-shadow: ", " !important;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      height: 0;\n      opacity: 0;\n      position: absolute;\n      width: 0;\n      z-index: -1;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: relative;\n      background-color: ", ";\n\n      &::before {\n        content: ' ';\n        position: absolute;\n        top: 0;\n        right: 0;\n        bottom: 0;\n        left: 0;\n        margin: auto;\n        height: 8px;\n        width: 8px;\n        border-radius: 50%;\n        background: ", " !important;\n      }\n\n      ", "&::before {\n        background: ", " !important;\n      }\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background-image: ", ";\n      border-radius: 50%;\n      border: ", ";\n      box-shadow: ", ";\n      box-sizing: border-box;\n      display: inline-block;\n      height: ", ";\n      margin-bottom: 2px;\n      margin-top: 2px;\n      position: relative;\n      vertical-align: ", ";\n      width: ", ";\n\n      ", ":hover &:not(", ") {\n        background: ", ";\n        box-shadow: ", ";\n      }\n      ", ":hover &", ":not(", ") {\n        background: ", ";\n      }\n      ", ":active & {\n        background: ", ";\n        box-shadow: ", ";\n      }\n      ", ":focus + &::after {\n        ", ";\n        box-shadow: ", ";\n        border-color: ", ";\n      }\n      &::after {\n        content: ' ';\n        display: inline-block;\n      }\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      content: ' ';\n      position: absolute;\n      left: -2px;\n      top: -2px;\n      width: 20px;\n      height: 20px;\n      border-width: 2px;\n      border-style: solid;\n      border-radius: 50%;\n      box-sizing: border-box;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: pointer;\n      position: relative;\n      white-space: nowrap;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject());




  },

  after: function after() {
    return (0, _Emotion.css)(_templateObject2());











  },

  radio: function radio(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.radioBgImage,

    t.radioBorder,
    t.radioBoxShadow,


    t.radioSize,



    t.radioVerticalAlign,
    t.radioSize,

    (0, _Emotion.cssName)(styles.root()), (0, _Emotion.cssName)(styles.disabled(t)),
    t.radioHoverBg,
    t.radioHoverShadow,

    (0, _Emotion.cssName)(styles.root()), (0, _Emotion.cssName)(styles.checked(t)), (0, _Emotion.cssName)(styles.disabled(t)),
    t.radioCheckedHoverBgColor,

    (0, _Emotion.cssName)(styles.root()),
    t.radioActiveBg,
    t.radioActiveShadow,

    (0, _Emotion.cssName)(styles.input()),
    styles.after(),
    t.radioFocusShadow,
    t.borderColorFocus);






  },

  focus: function focus(t) {
    return (0, _Emotion.css)(_templateObject4(),

    styles.after(),
    t.radioFocusShadow,
    t.borderColorFocus);


  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject5(),

    styles.after(),
    t.radioFocusShadow,
    t.borderColorWarning);


  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject6(),

    styles.after(),
    t.radioFocusShadow,
    t.borderColorError);


  },

  checked: function checked(t) {
    return (0, _Emotion.css)(_templateObject7(),

    t.radioCheckedBgColor,












    t.radioCheckedBulletColor,


    (0, _Emotion.cssName)(styles.disabled(t)),
    t.gray);


  },

  input: function input() {
    return (0, _Emotion.css)(_templateObject8());







  },

  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject9(),
    t.bgDisabled,

    t.radioDisabledShadow);

  },

  label: function label(t) {
    return (0, _Emotion.css)(_templateObject10(),
    t.radioLabelDisplay);




  },

  labelDisabled: function labelDisabled() {
    return (0, _Emotion.css)(_templateObject11());


  },

  placeholder: function placeholder() {
    return (0, _Emotion.css)(_templateObject12());


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
