// @skbkontur/react-ui/cjs/components/Radio/Radio.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.Radio = void 0;var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _classnames = _interopRequireDefault(require("classnames"));


var _ThemeContext = require("../../lib/theming/ThemeContext");


var _Radio = require("./Radio.styles");var


























Radio = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(Radio, _React$Component);function Radio() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.















    theme = void 0;_this.
    inputEl = _react.default.createRef();_this.

































































































    _isInRadioGroup = function () {return Boolean(_this.context.name);};_this.










    handleChange = function (e) {
      _this.props.onValueChange == null ? void 0 : _this.props.onValueChange(_this.props.value);

      if (_this._isInRadioGroup()) {
        _this.context.onSelect(_this.props.value);
      }

      _this.props.onChange == null ? void 0 : _this.props.onChange(e);
    };_this.

    handleMouseOver = function (e) {
      _this.props.onMouseOver == null ? void 0 : _this.props.onMouseOver(e);
    };_this.

    handleMouseEnter = function (e) {
      _this.props.onMouseEnter == null ? void 0 : _this.props.onMouseEnter(e);
    };_this.

    handleMouseLeave = function (e) {
      _this.props.onMouseLeave == null ? void 0 : _this.props.onMouseLeave(e);
    };return _this;}var _proto = Radio.prototype;_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});} /**
                                                                                                                                                                                                                                                             * @public
                                                                                                                                                                                                                                                             */;_proto.focus = function focus() {var _this$inputEl$current;(_this$inputEl$current = this.inputEl.current) == null ? void 0 : _this$inputEl$current.focus();} /**
                                                                                                                                                                                                                                                                                                                                                                                                                              * @public
                                                                                                                                                                                                                                                                                                                                                                                                                              */;_proto.blur = function blur() {var _this$inputEl$current2;(_this$inputEl$current2 = this.inputEl.current) == null ? void 0 : _this$inputEl$current2.blur();};_proto.renderMain = function renderMain() {var _cn;var _this$props = this.props,active = _this$props.active,children = _this$props.children,_this$props$disabled = _this$props.disabled,disabled = _this$props$disabled === void 0 ? this.context.disabled : _this$props$disabled,_this$props$warning = _this$props.warning,warning = _this$props$warning === void 0 ? this.context.warning : _this$props$warning,_this$props$error = _this$props.error,error = _this$props$error === void 0 ? this.context.error : _this$props$error,focused = _this$props.focused,pressed = _this$props.pressed,hovered = _this$props.hovered,onMouseOver = _this$props.onMouseOver,onMouseEnter = _this$props.onMouseEnter,onMouseLeave = _this$props.onMouseLeave,onValueChange = _this$props.onValueChange,className = _this$props.className,style = _this$props.style,rest = (0, _objectWithoutPropertiesLoose2.default)(_this$props, ["active", "children", "disabled", "warning", "error", "focused", "pressed", "hovered", "onMouseOver", "onMouseEnter", "onMouseLeave", "onValueChange", "className", "style"]);var radioClassNames = (0, _classnames.default)((_cn = {}, _cn[_Radio.jsStyles.radio(this.theme)] = true, _cn[_Radio.jsStyles.checked(this.theme)] = this.props.checked, _cn[_Radio.jsStyles.focus(this.theme)] = this.props.focused, _cn[_Radio.jsStyles.error(this.theme)] = error, _cn[_Radio.jsStyles.warning(this.theme)] = warning, _cn[_Radio.jsStyles.disabled(this.theme)] = disabled, _cn));var value;if (typeof this.props.value === 'string' || typeof this.props.value === 'number') {value = this.props.value;}var inputProps = (0, _extends2.default)({}, rest, { type: 'radio', className: _Radio.jsStyles.input(), disabled: disabled, tabIndex: this.props.tabIndex, value: value, ref: this.inputEl, onChange: this.handleChange });var labelProps = { className: _Radio.jsStyles.root(), onMouseOver: this.handleMouseOver, onMouseEnter: this.handleMouseEnter, onMouseLeave: this.handleMouseLeave };if (this._isInRadioGroup()) {var checked = this.props.value === this.context.activeItem;inputProps.checked = checked;inputProps.name = this.context.name;radioClassNames = (0, _classnames.default)(radioClassNames, checked && _Radio.jsStyles.checked(this.theme));}return _react.default.createElement("label", labelProps, _react.default.createElement("input", inputProps), _react.default.createElement("span", { className: radioClassNames }, _react.default.createElement("span", { className: _Radio.jsStyles.placeholder() })), this.props.children && this.renderLabel());};_proto.renderLabel = function renderLabel() {var _cn2;var labelClassNames = (0, _classnames.default)((_cn2 = {}, _cn2[_Radio.jsStyles.label(this.theme)] = true, _cn2[_Radio.jsStyles.labelDisabled()] = !!(this.props.disabled || this.context.disabled), _cn2));return _react.default.createElement("div", { className: labelClassNames }, this.props.children);};return Radio;}(_react.default.Component);exports.Radio = Radio;Radio.__KONTUR_REACT_UI__ = 'Radio';Radio.contextTypes = { activeItem: _propTypes.default.any, onSelect: _propTypes.default.func, name: _propTypes.default.string, disabled: _propTypes.default.bool, error: _propTypes.default.bool, warning: _propTypes.default.bool };Radio.defaultProps = { focused: false };
