// @skbkontur/react-ui/lib/theming/AnimationKeyframes/AnimationKeyframes.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n        100%, 0% { stroke: ", "; }\n        40% { stroke: ", "; }\n        66% { stroke: ", "; }\n        80%, 90% { stroke: ", "; }\n      "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n        100% { transform: rotate(360deg); }\n      "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n        0% { stroke-dasharray: 62.5%, 168.75%; }\n        50% {  stroke-dasharray: 187.5%, 43.75%; }\n        100% { stroke-dasharray: 62.5%, 168.75%; }\n      "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n        0% { stroke-dashoffset: 0; }\n        100% { stroke-dashoffset: -231.25%; }\n      "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n        0% { stroke-dasharray: 10, 96; }\n        50% { stroke-dasharray: 50, 56; }\n        100% { stroke-dasharray: 10, 96;}\n      "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n        0% { stroke-dashoffset: 10; }\n        100% { stroke-dashoffset: 116; }\n      "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { keyframes } from "../Emotion";
export var AnimationKeyframes = {
  spinnerCloudOffset: function spinnerCloudOffset(t) {
    return keyframes(_templateObject());
  },
  spinnerCloudLength: function spinnerCloudLength(t) {
    return keyframes(_templateObject2());
  },
  spinnerCircleOffset: function spinnerCircleOffset(t) {
    return keyframes(_templateObject3());
  },
  spinnerCircleLength: function spinnerCircleLength(t) {
    return keyframes(_templateObject4());
  },
  spinnerCircleRotate: function spinnerCircleRotate(t) {
    return keyframes(_templateObject5());
  },
  spinnerColor: function spinnerColor(t) {
    return keyframes(_templateObject6(), t.red, t.yellow, t.green, t.brand);
  }
};
