// @skbkontur/react-ui/cjs/components/Link/Link.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");


var _Link = require("./Link.mixins");function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      margin-right: ", ";\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", "& {\n        color: ", ";\n\n        &:hover {\n          color: ", ";\n        }\n      }\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        text-decoration: ", ";\n      }\n      ", "& {\n        color: ", ";\n      }\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 4px solid transparent;\n      border-bottom-width: 0;\n      border-top-color: #a0a0a0;\n      display: inline-block;\n      margin-bottom: 3px;\n      margin-left: 3px;\n      vertical-align: middle;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: #e1e1e1;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      line-height: 34px;\n      padding-left: 10px;\n      padding-right: 10px;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}

var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),
    (0, _Link.linkMixin)());

  },

  button: function button() {
    return (0, _Emotion.css)(_templateObject2());





  },

  buttonOpened: function buttonOpened() {
    return (0, _Emotion.css)(_templateObject3());


  },

  arrow: function arrow() {
    return (0, _Emotion.css)(_templateObject4());








  },

  useDefault: function useDefault(t) {
    return (0, _Emotion.css)(_templateObject5(),
    (0, _Link.linkUseColorsMixin)(t.linkColor, t.linkHoverColor, t.linkActiveColor));

  },

  useSuccess: function useSuccess(t) {
    return (0, _Emotion.css)(_templateObject6(),
    (0, _Link.linkUseColorsMixin)(t.linkSuccessColor, t.linkSuccessHoverColor, t.linkSuccessActiveColor));

  },

  useDanger: function useDanger(t) {
    return (0, _Emotion.css)(_templateObject7(),
    (0, _Link.linkUseColorsMixin)(t.linkDangerColor, t.linkDangerHoverColor, t.linkDangerActiveColor));

  },

  useGrayed: function useGrayed(t) {
    return (0, _Emotion.css)(_templateObject8(),
    (0, _Link.linkUseColorsMixin)(t.linkDisabledColor, t.linkDisabledColor, t.linkDisabledColor));

  },

  focus: function focus(t) {
    return (0, _Emotion.css)(_templateObject9(),
    (0, _Emotion.cssName)(styles.root(t)),
    t.linkHoverTextDecoration,

    (0, _Emotion.cssName)(styles.useGrayed(t)),
    t.linkDisabledColor);


  },

  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject10(),
    (0, _Link.linkDisabledMixin)(),

    (0, _Emotion.cssName)(styles.root(t)),
    t.linkDisabledColor,


    t.linkDisabledColor);



  },

  icon: function icon(t) {
    return (0, _Emotion.css)(_templateObject11(),

    t.linkIconMarginRight);

  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
