// @skbkontur/react-ui/cjs/components/Toggle/Toggle.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      opacity: 0.3;\n      cursor: default;\n\n      ", " {\n        background: ", ";\n      }\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: pointer;\n      display: inline-block;\n      height: 20px;\n      position: relative;\n      width: 34px;\n\n      &:hover:not(", ") {\n        ", " {\n          background: ", ";\n        }\n      }\n      &::after {\n        content: '';\n        display: inline-block;\n      }\n      &:active:not(", ") ", " {\n        width: 22px;\n      }\n      &:active:not(", ") ", ":checked ~ ", " {\n        transform: translateX(10px);\n      }\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ":checked ~ ", "& {\n        background: ", ";\n        box-shadow: inset 0 0 0 1px ", ";\n\n        ", " {\n          background: ", ";\n        }\n      }\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ":checked ~ ", "& {\n        background: ", ";\n        box-shadow: inset 0 0 0 1px ", ";\n\n        ", " {\n          background: ", ";\n        }\n      }\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ":checked ~ ", "& {\n        background: ", ";\n        box-shadow: inset 0 0 0 1px ", ";\n      }\n      ", " {\n        background: ", ";\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: inherit;\n      bottom: 0;\n      left: 0;\n      position: absolute;\n      top: 0;\n      transition: 0.2s ease-in;\n      width: 10px;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "&, ", ":checked ~ & {\n        box-shadow: 0 0 0 1px ", ", 0 0 0 3px ", ";\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-radius: 10px;\n      box-shadow: inset 0 0 0 1px ", ";\n      height: 100%;\n      overflow: hidden;\n      position: absolute;\n      width: 100%;\n      /* fixes overflow issue in Safari: https://bugs.webkit.org/show_bug.cgi?id=98538 */\n      z-index: 0;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      opacity: 0;\n\n      &:focus {\n        outline: none;\n      }\n      &:checked ~ ", " {\n        box-shadow: inset 0 0 0 1px ", ";\n        background: ", ";\n        transition: background 0s 0.2s;\n      }\n      &:checked ~ ", " ", " {\n        width: 70%;\n        background: ", ";\n      }\n      &:checked ~ ", " {\n        transform: translateX(14px);\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      border-radius: 9px;\n      bottom: 1px;\n      box-shadow: 0 1px 0 0 rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(0, 0, 0, 0.15);\n      height: 18px;\n      left: 1px;\n      position: absolute;\n      top: 1px;\n      transition: 0.2s ease-in;\n      width: 18px;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  handle: function handle(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.toggleBg);










  },

  input: function input(t) {
    return (0, _Emotion.css)(_templateObject2(),






    (0, _Emotion.cssName)(styles.container(t)),
    t.toggleBgChecked,
    t.toggleBgChecked,


    (0, _Emotion.cssName)(styles.container(t)), (0, _Emotion.cssName)(styles.activeBackground()),

    t.toggleBgChecked,

    (0, _Emotion.cssName)(styles.handle(t)));



  },

  container: function container(t) {
    return (0, _Emotion.css)(_templateObject3(),

    t.toggleBorderColor);







  },

  focused: function focused(t) {
    return (0, _Emotion.css)(_templateObject4(),
    (0, _Emotion.cssName)(styles.container(t)), (0, _Emotion.cssName)(styles.input(t)),
    t.outlineColorFocus, t.toggleFocusShadowColor);


  },

  activeBackground: function activeBackground() {
    return (0, _Emotion.css)(_templateObject5());








  },

  isLoading: function isLoading(t) {
    return (0, _Emotion.css)(_templateObject6(),
    (0, _Emotion.cssName)(styles.input(t)), (0, _Emotion.cssName)(styles.container(t)),
    t.toggleBorderColor,
    t.toggleBorderColor,

    (0, _Emotion.cssName)(styles.activeBackground()),
    t.toggleBgActive);


  },

  isWarning: function isWarning(t) {
    return (0, _Emotion.css)(_templateObject7(),
    (0, _Emotion.cssName)(styles.input(t)), (0, _Emotion.cssName)(styles.container(t)),
    t.toggleBgWarning,
    t.toggleBgWarning,

    (0, _Emotion.cssName)(styles.activeBackground()),
    t.toggleBgWarning);



  },

  isError: function isError(t) {
    return (0, _Emotion.css)(_templateObject8(),
    (0, _Emotion.cssName)(styles.input(t)), (0, _Emotion.cssName)(styles.container(t)),
    t.toggleBgError,
    t.toggleBgError,

    (0, _Emotion.cssName)(styles.activeBackground()),
    t.toggleBgError);



  },

  wrapper: function wrapper(t) {
    var wrapperDisabled = (0, _Emotion.cssName)(styles.wrapperDisabled(t));
    return (0, _Emotion.css)(_templateObject9(),






    wrapperDisabled,
    (0, _Emotion.cssName)(styles.handle(t)),
    t.toggleBgHover,






    wrapperDisabled, (0, _Emotion.cssName)(styles.handle(t)),


    wrapperDisabled, (0, _Emotion.cssName)(styles.input(t)), (0, _Emotion.cssName)(styles.handle(t)));



  },

  wrapperDisabled: function wrapperDisabled(t) {
    return (0, _Emotion.css)(_templateObject10(),



    (0, _Emotion.cssName)(styles.container(t)),
    t.toggleBgDisabled);


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
