// @skbkontur/react-ui/cjs/internal/ThemeShowcase/ThemeShowcase.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.ThemeShowcase = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));


var _DefaultTheme = require("../../lib/theming/themes/DefaultTheme");
var _FlatTheme = require("../../lib/theming/themes/FlatTheme");
var _ComboBox = require("../../components/ComboBox");
var _Gapped = require("../../components/Gapped");
var _Link = require("../../components/Link");
var _Sticky = require("../../components/Sticky");
var ColorFunctions = _interopRequireWildcard(require("../../lib/styles/ColorFunctions"));
var _Tooltip = require("../../components/Tooltip");
var _Supports = require("../../lib/Supports");

var _VariablesCollector = require("./VariablesCollector");








var _ThemeShowcase = require("./ThemeShowcase.styles");

var EMPTY_ARRAY = [];

var ALL_VARIABLES = Object.keys(_DefaultTheme.DEFAULT_THEME);var








ThemeShowcase = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(ThemeShowcase, _React$Component);function ThemeShowcase() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.
    state = {};_this.

    isUnmounting = false;_this.
    variablesDiff = [];_this.


















































































    getItems = function (query) {
      return Promise.resolve(_this.getValues(query));
    };_this.
    handleVariableChange = function (item) {
      if (!_this.isUnmounting) {
        _this.setState({ selectedVariable: item });
      }
    };_this.
    handleUnexpectedVariableInput = function (query) {
      var values = _this.getValues(query);
      if (values.length > 0) {
        return values[0];
      } else {
        return _this.resetVariable();
      }
    };_this.
    resetVariable = function () {
      if (!_this.isUnmounting) {
        _this.setState({ selectedVariable: undefined });
      }
    };return _this;}var _proto = ThemeShowcase.prototype;_proto.UNSAFE_componentWillMount = function UNSAFE_componentWillMount() {var _this2 = this;if (this.props.isDebugMode) {ALL_VARIABLES.forEach(function (variable) {var found = _VariablesCollector.ALL_USED_VARIABLES.includes(variable);if (!found) {_this2.variablesDiff.push(variable);}});}};_proto.render = function render() {var _this3 = this;if (!_Supports.IS_PROXY_SUPPORTED) {return _react.default.createElement("div", null, "\u0422\u0430\u0431\u043B\u0438\u0446\u0430 \u0438\u0441\u043F\u043E\u043B\u044C\u0437\u043E\u0432\u0430\u043D\u0438\u044F \u043F\u0435\u0440\u0435\u043C\u0435\u043D\u043D\u044B\u0445 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u0430 \u0442\u043E\u043B\u044C\u043A\u043E \u0432", ' ', _react.default.createElement(_Link.Link, { href: 'https://caniuse.com/#feat=proxy', target: '_blank' }, "\u0431\u0440\u0430\u0443\u0437\u0435\u0440\u0430\u0445 \u0441 \u043F\u043E\u0434\u0434\u0435\u0440\u0436\u043A\u043E\u0439 Proxy"), ".");}var selectedVariable = this.state.selectedVariable;var descriptionsToRender = selectedVariable ? _VariablesCollector.COMPONENT_DESCRIPTIONS_BY_VARIABLE[selectedVariable.value] || {} : _VariablesCollector.COMPONENT_DESCRIPTIONS;var isDebugMode = this.props.isDebugMode;var callsCount = isDebugMode ? "(" + _VariablesCollector.CALLS_COUNT + " \u0432\u044B\u0437\u043E\u0432\u043E\u0432)" : '';var executionTime = isDebugMode ? "\u0421\u0433\u0435\u043D\u0435\u0440\u0438\u0440\u043E\u0432\u0430\u043D\u043E \u0437\u0430 " + _VariablesCollector.EXECUTION_TIME.toFixed(3) + "ms" : '';return _react.default.createElement(_Gapped.Gapped, { wrap: true, gap: 30, verticalAlign: 'top' }, _react.default.createElement("div", null, _react.default.createElement(_Sticky.Sticky, { side: 'top' }, _react.default.createElement("div", { className: _ThemeShowcase.jsStyles.searchBar(), "data-perf-info": executionTime + " " + callsCount }, _react.default.createElement(_Gapped.Gapped, { gap: 15 }, _react.default.createElement(_ComboBox.ComboBox, { getItems: this.getItems, value: selectedVariable, onValueChange: this.handleVariableChange, onUnexpectedInput: this.handleUnexpectedVariableInput, placeholder: 'поиск по названию переменной' }), !!selectedVariable && _react.default.createElement(_Link.Link, { onClick: this.resetVariable }, "\u0441\u0431\u0440\u043E\u0441\u0438\u0442\u044C")))), Object.keys(descriptionsToRender).sort().map(function (componentName) {return _react.default.createElement(ComponentShowcase, { key: componentName, name: componentName, description: descriptionsToRender[componentName], isDebugMode: isDebugMode, onVariableSelect: _this3.handleVariableChange });})), _react.default.createElement(ShowUnusedVariables, { diff: this.variablesDiff }));};_proto.componentWillUnmount = function componentWillUnmount() {this.isUnmounting = true;};_proto.getValues = function getValues(query) {var lowerCaseQuery = query && query.toLowerCase().trim();var allItems = _VariablesCollector.ALL_USED_VARIABLES;if (lowerCaseQuery) {allItems = _VariablesCollector.ALL_USED_VARIABLES.filter(function (usedVariable) {return usedVariable.toLowerCase().startsWith(lowerCaseQuery);});}return allItems.map(function (usedVariableName) {return { value: usedVariableName, label: usedVariableName };});};return ThemeShowcase;}(_react.default.Component);exports.ThemeShowcase = ThemeShowcase;var








ComponentShowcase = /*#__PURE__*/function (_React$Component2) {(0, _inheritsLoose2.default)(ComponentShowcase, _React$Component2);function ComponentShowcase() {return _React$Component2.apply(this, arguments) || this;}var _proto2 = ComponentShowcase.prototype;_proto2.
  render = function render() {var _this4 = this;var _this$props =
    this.props,name = _this$props.name,description = _this$props.description,onVariableSelect = _this$props.onVariableSelect,isDebugMode = _this$props.isDebugMode;
    var elements = Object.keys(description);

    return (
      _react.default.createElement(_react.default.Fragment, null,
      _react.default.createElement(_Sticky.Sticky, { side: 'top', offset: 40 },
      function (isSticky) {var _cn;return (
          _react.default.createElement("h2", {
            className: (0, _classnames.default)((_cn = {}, _cn[
            _ThemeShowcase.jsStyles.heading()] = true, _cn[
            _ThemeShowcase.jsStyles.headingSticky()] = isSticky, _cn)) },


          _this4.props.name));}),



      _react.default.createElement("table", { className: _ThemeShowcase.jsStyles.table() },
      _react.default.createElement("thead", null,
      _react.default.createElement("tr", null,
      _react.default.createElement("th", { style: { width: 170 } }, "ClassName"),
      _react.default.createElement("th", { style: { width: 210 } }, "Variable Name"),
      _react.default.createElement("th", { style: { width: 250 } }, "Default Value"),
      _react.default.createElement("th", { style: { width: 250 } }, "Flat Value"))),


      _react.default.createElement("tbody", null,
      elements.map(function (el) {return (
          _react.default.createElement(ComponentShowcaseRow, {
            key: name + "_" + el,
            element: el,
            row: description[el],
            onVariableSelect: onVariableSelect,
            isDebugMode: isDebugMode }));})))));






  };return ComponentShowcase;}(_react.default.Component);var









ComponentShowcaseRow = /*#__PURE__*/function (_React$Component3) {(0, _inheritsLoose2.default)(ComponentShowcaseRow, _React$Component3);function ComponentShowcaseRow() {return _React$Component3.apply(this, arguments) || this;}var _proto3 = ComponentShowcaseRow.prototype;_proto3.
  render = function render() {var _this5 = this;var _this$props2 =
    this.props,el = _this$props2.element,row = _this$props2.row,isDebugMode = _this$props2.isDebugMode;
    var rowSpan = row.variables.length + 1;

    return (
      _react.default.createElement(_react.default.Fragment, null,
      _react.default.createElement("tr", { className: _ThemeShowcase.jsStyles.invisibleRow() },
      _react.default.createElement("td", { rowSpan: rowSpan },
      _react.default.createElement("span", { className: _ThemeShowcase.jsStyles.elementName() }, ".", el)),

      _react.default.createElement("td", { className: _ThemeShowcase.jsStyles.invisibleCell() }),
      _react.default.createElement("td", { className: _ThemeShowcase.jsStyles.invisibleCell() }),
      _react.default.createElement("td", { className: _ThemeShowcase.jsStyles.invisibleCell() })),

      row.variables.map(function (varName) {var _cn2;
        var dependencies = row.dependencies[varName] || EMPTY_ARRAY;
        var variableDefault = _DefaultTheme.DEFAULT_THEME[varName];
        var variableFlat = _FlatTheme.FLAT_THEME[varName];
        var hasNoVariables = isDebugMode && !variableDefault && !variableFlat;

        return (
          _react.default.createElement("tr", { key: el + "_" + varName, className: (0, _classnames.default)((_cn2 = {}, _cn2[_ThemeShowcase.jsStyles.suspiciousRow()] = hasNoVariables, _cn2)) },
          _react.default.createElement("td", null,
          _react.default.createElement(VariableName, {
            variableName: varName,
            dependencies: dependencies,
            onVariableSelect: _this5.props.onVariableSelect })),


          _react.default.createElement("td", null,
          _react.default.createElement(VariableValue, { value: variableDefault })),

          _react.default.createElement("td", null,
          _react.default.createElement(VariableValue, { value: variableFlat }))));



      })));


  };return ComponentShowcaseRow;}(_react.default.Component);var








VariableName = /*#__PURE__*/function (_React$Component4) {(0, _inheritsLoose2.default)(VariableName, _React$Component4);function VariableName() {var _this6;for (var _len3 = arguments.length, args = new Array(_len3), _key3 = 0; _key3 < _len3; _key3++) {args[_key3] = arguments[_key3];}_this6 = _React$Component4.call.apply(_React$Component4, [this].concat(args)) || this;_this6.





























    handleVariableSelect = function () {var _this6$props =
      _this6.props,variableName = _this6$props.variableName,onVariableSelect = _this6$props.onVariableSelect;
      if (onVariableSelect) {
        onVariableSelect({ value: variableName, label: variableName });
      }
    };return _this6;}var _proto4 = VariableName.prototype;_proto4.render = function render() {return _react.default.createElement("span", null, _react.default.createElement("span", { className: _ThemeShowcase.jsStyles.variableName(), onClick: this.handleVariableSelect }, this.props.variableName), this.props.dependencies.length > 0 && this.renderDependencies());};_proto4.renderDependencies = function renderDependencies() {var _this$props3 = this.props,dependencies = _this$props3.dependencies,onVariableSelect = _this$props3.onVariableSelect;return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("br", null), _react.default.createElement("br", null), "\u0437\u0430\u0432\u0438\u0441\u0438\u0442 \u043E\u0442:", dependencies.map(function (dependency) {return _react.default.createElement(DependencyName, { key: "dependency_" + dependency, dependencyName: dependency, onDependencySelect: onVariableSelect });}));};return VariableName;}(_react.default.Component);var






DependencyName = /*#__PURE__*/function (_React$Component5) {(0, _inheritsLoose2.default)(DependencyName, _React$Component5);function DependencyName() {var _this7;for (var _len4 = arguments.length, args = new Array(_len4), _key4 = 0; _key4 < _len4; _key4++) {args[_key4] = arguments[_key4];}_this7 = _React$Component5.call.apply(_React$Component5, [this].concat(args)) || this;_this7.














    getValues = function () {
      var dependencyName = _this7.props.dependencyName;
      var dependencyDefault = _DefaultTheme.DEFAULT_THEME[dependencyName];
      var dependencyFlat = _FlatTheme.FLAT_THEME[dependencyName];
      return (
        _react.default.createElement(_react.default.Fragment, null,
        _react.default.createElement("span", null, "Default value: ", _react.default.createElement(VariableValue, { value: dependencyDefault })),
        _react.default.createElement("br", null),
        _react.default.createElement("span", null, "Flat value: ", _react.default.createElement(VariableValue, { value: dependencyFlat }))));


    };_this7.

    handleDependencySelect = function () {var _this7$props =
      _this7.props,dependencyName = _this7$props.dependencyName,onDependencySelect = _this7$props.onDependencySelect;
      if (onDependencySelect) {
        onDependencySelect({ value: dependencyName, label: dependencyName });
      }
    };return _this7;}var _proto5 = DependencyName.prototype;_proto5.render = function render() {return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("br", null), "\u2013", ' ', _react.default.createElement(_Tooltip.Tooltip, { trigger: 'hover', render: this.getValues, pos: 'right middle' }, _react.default.createElement("span", { className: _ThemeShowcase.jsStyles.variableName(), onClick: this.handleDependencySelect }, this.props.dependencyName)));};return DependencyName;}(_react.default.Component);


var VariableValue = function VariableValue(props) {var _cn3;
  var value = props.value;
  var valueIsColor = isColor(value);
  var valueIsGradient = isGradient(value);
  var hasExample = valueIsColor || valueIsGradient;
  var borderColor = 'transparent';
  if (hasExample) {
    borderColor = valueIsColor ? ColorFunctions.contrast(value) : '#000';
  }

  return (
    _react.default.createElement("span", { className: (0, _classnames.default)((_cn3 = {}, _cn3[_ThemeShowcase.jsStyles.undefined()] = !value, _cn3)) },
    hasExample && _react.default.createElement("span", { className: _ThemeShowcase.jsStyles.colorExample(), style: { background: value, borderColor: borderColor } }),
    value || 'undefined'));


};

var ShowUnusedVariables = function ShowUnusedVariables(props) {
  if (props.diff.length === 0) {
    return null;
  }

  return (
    _react.default.createElement("div", { className: _ThemeShowcase.jsStyles.unusedVariablesWarning() }, "\u041D\u0435\u0438\u0441\u043F\u043E\u043B\u044C\u0437\u043E\u0432\u0430\u043D\u043D\u044B\u0435 \u043F\u0435\u0440\u0435\u043C\u0435\u043D\u043D\u044B\u0435 (",
    props.diff.length, "):",

    _react.default.createElement("ul", null,
    props.diff.sort().map(function (v) {return (
        _react.default.createElement("li", { key: v }, v));}))));




};

function isColor(input) {
  return !!input && (input.startsWith('#') || input.startsWith('rgb') || input.startsWith('hsl'));
}

function isGradient(input) {
  return !!input && input.startsWith('linear-gradient');
}
