// @skbkontur/react-ui/internal/InputLikeText/InputLikeText.styles/InputLikeText.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 10px;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      user-select: none;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      user-select: text;\n      -ms-user-select: element;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      top: 0;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-right: 10px;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root() {
    return css(_templateObject());
  },
  input: function input() {
    return css(_templateObject2());
  },
  userSelectContain: function userSelectContain() {
    return css(_templateObject3());
  },
  userSelectNone: function userSelectNone() {
    return css(_templateObject4());
  },
  withoutLeftSide: function withoutLeftSide() {
    return css(_templateObject5());
  }
};
export var jsStyles = memoizeStyle(styles);
