// @skbkontur/react-ui/components/MenuItem/MenuItem.styles/MenuItem.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: absolute;\n      left: 15px;\n      top: 5px;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #fff;\n      opacity: 0.6;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #a0a0a0;\n      white-space: normal;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      & {\n        padding-left: ", ";\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 15px;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      background: transparent;\n      color: ", ";\n      cursor: default;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      color: ", " !important;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      cursor: pointer;\n      display: block;\n      line-height: 18px;\n      padding: 6px 18px 7px 8px;\n      position: relative;\n      text-decoration: none;\n\n      button& {\n        min-width: 100%;\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
import { resetButton } from "../../../lib/styles/Mixins";
var styles = {
  root: function root() {
    return css(_templateObject(), resetButton());
  },
  hover: function hover(t) {
    // Color with !important in purpose to override `a:hover`
    return css(_templateObject2(), t.dropdownMenuHoverBg, t.textColorInvert);
  },
  selected: function selected(t) {
    return css(_templateObject3(), t.dropdownMenuSelectedBg);
  },
  disabled: function disabled(t) {
    return css(_templateObject4(), t.textColorDisabled);
  },
  link: function link(t) {
    return css(_templateObject5(), t.linkColor);
  },
  loose: function loose() {
    return css(_templateObject6());
  },
  withIcon: function withIcon(t) {
    return css(_templateObject7(), t.menuItemPaddingForIcon);
  },
  comment: function comment() {
    return css(_templateObject8());
  },
  commentHover: function commentHover() {
    return css(_templateObject9());
  },
  icon: function icon() {
    return css(_templateObject10());
  }
};
export var jsStyles = memoizeStyle(styles);
