// @skbkontur/react-ui/components/Checkbox/Checkbox.styles/Checkbox.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      display: table-cell;\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 10px;\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      color: transparent;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-top: -2px;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      cursor: default;\n\n      & ", " {\n        box-shadow: ", " !important; //to override hover and active\n        background: ", " !important;\n        color: ", " !important;\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      & ", " {\n        box-shadow: inset 0 0 0 1px ", ", 0 0 0 ", " ", " !important;\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      & ", " {\n        background: ", ";\n        color: ", ";\n        box-shadow: ", ";\n      }\n\n      &:hover ", " {\n        box-shadow: ", ";\n        background: ", ";\n      }\n      &:active ", " {\n        box-shadow: ", ";\n        background: ", ";\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      & ", " {\n        background: ", ";\n        color: ", ";\n        box-shadow: ", ";\n      }\n\n      &:hover ", " {\n        box-shadow: ", ";\n        background: ", ";\n      }\n\n      &:active ", " {\n        box-shadow: ", ";\n        background: ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      & ", " {\n        box-shadow: inset 0 0 0 1px ", ", 0 0 0 ", " ", " !important;\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      & ", " {\n        box-shadow: inset 0 0 0 1px ", ", 0 0 0 ", " ", " !important;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      opacity: 0;\n      width: 0;\n      height: 0;\n      position: absolute;\n      z-index: -1;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-flex;\n      align-items: center;\n      justify-content: center;\n      box-sizing: border-box;\n      width: 16px;\n      height: 16px;\n      flex: none;\n      font-size: 14px;\n      color: ", ";\n      border: ", ";\n      border-radius: ", ";\n      box-shadow: ", ";\n      background: ", ";\n      align-self: baseline;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-table !important;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-flex;\n      align-items: baseline;\n      cursor: pointer;\n      position: relative;\n      user-select: none;\n      line-height: 20px;\n\n      &:hover ", " {\n        background: ", ";\n        box-shadow: ", ";\n      }\n\n      &:active ", " {\n        box-shadow: ", ";\n        background: ", ";\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), cssName(styles.box(t)), t.chbHoverBg, t.chbShadowHover, cssName(styles.box(t)), t.chbShadowActive, t.chbActiveBg);
  },
  rootFallback: function rootFallback() {
    return css(_templateObject2());
  },
  rootWrapperIE11: function rootWrapperIE11() {
    return css(_templateObject3());
  },
  box: function box(t) {
    return css(_templateObject4(), t.textColorDefault, t.chbBorder, t.chbBorderRadius, t.chbShadow, t.chbBg);
  },
  input: function input() {
    return css(_templateObject5());
  },
  warning: function warning(t) {
    return css(_templateObject6(), cssName(styles.box(t)), t.outlineColorFocus, t.chbShadowWidth, t.borderColorWarning);
  },
  error: function error(t) {
    return css(_templateObject7(), cssName(styles.box(t)), t.outlineColorFocus, t.chbShadowWidth, t.borderColorError);
  },
  checked: function checked(t) {
    return css(_templateObject8(), cssName(styles.box(t)), t.chbCheckedBg, t.chbCheckedColor, t.chbCheckedShadow, cssName(styles.box(t)), t.chbCheckedHoverShadow, t.chbCheckedHoverBg, cssName(styles.box(t)), t.chbCheckedActiveShadow, t.chbCheckedActiveBg);
  },
  indeterminate: function indeterminate(t) {
    return css(_templateObject9(), cssName(styles.box(t)), t.chbBoxIndeterminateBg, t.chbIndeterminateBg, t.chbCheckedShadow, cssName(styles.box(t)), t.chbCheckedHoverShadow, t.chbCheckedHoverBg, cssName(styles.box(t)), t.chbCheckedActiveShadow, t.chbCheckedActiveBg);
  },
  focus: function focus(t) {
    return css(_templateObject10(), cssName(styles.box(t)), t.outlineColorFocus, t.chbShadowWidth, t.chbBorderColorFocus);
  },
  disabled: function disabled(t) {
    return css(_templateObject11(), t.textColorDisabled, cssName(styles.box(t)), t.chbShadowDisabled, t.bgDisabled, t.textColorDisabled);
  },
  iconFixBaseline: function iconFixBaseline() {
    return css(_templateObject12());
  },
  iconUnchecked: function iconUnchecked() {
    return css(_templateObject13());
  },
  caption: function caption() {
    return css(_templateObject14());
  },
  captionIE11: function captionIE11() {
    return css(_templateObject15());
  }
};
export var jsStyles = memoizeStyle(styles);
