// @skbkontur/react-ui/cjs/components/Tabs/Indicator.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      height: 3px;\n      position: absolute;\n      transition: all 0.2s ease-out;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),
    t.tabColorFocus);




  },

  primary: function primary(t) {
    return (0, _Emotion.css)(_templateObject2(),
    t.tabColorPrimary);

  },

  success: function success(t) {
    return (0, _Emotion.css)(_templateObject3(),
    t.tabColorSuccess);

  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.tabColorWarning);

  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.tabColorError);

  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
