// @skbkontur/react-ui/cjs/components/Tabs/Tab.styles.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");
var ColorFunctions = _interopRequireWildcard(require("../../lib/styles/ColorFunctions"));function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:hover {\n        border-bottom-color: ", ";\n      }\n      &", ":hover {\n        border-left-color: ", ";\n      }\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:hover {\n        border-bottom-color: ", ";\n      }\n      &", ":hover {\n        border-left-color: ", ";\n      }\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:hover {\n        border-bottom-color: ", ";\n      }\n      &", ":hover {\n        border-left-color: ", ";\n      }\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:hover {\n        border-bottom-color: ", ";\n      }\n      &", ":hover {\n        border-left-color: ", ";\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: rgba(\n        ", ",\n        ", ",\n        ", ",\n        0.5\n      );\n      cursor: default;\n\n      &:hover {\n        border-bottom-color: transparent !important;\n      }\n\n      &", ":hover {\n        border-left-color: transparent !important;\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 2px solid ", ";\n      bottom: -3px;\n      left: -20px;\n      position: absolute;\n      right: -20px;\n      top: 0;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:hover {\n        cursor: default;\n        border-bottom: 3px solid transparent;\n      }\n\n      &", ":hover {\n        border-left: 3px solid transparent;\n      }\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-bottom: none;\n      border-left: 3px solid transparent;\n      display: block;\n      margin-left: 0;\n      margin-right: 0;\n      padding-left: 17px;\n      padding-right: 20px;\n\n      ", "&:hover {\n        border-bottom: none;\n        border-left: 3px solid ", ";\n      }\n\n      ", " {\n        bottom: 0;\n        left: -3px;\n        right: 0;\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-bottom: 3px solid transparent;\n      box-sizing: border-box;\n      color: inherit;\n      cursor: pointer;\n      display: inline-block;\n      font-size: 18px;\n      margin-left: 20px;\n      margin-right: 20px;\n      padding-bottom: 11px;\n      padding-top: 9px;\n      position: relative;\n      text-decoration: inherit;\n      transition: border-bottom 0.2s ease-out;\n\n      &:hover {\n        outline: inherit;\n        border-bottom: 3px solid ", ";\n      }\n\n      &:focus {\n        outline: inherit;\n      }\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),
















    t.tabColorHover);






  },

  vertical: function vertical(t) {
    return (0, _Emotion.css)(_templateObject2(),








    (0, _Emotion.cssName)(styles.root(t)),

    t.tabColorHover,


    (0, _Emotion.cssName)(styles.focus(t)));





  },

  active: function active(t) {
    return (0, _Emotion.css)(_templateObject3(),





    (0, _Emotion.cssName)(styles.vertical(t)));



  },

  focus: function focus(t) {
    return (0, _Emotion.css)(_templateObject4(),
    t.tabColorFocus);






  },

  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject5(),

    ColorFunctions.red(t.textColorDefault),
    ColorFunctions.green(t.textColorDefault),
    ColorFunctions.blue(t.textColorDefault),








    (0, _Emotion.cssName)(styles.vertical(t)));



  },

  primary: function primary(t) {
    return (0, _Emotion.css)(_templateObject6(),

    t.tabColorHoverPrimary,

    (0, _Emotion.cssName)(styles.vertical(t)),
    t.tabColorHoverPrimary);


  },

  success: function success(t) {
    return (0, _Emotion.css)(_templateObject7(),

    t.tabColorHoverSuccess,

    (0, _Emotion.cssName)(styles.vertical(t)),
    t.tabColorHoverSuccess);


  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject8(),

    t.tabColorHoverWarning,

    (0, _Emotion.cssName)(styles.vertical(t)),
    t.tabColorHoverWarning);


  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject9(),

    t.tabColorHoverError,

    (0, _Emotion.cssName)(styles.vertical(t)),
    t.tabColorHoverError);


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
