// @skbkontur/react-ui/cjs/lib/date/constants.js
"use strict";exports.__esModule = true;exports.defaultFirstDayWeek = exports.defaultDateComponentsSeparator = exports.defaultDateComponentsOrder = exports.emptyDateComponents = exports.RE_ORDER_YMD = exports.RE_ORDER_DMY = exports.RE_ORDER_MDY = exports.SEPARATOR = exports.CHAR_MASK = exports.CHAR_PAD = exports.LENGTH_FULLDATE = exports.LENGTH_SEPARATOR = exports.LENGTH_DATE = exports.LENGTH_MONTH = exports.LENGTH_YEAR = exports.MAX_FULLDATE = exports.MIN_FULLDATE = exports.MAX_DATE = exports.MIN_DATE = exports.MAX_MONTH = exports.MIN_MONTH = exports.MAX_YEAR = exports.MIN_YEAR = void 0;var _types = require("./types");

var MIN_YEAR = 0;exports.MIN_YEAR = MIN_YEAR;
var MAX_YEAR = 9999;exports.MAX_YEAR = MAX_YEAR;
var MIN_MONTH = 1;exports.MIN_MONTH = MIN_MONTH;
var MAX_MONTH = 12;exports.MAX_MONTH = MAX_MONTH;
var MIN_DATE = 1;exports.MIN_DATE = MIN_DATE;
var MAX_DATE = 31;exports.MAX_DATE = MAX_DATE;
var MIN_FULLDATE = '01.01.1900';exports.MIN_FULLDATE = MIN_FULLDATE;
var MAX_FULLDATE = '31.12.2099';exports.MAX_FULLDATE = MAX_FULLDATE;
var LENGTH_YEAR = 4;exports.LENGTH_YEAR = LENGTH_YEAR;
var LENGTH_MONTH = 2;exports.LENGTH_MONTH = LENGTH_MONTH;
var LENGTH_DATE = 2;exports.LENGTH_DATE = LENGTH_DATE;
var LENGTH_SEPARATOR = 1;exports.LENGTH_SEPARATOR = LENGTH_SEPARATOR;
var LENGTH_FULLDATE = LENGTH_YEAR + LENGTH_MONTH + LENGTH_DATE + LENGTH_SEPARATOR * 2;exports.LENGTH_FULLDATE = LENGTH_FULLDATE;
var CHAR_PAD = '0';exports.CHAR_PAD = CHAR_PAD;
var CHAR_MASK = '_';exports.CHAR_MASK = CHAR_MASK;
var SEPARATOR = "(?:\\.|\\/|\\-|\\s|\\,)";exports.SEPARATOR = SEPARATOR;
var RE_ORDER_MDY = new RegExp("^(\\d{1," +
LENGTH_MONTH + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_DATE + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_YEAR + "})?$");exports.RE_ORDER_MDY = RE_ORDER_MDY;

var RE_ORDER_DMY = new RegExp("^(\\d{1," +
LENGTH_DATE + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_MONTH + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_YEAR + "})?$");exports.RE_ORDER_DMY = RE_ORDER_DMY;

var RE_ORDER_YMD = new RegExp("^(\\d{1," +
LENGTH_YEAR + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_MONTH + "})?" + SEPARATOR + "?(\\d{1," + LENGTH_DATE + "})?$");exports.RE_ORDER_YMD = RE_ORDER_YMD;

var emptyDateComponents = {
  year: null,
  month: null,
  date: null };exports.emptyDateComponents = emptyDateComponents;

var defaultDateComponentsOrder = _types.InternalDateOrder.DMY;exports.defaultDateComponentsOrder = defaultDateComponentsOrder;
var defaultDateComponentsSeparator = _types.InternalDateSeparator.Dot;exports.defaultDateComponentsSeparator = defaultDateComponentsSeparator;
var defaultFirstDayWeek = _types.InternalDateFirstDayWeek.Monday;exports.defaultFirstDayWeek = defaultFirstDayWeek;
