// @skbkontur/react-ui/cjs/components/Button/Button.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");

var _Mixins = require("../../lib/styles/Mixins");

var _Button = require("./Button.mixins");function _templateObject39() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      border-radius: inherit;\n      overflow: hidden;\n\n      &::before {\n        content: '';\n        height: 100%;\n        position: absolute;\n        opacity: 0.2;\n        background: linear-gradient(-110deg, #ccc 30%, transparent 0, transparent 60%, #ccc 0);\n        background-size: 30px 100%;\n        top: 0;\n        left: 0;\n        right: -30px;\n        animation: ", " 1s linear infinite;\n        transform: rotateY(180deg);\n      }\n    "]);_templateObject39 = function _templateObject39() {return data;};return data;}function _templateObject38() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n    0% {\n      transform: translateX(0) rotateY(180deg);\n    }\n\n    100% {\n      transform: translateX(-30px) rotateY(180deg);\n    }\n  "]);_templateObject38 = function _templateObject38() {return data;};return data;}function _templateObject37() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:not(", "):not(", "):not(", "):not(", ") {\n        &,\n        &:hover,\n        &:active {\n          box-shadow: none !important;\n        }\n      }\n    "]);_templateObject37 = function _templateObject37() {return data;};return data;}function _templateObject36() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-right: 15px;\n      padding-left: 15px;\n    "]);_templateObject36 = function _templateObject36() {return data;};return data;}function _templateObject35() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      padding-right: 7px;\n    "]);_templateObject35 = function _templateObject35() {return data;};return data;}function _templateObject34() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      border-radius: 2px 2px 2px 16px;\n      box-sizing: border-box;\n      z-index: 1;\n    "]);_templateObject34 = function _templateObject34() {return data;};return data;}function _templateObject33() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-right: 0;\n      margin-left: 10px;\n    "]);_templateObject33 = function _templateObject33() {return data;};return data;}function _templateObject32() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin-right: 10px;\n    "]);_templateObject32 = function _templateObject32() {return data;};return data;}function _templateObject31() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      padding: 0;\n    "]);_templateObject31 = function _templateObject31() {return data;};return data;}function _templateObject30() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-right: 0 !important;\n    "]);_templateObject30 = function _templateObject30() {return data;};return data;}function _templateObject29() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 0 !important;\n      padding-right: 0 !important;\n    "]);_templateObject29 = function _templateObject29() {return data;};return data;}function _templateObject28() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 5px !important;\n      padding-right: 5px !important;\n    "]);_templateObject28 = function _templateObject28() {return data;};return data;}function _templateObject27() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding: ", ";\n      box-sizing: border-box;\n      display: inline-block;\n    "]);_templateObject27 = function _templateObject27() {return data;};return data;}function _templateObject26() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: relative;\n      white-space: nowrap;\n      display: inline-block;\n      width: 100%;\n      vertical-align: top;\n    "]);_templateObject26 = function _templateObject26() {return data;};return data;}function _templateObject25() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n    "]);_templateObject25 = function _templateObject25() {return data;};return data;}function _templateObject24() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: ", " !important;\n      background: ", " !important;\n      color: ", " !important;\n      border: ", " !important;\n\n      &:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n\n      &,\n      &:not(", ") {\n        ", " {\n          background: ", " !important;\n          box-shadow: ", " !important;\n        }\n      }\n    "]);_templateObject24 = function _templateObject24() {return data;};return data;}function _templateObject23() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);_templateObject23 = function _templateObject23() {return data;};return data;}function _templateObject22() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);_templateObject22 = function _templateObject22() {return data;};return data;}function _templateObject21() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);_templateObject21 = function _templateObject21() {return data;};return data;}function _templateObject20() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);_templateObject20 = function _templateObject20() {return data;};return data;}function _templateObject19() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);_templateObject19 = function _templateObject19() {return data;};return data;}function _templateObject18() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      visibility: visible;\n\n      ", ":not(", ") & {\n        box-shadow: ", " !important;\n      }\n    "]);_templateObject18 = function _templateObject18() {return data;};return data;}function _templateObject17() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: 2px -2px 0 0 ", " !important;\n    "]);_templateObject17 = function _templateObject17() {return data;};return data;}function _templateObject16() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-shadow: 2px -2px 0 0 ", " !important;\n    "]);_templateObject16 = function _templateObject16() {return data;};return data;}function _templateObject15() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", " {\n        box-shadow: inset 0 0 0 1px ", ";\n        border-color: transparent !important;\n      }\n    "]);_templateObject15 = function _templateObject15() {return data;};return data;}function _templateObject14() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      &", " {\n        outline-color: transparent;\n      }\n      &:not(", ") {\n        line-height: normal !important;\n      }\n    "]);_templateObject14 = function _templateObject14() {return data;};return data;}function _templateObject13() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      cursor: default;\n      pointer-events: none;\n      border-color: transparent !important;\n\n      &:not(", ") {\n        background: ", " !important;\n        color: ", " !important;\n        box-shadow: ", " !important;\n\n        ", " {\n          background: ", " !important;\n          box-shadow: ", " !important;\n        }\n      }\n\n      &", " {\n        color: ", " !important;\n      }\n    "]);_templateObject13 = function _templateObject13() {return data;};return data;}function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: relative;\n      z-index: 2;\n\n      &", " {\n        color: ", ";\n        text-decoration: ", ";\n      }\n\n      &:not(", "):not(", ") {\n        border: ", ";\n\n        &,\n        &:hover,\n        &:active,\n        ", " {\n          box-shadow: inset 0 0 0 1px ", ", 0 0 0 ", " ", ";\n\n          &", ", &", " {\n            box-shadow: inset 0 0 0 1px ", " !important;\n            border-color: transparent !important;\n          }\n          ", " {\n            box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", ";\n\n            &", " {\n              box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", " !important;\n            }\n\n            &", " {\n              box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", " !important;\n            }\n          }\n        }\n      }\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: none;\n      border-radius: ", " !important;\n      border: none;\n      box-shadow: none;\n      color: ", " !important;\n      display: inline;\n      line-height: inherit !important;\n      margin: 0;\n      padding: 0;\n\n      &:hover {\n        color: ", ";\n        text-decoration: ", ";\n      }\n      &:active {\n        color: ", " !important;\n      }\n      ", " {\n        display: inline;\n      }\n      ", " {\n        padding-right: ", ";\n      }\n      ", " ,\n      ", "  {\n        box-shadow: none;\n        left: -2px !important;\n        right: -2px !important;\n        bottom: -2px !important;\n      }\n      ", "  {\n        background: ", " !important;\n      }\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      ", ";\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-radius: ", ";\n\n      ", ";\n\n      ", " {\n        border-radius: ", ";\n      }\n\n      ", ";\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-radius: inherit;\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      box-shadow: 0 0 0 2px ", ";\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-radius: inherit;\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      box-shadow: 0 0 0 2px ", ";\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n      ", ";\n\n      background-clip: border-box;\n      background-position: center;\n      background-repeat: no-repeat;\n      background-size: contain;\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      position: relative;\n      text-align: center;\n      width: 100%;\n\n      &::-moz-focus-inner {\n        border: 0;\n        padding: 0;\n      }\n      &::after {\n        content: '';\n        display: inline-block;\n        vertical-align: baseline;\n        width: 0;\n      }\n      &:not(", ") {\n        border-radius: ", ";\n      }\n      ", "& {\n        padding: 0;\n      }\n      &:active:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n0% {\n  transform: translateX(50px) rotate(-44.3deg);\n}\n\n100% {\n  transform: translateX(21px) translateY(30px) rotate(-44.3deg);\n}\n"]);_templateObject = function _templateObject() {return data;};return data;}








var btn_loading_arrow = (0, _Emotion.keyframes)(_templateObject());









var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject2(),
    (0, _Mixins.resetButton)(),
    (0, _Mixins.resetText)(),





    t.textColorDefault,
















    (0, _Emotion.cssName)(styles.sizeSmall(t)),
    t.btnBorderRadius,

    (0, _Emotion.cssName)(styles.link(t)),


    (0, _Emotion.cssName)(styles.link(t)), (0, _Emotion.cssName)(styles.disabled(t)),
    (0, _Emotion.cssName)(styles.caption()));




  },

  warning: function warning(t) {
    return (0, _Emotion.css)(_templateObject3(),






    t.borderColorWarning);

  },

  error: function error(t) {
    return (0, _Emotion.css)(_templateObject4(),






    t.borderColorError);

  },

  sizeSmall: function sizeSmall(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.btnSmallBorderRadius,

    (0, _Button.buttonSizeMixin)(
    t.btnFontSizeSmall,
    t.controlHeightSmall,
    t.btnHeightShift,
    t.controlLineHeightSmall,
    t.btnPaddingXSmall,
    t.btnPaddingYSmall,
    (0, _Emotion.cssName)(styles.link(t)),
    (0, _Emotion.cssName)(styles.fallback(t))),


    (0, _Emotion.cssName)(styles.arrow()),
    t.btnSmallArrowBorderRadius,


    (0, _Button.buttonArrowMixin)(
    t.btnSmallArrowTop,
    t.btnSmallArrowLeft,
    t.btnSmallArrowRight,
    t.btnSmallArrowLength,
    'rotate(53deg) skewX(24deg) skewY(10deg)',
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  sizeMedium: function sizeMedium(t) {
    return (0, _Emotion.css)(_templateObject6(),
    (0, _Button.buttonSizeMixin)(
    t.btnFontSizeMedium,
    t.controlHeightMedium,
    t.btnHeightShift,
    t.controlLineHeightMedium,
    t.btnPaddingXMedium,
    t.btnPaddingYMedium,
    (0, _Emotion.cssName)(styles.link(t)),
    (0, _Emotion.cssName)(styles.fallback(t))),


    (0, _Button.buttonArrowMixin)(
    '9px',
    t.btnMediumArrowLeft,
    t.btnMediumArrowRight,
    '20.2px',
    t.btnMediumArrowTransform,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  sizeLarge: function sizeLarge(t) {
    return (0, _Emotion.css)(_templateObject7(),
    (0, _Button.buttonSizeMixin)(
    t.btnFontSizeLarge,
    t.controlHeightLarge,
    t.btnHeightShift,
    t.controlLineHeightLarge,
    t.btnPaddingXLarge,
    t.btnPaddingYLarge,
    (0, _Emotion.cssName)(styles.link(t)),
    (0, _Emotion.cssName)(styles.fallback(t))),


    (0, _Button.buttonArrowMixin)(
    '10.2px',
    t.btnLargeArrowLeft,
    '-10.8px',
    '22.2px',
    t.btnLargeArrowTransform,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  sizeSmallLoading: function sizeSmallLoading(t) {
    return (0, _Emotion.css)(_templateObject8(),
    (0, _Button.buttonLoadingArrowMixin)(
    t.btnSmallArrowTop,
    t.btnSmallArrowTop,
    '-207px',
    '441%',
    t.btnSmallArrowBg,
    t.btnSmallArrowLeftLoadingDelay,
    btn_loading_arrow,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  sizeMediumLoading: function sizeMediumLoading(t) {
    return (0, _Emotion.css)(_templateObject9(),
    (0, _Button.buttonLoadingArrowMixin)(
    '0',
    '0',
    t.btnMediumArrowLeftLoadingLeft,
    '441%',
    t.btnMediumArrowBg,
    t.btnMediumArrowLeftLoadingDelay,
    btn_loading_arrow,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  sizeLargeLoading: function sizeLargeLoading(t) {
    return (0, _Emotion.css)(_templateObject10(),
    (0, _Button.buttonLoadingArrowMixin)(
    '-32px',
    '-36px',
    ' -198px',
    '700%',
    t.btnLargeArrowBg,
    t.btnLargeArrowLeftLoadingDelay,
    btn_loading_arrow,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  link: function link(t) {
    return (0, _Emotion.css)(_templateObject11(),

    t.btnLinkBorderRadius,


    t.linkColor,






    t.linkHoverColor,
    t.linkHoverTextDecoration,


    t.linkActiveColor,

    (0, _Emotion.cssName)(styles.caption()),


    (0, _Emotion.cssName)(styles.icon()),
    t.linkIconMarginRight,

    (0, _Emotion.cssName)(styles.warning(t)),
    (0, _Emotion.cssName)(styles.error(t)),





    (0, _Emotion.cssName)(styles.error(t)),
    t.errorSecondary);


  },

  focus: function focus(t) {
    return (0, _Emotion.css)(_templateObject12(),



    (0, _Emotion.cssName)(styles.link(t)),
    t.linkColor,
    t.linkHoverTextDecoration,


    (0, _Emotion.cssName)(styles.disabled(t)), (0, _Emotion.cssName)(styles.link(t)),
    t.btnFocusBorder,




    (0, _Emotion.cssName)(styles.active(t)),
    t.outlineColorFocus, t.btnFocusShadowWidth, t.borderColorFocus,

    (0, _Emotion.cssName)(styles.warning(t)), (0, _Emotion.cssName)(styles.error(t)),
    t.outlineColorFocus,


    (0, _Emotion.cssName)(styles.arrow()),
    t.outlineColorFocus, t.borderColorFocus,

    (0, _Emotion.cssName)(styles.arrowWarning(t)),
    t.outlineColorFocus, t.borderColorWarning,


    (0, _Emotion.cssName)(styles.arrowError(t)),
    t.outlineColorFocus, t.borderColorError);





  },

  disabled: function disabled(t) {
    return (0, _Emotion.css)(_templateObject13(),




    (0, _Emotion.cssName)(styles.link(t)),
    t.btnDisabledBg,
    t.btnDisabledTextColor,
    t.btnDisabledShadow,

    (0, _Emotion.cssName)(styles.arrow()),
    t.btnDisabledBg,
    t.btnDisabledShadowArrow,



    (0, _Emotion.cssName)(styles.link(t)),
    t.linkDisabledColor);


  },

  fallback: function fallback(t) {
    return (0, _Emotion.css)(_templateObject14(),
    (0, _Emotion.cssName)(styles.disabled(t)),


    (0, _Emotion.cssName)(styles.link(t)));



  },

  validationRoot: function validationRoot(t) {
    return (0, _Emotion.css)(_templateObject15(),
    (0, _Emotion.cssName)(styles.focus(t)),
    t.outlineColorFocus);



  },

  arrowWarning: function arrowWarning(t) {
    return (0, _Emotion.css)(_templateObject16(),
    t.borderColorWarning);

  },

  arrowError: function arrowError(t) {
    return (0, _Emotion.css)(_templateObject17(),
    t.borderColorError);

  },

  arrowLeft: function arrowLeft(t) {
    return (0, _Emotion.css)(_templateObject18(),


    (0, _Emotion.cssName)(styles.checked(t)), (0, _Emotion.cssName)(styles.focus(t)),
    t.btnCheckedShadowArrowLeft);


  },

  default: function _default(t) {
    return (0, _Emotion.css)(_templateObject19(),
    (0, _Button.buttonUseMixin)(
    t.btnDefaultBg,
    t.btnDefaultBgStart,
    t.btnDefaultBgEnd,
    t.btnDefaultBgArrowStart,
    t.btnDefaultBgArrowEnd,
    t.btnDefaultShadow,
    t.btnDefaultShadowArrow,
    t.btnDefaultShadowArrowLeft,
    t.btnDefaultTextColor,
    t.btnDefaultBorder,
    (0, _Emotion.cssName)(styles.checked(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonHoverMixin)(
    t.btnDefaultHoverBg,
    t.btnDefaultHoverBgStart,
    t.btnDefaultHoverBgEnd,
    t.btnDefaultHoverBgStart,
    t.btnDefaultHoverBgEnd,
    t.btnDefaultHoverShadow,
    t.btnDefaultHoverShadowArrow,
    t.btnDefaultHoverShadowArrowLeft,
    t.btnDefaultHoverBorderColor,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonActiveMixin)(
    t.btnDefaultActiveBg,
    t.btnDefaultActiveBg,
    t.btnDefaultActiveBg,
    t.btnDefaultActiveShadow,
    t.btnDefaultActiveShadowArrow,
    t.btnDefaultActiveShadowArrowLeft,
    (0, _Emotion.cssName)(styles.active(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  primary: function primary(t) {
    return (0, _Emotion.css)(_templateObject20(),
    (0, _Button.buttonUseMixin)(
    t.btnPrimaryBg,
    t.btnPrimaryBgStart,
    t.btnPrimaryBgEnd,
    t.btnPrimaryBgArrowStart,
    t.btnPrimaryBgArrowEnd,
    t.btnPrimaryShadow,
    t.btnPrimaryShadowArrow,
    t.btnPrimaryShadowArrowLeft,
    t.btnPrimaryTextColor,
    t.btnPrimaryBorder,
    (0, _Emotion.cssName)(styles.checked(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonHoverMixin)(
    t.btnPrimaryHoverBg,
    t.btnPrimaryHoverBgStart,
    t.btnPrimaryHoverBgEnd,
    t.btnPrimaryHoverBgStart,
    t.btnPrimaryHoverBgEnd,
    t.btnPrimaryHoverShadow,
    t.btnPrimaryHoverShadowArrow,
    t.btnPrimaryHoverShadowArrowLeft,
    t.btnPrimaryHoverBorderColor,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonActiveMixin)(
    t.btnPrimaryActiveBg,
    t.btnPrimaryActiveBg,
    t.btnPrimaryActiveBg,
    t.btnPrimaryActiveShadow,
    t.btnPrimaryActiveShadowArrow,
    t.btnPrimaryActiveShadowArrowLeft,
    (0, _Emotion.cssName)(styles.active(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  success: function success(t) {
    return (0, _Emotion.css)(_templateObject21(),
    (0, _Button.buttonUseMixin)(
    t.btnSuccessBg,
    t.btnSuccessBgStart,
    t.btnSuccessBgEnd,
    t.btnSuccessBgArrowStart,
    t.btnSuccessBgArrowEnd,
    t.btnSuccessShadow,
    t.btnSuccessShadowArrow,
    t.btnSuccessShadowArrowLeft,
    t.btnSuccessTextColor,
    t.btnSuccessBorder,
    (0, _Emotion.cssName)(styles.checked(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonHoverMixin)(
    t.btnSuccessHoverBg,
    t.btnSuccessHoverBgStart,
    t.btnSuccessHoverBgEnd,
    t.btnSuccessHoverBgStart,
    t.btnSuccessHoverBgEnd,
    t.btnSuccessHoverShadow,
    t.btnSuccessHoverShadowArrow,
    t.btnSuccessHoverShadowArrowLeft,
    t.btnSuccessHoverBorderColor,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonActiveMixin)(
    t.btnSuccessActiveBg,
    t.btnSuccessActiveBg,
    t.btnSuccessActiveBg,
    t.btnSuccessActiveShadow,
    t.btnSuccessActiveShadowArrow,
    t.btnSuccessActiveShadowArrowLeft,
    (0, _Emotion.cssName)(styles.active(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  danger: function danger(t) {
    return (0, _Emotion.css)(_templateObject22(),
    (0, _Button.buttonUseMixin)(
    t.btnDangerBg,
    t.btnDangerBgStart,
    t.btnDangerBgEnd,
    t.btnDangerBgArrowStart,
    t.btnDangerBgArrowEnd,
    t.btnDangerShadow,
    t.btnDangerShadowArrow,
    t.btnDangerShadowArrowLeft,
    t.btnDangerTextColor,
    t.btnDangerBorder,
    (0, _Emotion.cssName)(styles.checked(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonHoverMixin)(
    t.btnDangerHoverBg,
    t.btnDangerHoverBgStart,
    t.btnDangerHoverBgEnd,
    t.btnDangerHoverBgStart,
    t.btnDangerHoverBgEnd,
    t.btnDangerHoverShadow,
    t.btnDangerHoverShadowArrow,
    t.btnDangerHoverShadowArrowLeft,
    t.btnDangerHoverBorderColor,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonActiveMixin)(
    t.btnDangerActiveBg,
    t.btnDangerActiveBg,
    t.btnDangerActiveBg,
    t.btnDangerActiveShadow,
    t.btnDangerActiveShadowArrow,
    t.btnDangerActiveShadowArrowLeft,
    (0, _Emotion.cssName)(styles.active(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  pay: function pay(t) {
    return (0, _Emotion.css)(_templateObject23(),
    (0, _Button.buttonUseMixin)(
    t.btnPayBg,
    t.btnPayBgStart,
    t.btnPayBgEnd,
    t.btnPayBgArrowStart,
    t.btnPayBgArrowEnd,
    t.btnPayShadow,
    t.btnPayShadowArrow,
    t.btnPayShadowArrowLeft,
    t.btnPayTextColor,
    t.btnPayBorder,
    (0, _Emotion.cssName)(styles.checked(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonHoverMixin)(
    t.btnPayHoverBg,
    t.btnPayHoverBgStart,
    t.btnPayHoverBgEnd,
    t.btnPayHoverBgStart,
    t.btnPayHoverBgEnd,
    t.btnPayHoverShadow,
    t.btnPayHoverShadowArrow,
    t.btnPayHoverShadowArrowLeft,
    t.btnPayHoverBorderColor,
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))),


    (0, _Button.buttonActiveMixin)(
    t.btnPayActiveBg,
    t.btnPayActiveBg,
    t.btnPayActiveBg,
    t.btnPayActiveShadow,
    t.btnPayActiveShadowArrow,
    t.btnPayActiveShadowArrowLeft,
    (0, _Emotion.cssName)(styles.active(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    (0, _Emotion.cssName)(styles.arrowLeft(t))));


  },

  checked: function checked(t) {
    return (0, _Emotion.css)(_templateObject24(),
    t.btnCheckedShadow,
    t.btnCheckedBg,
    t.btnCheckedTextColor,
    t.btnDefaultCheckedBorder,

    (0, _Emotion.cssName)(styles.link(t)), (0, _Emotion.cssName)(styles.disabled(t)),
    (0, _Emotion.cssName)(styles.caption()),





    (0, _Emotion.cssName)(styles.focus(t)),
    (0, _Emotion.cssName)(styles.arrow()),
    t.btnCheckedBg,
    t.btnCheckedShadowArrow);



  },

  active: function active(t) {
    return (0, _Emotion.css)(_templateObject25(),
    (0, _Emotion.cssName)(styles.link(t)), (0, _Emotion.cssName)(styles.disabled(t)),
    (0, _Emotion.cssName)(styles.caption()));




  },

  caption: function caption() {
    return (0, _Emotion.css)(_templateObject26());






  },

  wrap: function wrap(t) {
    return (0, _Emotion.css)(_templateObject27(),
    t.btnWrapPadding);



  },

  narrow: function narrow() {
    return (0, _Emotion.css)(_templateObject28());



  },

  noPadding: function noPadding() {
    return (0, _Emotion.css)(_templateObject29());



  },

  noRightPadding: function noRightPadding() {
    return (0, _Emotion.css)(_templateObject30());


  },

  wrapLink: function wrapLink(t) {
    return (0, _Emotion.css)(_templateObject31(),
    styles.wrap(t));



  },

  wrapArrow: function wrapArrow() {
    return (0, _Emotion.css)(_templateObject32());


  },

  wrapArrowLeft: function wrapArrowLeft() {
    return (0, _Emotion.css)(_templateObject33());



  },

  arrow: function arrow() {
    return (0, _Emotion.css)(_templateObject34());





  },

  icon: function icon() {
    return (0, _Emotion.css)(_templateObject35());



  },

  buttonWithIcon: function buttonWithIcon() {
    return (0, _Emotion.css)(_templateObject36());



  },

  borderless: function borderless(t) {
    var focus = (0, _Emotion.cssName)(styles.focus(t));
    var disabled = (0, _Emotion.cssName)(styles.disabled(t));
    var checked = (0, _Emotion.cssName)(styles.checked(t));
    var active = (0, _Emotion.cssName)(styles.active(t));

    return (0, _Emotion.css)(_templateObject37(),
    focus, disabled, active, checked);







  },

  loading: function loading() {
    var btn_loading = (0, _Emotion.keyframes)(_templateObject38());








    return (0, _Emotion.css)(_templateObject39(),


















    btn_loading);



  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
