// @skbkontur/react-ui/components/Link/Link.styles/Link.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      margin-right: ", ";\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", "& {\n        color: ", ";\n\n        &:hover {\n          color: ", ";\n        }\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        text-decoration: ", ";\n      }\n      ", "& {\n        color: ", ";\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      border: 4px solid transparent;\n      border-bottom-width: 0;\n      border-top-color: #a0a0a0;\n      display: inline-block;\n      margin-bottom: 3px;\n      margin-left: 3px;\n      vertical-align: middle;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      background: #e1e1e1;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      line-height: 34px;\n      padding-left: 10px;\n      padding-right: 10px;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
import { linkMixin, linkDisabledMixin, linkUseColorsMixin } from "../Link.mixins";
var styles = {
  root: function root(t) {
    return css(_templateObject(), linkMixin());
  },
  button: function button() {
    return css(_templateObject2());
  },
  buttonOpened: function buttonOpened() {
    return css(_templateObject3());
  },
  arrow: function arrow() {
    return css(_templateObject4());
  },
  useDefault: function useDefault(t) {
    return css(_templateObject5(), linkUseColorsMixin(t.linkColor, t.linkHoverColor, t.linkActiveColor));
  },
  useSuccess: function useSuccess(t) {
    return css(_templateObject6(), linkUseColorsMixin(t.linkSuccessColor, t.linkSuccessHoverColor, t.linkSuccessActiveColor));
  },
  useDanger: function useDanger(t) {
    return css(_templateObject7(), linkUseColorsMixin(t.linkDangerColor, t.linkDangerHoverColor, t.linkDangerActiveColor));
  },
  useGrayed: function useGrayed(t) {
    return css(_templateObject8(), linkUseColorsMixin(t.linkDisabledColor, t.linkDisabledColor, t.linkDisabledColor));
  },
  focus: function focus(t) {
    return css(_templateObject9(), cssName(styles.root(t)), t.linkHoverTextDecoration, cssName(styles.useGrayed(t)), t.linkDisabledColor);
  },
  disabled: function disabled(t) {
    return css(_templateObject10(), linkDisabledMixin(), cssName(styles.root(t)), t.linkDisabledColor, t.linkDisabledColor);
  },
  icon: function icon(t) {
    return css(_templateObject11(), t.linkIconMarginRight);
  }
};
export var jsStyles = memoizeStyle(styles);
