// @skbkontur/react-ui/cjs/internal/themes/FlatTheme.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.FlatThemeInternal = exports.FlatTheme = void 0;var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var ColorFunctions = _interopRequireWildcard(require("../../lib/styles/ColorFunctions"));

var _DefaultTheme = require("./DefaultTheme");

// TODO Delete repeated values from flat theme
var
FlatTheme = /*#__PURE__*/function (_ref) {(0, _inheritsLoose2.default)(FlatTheme, _ref);function FlatTheme() {return _ref.apply(this, arguments) || this;}(0, _createClass2.default)(FlatTheme, null, [{ key: "btnCheckedShadowArrow", get: function get()




























































































































































    {
      return "0 0 0 1px " + this.btnCheckedBg;
    } }, { key: "btnCheckedShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnCheckedBg;
    } }, { key: "btnFocusBorder", get: function get()
    {
      return "1px solid " + this.borderColorFocus;
    } }, { key: "btnCheckedHoverBorderColor", get: function get()
    {
      return this.btnCheckedBg;
    } }, { key: "btnDefaultShadowArrowLeft", get: function get()
    {
      return this.btnDefaultShadowArrow;
    } }, { key: "btnDefaultCheckedShadowArrow", get: function get()

    {
      return "1px -1px 0 0 rgba(0, 0, 0, 0.15), 0 0 0 1px " + this.btnCheckedBg;
    } }, { key: "btnDefaultHoverShadowArrow", get: function get()
    {
      return "1px -1px 0 0 " + this.btnDefaultHoverBorderColor + ", 1px -1px 0 0 " + this.btnDefaultHoverBg;
    } }, { key: "btnDefaultHoverShadowArrowLeft", get: function get()
    {
      return "1px -1px 0 0 " + this.btnDefaultHoverBorderColor + ", 1px -1px 0 0 " + this.btnDefaultHoverBg;
    } }, { key: "btnDefaultActiveShadowArrow", get: function get()
    {
      return "1px -1px 0 0 " + this.btnDefaultActiveBorderColor + ", 1px -1px 0 0 " + this.btnDefaultActiveBg;
    } }, { key: "btnDefaultActiveShadowArrowLeft", get: function get()
    {
      return "1px -1px 0 0 " + this.btnDefaultActiveBorderColor + ", 1px -1px 0 0 " + this.btnDefaultActiveBg;
    } }, { key: "btnPrimaryShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryDisabledBg;
    } }, { key: "btnPrimaryShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryDisabledBg;
    } }, { key: "btnPrimaryHoverShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryHoverBg;
    } }, { key: "btnPrimaryHoverShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryHoverBg;
    } }, { key: "btnPrimaryActiveShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryActiveBg;
    } }, { key: "btnPrimaryActiveShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPrimaryActiveBg;
    } }, { key: "btnSuccessShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessDisabledBg;
    } }, { key: "btnSuccessShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessDisabledBg;
    } }, { key: "btnSuccessHoverShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessHoverBg;
    } }, { key: "btnSuccessHoverShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessHoverBg;
    } }, { key: "btnSuccessActiveShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessActiveBg;
    } }, { key: "btnSuccessActiveShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnSuccessActiveBg;
    } }, { key: "btnDangerShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerDisabledBg;
    } }, { key: "btnDangerShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerDisabledBg;
    } }, { key: "btnDangerHoverShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerHoverBg;
    } }, { key: "btnDangerHoverShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerHoverBg;
    } }, { key: "btnDangerActiveShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerActiveBg;
    } }, { key: "btnDangerActiveShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnDangerActiveBg;
    } }, { key: "btnPayShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPayDisabledBg;
    } }, { key: "btnPayShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPayDisabledBg;
    } }, { key: "btnPayHoverShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPayHoverBg;
    } }, { key: "btnPayHoverShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPayHoverBg;
    } }, { key: "btnPayActiveShadowArrow", get: function get()
    {
      return "0 0 0 1px " + this.btnPayActiveBg;
    } }, { key: "btnPayActiveShadowArrowLeft", get: function get()
    {
      return "0 0 0 1px " + this.btnPayActiveBg;
    } }, { key: "inputDisabledBorderColor", get: function get()
    {
      return this.inputDisabledBg;
    } }, { key: "chbBorderColorFocus", get: function get()
    {
      return this.borderColorFocus;
    } }, { key: "chbHoverBg", get: function get()
    {
      return this.btnDefaultHoverBg;
    } }, { key: "chbActiveBg", get: function get()
    {
      return this.btnDefaultActiveBg;
    } }, { key: "chbCheckedBg", get: function get()
    {
      return this.borderColorFocus;
    } }, { key: "chbBoxIndeterminateBg", get: function get()
    {
      return this.chbCheckedBg;
    } }, { key: "chbCheckedHoverBg", get: function get()
    {
      return ColorFunctions.darken(this.chbCheckedBg, '5%');
    } }, { key: "chbCheckedActiveBg", get: function get()
    {
      return ColorFunctions.darken(this.chbCheckedBg, '15%');
    } }, { key: "radioHoverBg", get: function get()
    {
      return this.chbHoverBg;
    } }, { key: "radioActiveBg", get: function get()
    {
      return this.chbActiveBg;
    } }, { key: "radioFocusShadow", get: function get()
    {
      return "inset 0 0 0 1px " + this.outlineColorFocus;
    } }, { key: "radioCheckedBgColor", get: function get()
    {
      return this.chbCheckedBg;
    } }, { key: "radioCheckedHoverBgColor", get: function get()
    {
      return ColorFunctions.darken(this.chbCheckedBg, '5%');
    } }, { key: "textareaDisabledBorderColor", get: function get()
    {
      return this.textareaDisabledBg;
    } }]);return FlatTheme;}( /*#__PURE__*/function () {function _class() {}return _class;}());exports.FlatTheme = FlatTheme;FlatTheme.outlineColorFocus = '#fff';FlatTheme.borderColorFocus = '#1d85d0';FlatTheme.blinkColor = 'rgba(0, 136, 255, 0.2)';FlatTheme.btnHeightShift = '0';FlatTheme.btnWrapPadding = '0';FlatTheme.btnLinkBorderRadius = '2px';FlatTheme.btnArrowBorderRadius = '2px 2px 2px 16px';FlatTheme.btnSmallBorderRadius = '2px';FlatTheme.btnFocusShadowWidth = '1px';FlatTheme.btnCheckedBg = '#7e7e7e';FlatTheme.btnCheckedTextColor = '#fff';FlatTheme.btnCheckedShadow = 'none';FlatTheme.btnDisabledBg = '#f2f2f2';FlatTheme.btnDisabledShadow = 'none';FlatTheme.btnDisabledShadowColor = '#f2f2f2';FlatTheme.btnDefaultBgArrowStart = '#fff';FlatTheme.btnDefaultBgArrowEnd = '#fff';FlatTheme.btnDefaultTextColor = 'none';FlatTheme.btnDefaultHoverBg = 'rgba(0, 0, 0, 0.05)';FlatTheme.btnDefaultHoverBorderColor = 'rgba(0, 0, 0, 0.15)';FlatTheme.btnDefaultActiveBg = '#e5e5e5';FlatTheme.btnDefaultActiveBorderColor = '#cccccc';FlatTheme.btnDefaultBorder = '1px solid rgba(0, 0, 0, 0.15)';FlatTheme.btnDefaultShadow = 'none';FlatTheme.btnDefaultShadowArrow = '1px -1px 0 0 rgba(0, 0, 0, 0.15)';FlatTheme.btnDefaultBg = '#fff';FlatTheme.btnDefaultBgStart = 'none';FlatTheme.btnDefaultBgEnd = 'none';FlatTheme.btnDefaultHoverBgStart = '#f2f2f2';FlatTheme.btnDefaultHoverBgEnd = '#f2f2f2';FlatTheme.btnDefaultHoverShadow = 'none';FlatTheme.btnDefaultActiveShadow = 'none';FlatTheme.btnPrimaryBg = '#1d85d0';FlatTheme.btnPrimaryTextColor = '#fff';FlatTheme.btnPrimaryHoverBg = '#1b7dc3';FlatTheme.btnPrimaryActiveBg = '#1974b6';FlatTheme.btnPrimaryDisabledBg = '#1d85d0';FlatTheme.btnPrimaryBorder = '1px solid transparent';FlatTheme.btnPrimaryBgStart = '#1d85d0';FlatTheme.btnPrimaryBgEnd = '#1d85d0';FlatTheme.btnPrimaryBgArrowStart = '#1d85d0';FlatTheme.btnPrimaryBgArrowEnd = '#1d85d0';FlatTheme.btnPrimaryHoverBgStart = '#1b7dc3';FlatTheme.btnPrimaryHoverBgEnd = '#1b7dc3';FlatTheme.btnPrimaryShadow = 'none';FlatTheme.btnPrimaryHoverShadow = 'none';FlatTheme.btnPrimaryActiveShadow = 'none';FlatTheme.btnSuccessBg = '#419d14';FlatTheme.btnSuccessTextColor = '#fff';FlatTheme.btnSuccessHoverBg = '#3c9312';FlatTheme.btnSuccessActiveBg = '#398911';FlatTheme.btnSuccessDisabledBg = '#419d14';FlatTheme.btnSuccessBorder = '1px solid transparent';FlatTheme.btnSuccessBgStart = '#419d14';FlatTheme.btnSuccessBgEnd = '#419d14';FlatTheme.btnSuccessBgArrowStart = '#419d14';FlatTheme.btnSuccessBgArrowEnd = '#419d14';FlatTheme.btnSuccessHoverBgStart = '#3c9312';FlatTheme.btnSuccessHoverBgEnd = '#3c9312';FlatTheme.btnSuccessShadow = 'none';FlatTheme.btnSuccessHoverShadow = 'none';FlatTheme.btnSuccessActiveShadow = 'none';FlatTheme.btnDangerBg = '#d9472b';FlatTheme.btnDangerTextColor = '#fff';FlatTheme.btnDangerHoverBg = '#cc4228';FlatTheme.btnDangerActiveBg = '#be3e25';FlatTheme.btnDangerDisabledBg = '#d9472b';FlatTheme.btnDangerBorder = '1px solid transparent';FlatTheme.btnDangerBgStart = '#d9472b';FlatTheme.btnDangerBgEnd = '#d9472b';FlatTheme.btnDangerBgArrowStart = '#d9472b';FlatTheme.btnDangerBgArrowEnd = '#d9472b';FlatTheme.btnDangerHoverBgStart = '#cc4228';FlatTheme.btnDangerHoverBgEnd = '#cc4228';FlatTheme.btnDangerShadow = 'none';FlatTheme.btnDangerHoverShadow = 'none';FlatTheme.btnDangerActiveShadow = 'none';FlatTheme.btnPayBg = '#ffca43';FlatTheme.btnPayTextColor = 'none';FlatTheme.btnPayHoverBg = '#f0be3f';FlatTheme.btnPayActiveBg = '#e0b13a';FlatTheme.btnPayDisabledBg = '#ffca43';FlatTheme.btnPayBorder = '1px solid transparent';FlatTheme.btnPayBgStart = '#ffca43';FlatTheme.btnPayBgEnd = '#ffca43';FlatTheme.btnPayBgArrowStart = '#ffca43';FlatTheme.btnPayBgArrowEnd = '#ffca43';FlatTheme.btnPayHoverBgStart = '#f0be3f';FlatTheme.btnPayHoverBgEnd = '#f0be3f';FlatTheme.btnPayShadow = 'none';FlatTheme.btnPayHoverShadow = 'none';FlatTheme.btnPayActiveShadow = 'none';FlatTheme.btnSmallArrowLength = '17px';FlatTheme.btnSmallArrowTop = '7.5px';FlatTheme.btnSmallArrowRight = '-8px';FlatTheme.btnSmallArrowLeft = '-8.2px';FlatTheme.btnSmallArrowLeftLoadingDelay = '0.23s';FlatTheme.btnSmallArrowBorderRadius = '2px 2px 4px 16px';FlatTheme.btnSmallArrowBg = 'linear-gradient(-53deg, transparent 48%, #ccc 0, #ccc 72%, transparent 0)';FlatTheme.btnMediumArrowLeft = '-10px';FlatTheme.btnMediumArrowRight = '-10px';FlatTheme.btnMediumArrowLeftLoadingLeft = '-207px';FlatTheme.btnMediumArrowTransform = 'rotate(53deg) skewX(25deg) skewY(8deg)';FlatTheme.btnMediumArrowBg = 'linear-gradient(-56deg, transparent 48%, #ccc 0, #ccc 70%, transparent 0)';FlatTheme.btnMediumArrowLeftLoadingDelay = '0.45s';FlatTheme.btnDeprecatedSizeMediumFontSize = '16px';FlatTheme.btnDeprecatedSizeMediumPaddingShift = '1';FlatTheme.btnDeprecatedSizeMediumPaddingShiftIe = '2';FlatTheme.btnLargeArrowLeft = '-10.8px';FlatTheme.btnLargeArrowTransform = 'rotate(53deg) skewX(26deg) skewY(10deg)';FlatTheme.btnLargeArrowBg = 'linear-gradient(-56deg, transparent 48.2%, #ccc 0, #ccc 63.4%, transparent 0)';FlatTheme.btnLargeArrowLeftLoadingDelay = '0s';FlatTheme.chbShadow = '0 0 0 1px rgba(0, 0, 0, 0.15)';FlatTheme.chbBg = 'none';FlatTheme.chbIndeterminateBg = '#fff';FlatTheme.chbShadowDisabled = '0 0 0 1px #f2f2f2';FlatTheme.chbShadowActive = '0 0 0 1px #c3c3c3';FlatTheme.chbShadowHover = '0 0 0 1px #c3c3c3';FlatTheme.chbShadowWidth = '2px';FlatTheme.chbCheckedColor = '#fff';FlatTheme.chbCheckedShadow = '0 0 0 1px #1768a3';FlatTheme.chbCheckedHoverShadow = '0 0 0 1px #175590';FlatTheme.chbCheckedActiveShadow = '0 0 0 1px #175590';FlatTheme.chbBorder = 'none';FlatTheme.chbBorderRadius = '1px';FlatTheme.inputBg = 'white';FlatTheme.inputIconColor = '#a9a9a9';FlatTheme.inputColor = 'inherit';FlatTheme.inputShadow = 'none';FlatTheme.inputBorderWidth = '1px';FlatTheme.inputBorderTopColor = 'rgba(0, 0, 0, 0.15)';FlatTheme.inputDisabledBg = 'rgba(0, 0, 0, 0.05)';FlatTheme.toggleBg = '#fff';FlatTheme.toggleBgDisabled = 'none';FlatTheme.toggleBgHover = '#f3f3f2';FlatTheme.toggleBgChecked = '#3072c4';FlatTheme.toggleBgWarning = '#f69c00';FlatTheme.toggleBgError = '#d70c17';FlatTheme.toggleBgActive = '#e5e5e5';FlatTheme.toggleBorderColor = '#d0d0d0';FlatTheme.textareaBg = 'white';FlatTheme.textareaColor = 'inherit';FlatTheme.textareaShadow = 'none';FlatTheme.textareaBorderTopColor = 'rgba(0, 0, 0, 0.15)';FlatTheme.textareaDisabledBg = 'rgba(0, 0, 0, 0.05)';FlatTheme.radioSize = '18px';FlatTheme.radioVerticalAlign = 'top';FlatTheme.radioBgImage = 'none';FlatTheme.radioBoxShadow = 'none';FlatTheme.radioBorder = '1px solid rgba(0, 0, 0, 0.15)';FlatTheme.radioHoverShadow = 'none';FlatTheme.radioActiveShadow = 'none';FlatTheme.radioCheckedBulletColor = '#fff';FlatTheme.radioDisabledShadow = 'none';FlatTheme.radioLabelDisplay = 'inline-block';FlatTheme.btnFontSizeMedium = '16px';FlatTheme.btnDefaultCheckedBorder = '1px solid transparent';


var FlatThemeInternal = Object.setPrototypeOf(FlatTheme, _DefaultTheme.DefaultThemeInternal);exports.FlatThemeInternal = FlatThemeInternal;
