// @skbkontur/react-ui/internal/SpinnerOld/SpinnerOld.styles/SpinnerOld.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: relative;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      display: block;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-left: 5px;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      text-align: center;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: 14px;\n      color: ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 2s linear infinite,\n        ", " 6s ease-in-out infinite;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke-dasharray: 20, 86;\n      stroke-dashoffset: 15;\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke-dasharray: 20, 86;\n      stroke-dashoffset: 15;\n      stroke: ", ";\n      animation: ", " 1s cubic-bezier(0.5, 0.2, 0.5, 0.8) infinite,\n        ", " 2s cubic-bezier(0.36, 0.14, 0.38, 0.69) infinite,\n        ", " 6s ease-in-out infinite;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      fill: none;\n      stroke-linecap: round;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      stroke: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, memoizeStyle } from "../../../lib/theming/Emotion";
import { AnimationKeyframes } from "../../../lib/theming/AnimationKeyframes";
var styles = {
  cloudStroke: function cloudStroke(t) {
    return css(_templateObject(), t.spinnerBgColor);
  },
  cloudWrapper: function cloudWrapper() {
    return css(_templateObject2());
  },
  cloud: function cloud(t) {
    return css(_templateObject3(), t.red, AnimationKeyframes.spinnerCloudOffset(t), AnimationKeyframes.spinnerCloudLength(t), AnimationKeyframes.spinnerColor(t));
  },
  cloudDimmed: function cloudDimmed(t) {
    return css(_templateObject4(), t.spinnerDimmedColor, AnimationKeyframes.spinnerCloudOffset(t), AnimationKeyframes.spinnerCloudLength(t));
  },
  circle: function circle(t) {
    return css(_templateObject5(), t.red, AnimationKeyframes.spinnerCircleOffset(t), AnimationKeyframes.spinnerCircleLength(t), AnimationKeyframes.spinnerCircleRotate(t), AnimationKeyframes.spinnerColor(t));
  },
  circleDimmed: function circleDimmed(t) {
    return css(_templateObject6(), t.spinnerDimmedColor, AnimationKeyframes.spinnerCircleOffset(t), AnimationKeyframes.spinnerCircleLength(t), AnimationKeyframes.spinnerCircleRotate(t));
  },
  caption: function caption(t) {
    return css(_templateObject7(), t.spinnerCaptionColor);
  },
  spinner: function spinner() {
    return css(_templateObject8());
  },
  inner: function inner() {
    return css(_templateObject9());
  },
  captionRight: function captionRight() {
    return css(_templateObject10());
  },
  captionBottom: function captionBottom() {
    return css(_templateObject11());
  },
  fallback: function fallback() {
    return css(_templateObject12());
  }
};
export var jsStyles = memoizeStyle(styles);
