// @skbkontur/react-ui/cjs/lib/LayoutEvents.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.addListener = addListener;exports.emit = emit;var _eventemitter = _interopRequireDefault(require("eventemitter3"));
var _reactDom = require("react-dom");

var emitterCache;
function getEmitter() {
  if (!emitterCache) {
    emitterCache = new _eventemitter.default();
  }
  return emitterCache;
}

function listenBrowserEvents() {
  window.addEventListener('scroll', emit);
  window.addEventListener('resize', emit);
}

function unlistenBrowserEvents() {
  window.removeEventListener('scroll', emit);
  window.removeEventListener('resize', emit);
}

function addListener(callback) {
  var emitter = getEmitter();
  if (emitter.listeners('layout').length === 0) {
    listenBrowserEvents();
  }
  emitter.addListener('layout', callback);
  return {
    remove: function remove() {
      emitter.removeListener('layout', callback);
      if (emitter.listeners('layout').length === 0) {
        unlistenBrowserEvents();
      }
    } };

}

function emit() {
  (0, _reactDom.unstable_batchedUpdates)(function () {
    getEmitter().emit('layout');
  });
}
