// @skbkontur/react-ui/cjs/lib/utils.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.taskWithDelay = taskWithDelay;exports.isFunction = isFunction;exports.isFunctionalComponent = isFunctionalComponent;exports.escapeRegExpSpecChars = escapeRegExpSpecChars;exports.isExternalLink = exports.hasSvgAnimationSupport = exports.getRandomID = exports.CancelationError = exports.emptyHandler = exports.delay = exports.isIE11 = exports.isEdge = exports.isChrome = exports.isOpera = exports.isFirefox = exports.isSafari = exports.isWindows = exports.isMac = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _wrapNativeSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/wrapNativeSuper"));

// NOTE: Copy-paste from @types/react










// NOTE Some checks are used from https://github.com/arasatasaygin/is.js
var platform = (navigator && navigator.platform || '').toLowerCase();
var userAgent = (navigator && navigator.userAgent || '').toLowerCase();
var vendor = (navigator && navigator.vendor || '').toLowerCase();

var isMac = platform.includes("mac");exports.isMac = isMac;
var isWindows = platform.includes("win");exports.isWindows = isWindows;

var isSafari = /version\/(\d+).+?safari/.test(userAgent);exports.isSafari = isSafari;
var isFirefox = /(?:firefox|fxios)\/(\d+)/.test(userAgent);exports.isFirefox = isFirefox;
var isOpera = /(?:^opera.+?version|opr)\/(\d+)/.test(userAgent);exports.isOpera = isOpera;
var isChrome = vendor.includes("google inc") && /(?:chrome|crios)\/(\d+)/.test(userAgent) && !isOpera;exports.isChrome = isChrome;
var isEdge = userAgent.includes("edge/");exports.isEdge = isEdge;
var isIE11 = userAgent.includes("trident/");exports.isIE11 = isIE11;

var delay = function delay(ms) {return new Promise(function (resolve) {return setTimeout(resolve, ms);});};exports.delay = delay;

var emptyHandler = function emptyHandler() {
  /* noop */
};exports.emptyHandler = emptyHandler;var

CancelationError = /*#__PURE__*/function (_Error) {(0, _inheritsLoose2.default)(CancelationError, _Error);function CancelationError() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _Error.call.apply(_Error, [this].concat(args)) || this;_this.
    code = 'CancelationError';return _this;}return CancelationError;}((0, _wrapNativeSuper2.default)(Error));exports.CancelationError = CancelationError;


function taskWithDelay(task, ms) {
  var cancelationToken = function cancelationToken() {return null;};

  new Promise(function (resolve, reject) {
    cancelationToken = reject;
    setTimeout(resolve, ms);
  }).
  then(task).
  catch(function () {return null;});

  return cancelationToken;
}

function isFunction(x) {
  return typeof x === 'function';
}

function isFunctionalComponent(Component) {
  return typeof Component === 'function' && !(Component.prototype && Component.prototype.isReactComponent);
}

function escapeRegExpSpecChars(s) {
  return s.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
}

var getRandomID = function getRandomID() {return (
    Math.random().
    toString(16).
    slice(2));};exports.getRandomID = getRandomID;

var hasSvgAnimationSupport = function () {
  if (document.createElementNS) {
    var namespaceURI = 'http://www.w3.org/2000/svg';
    var element = document.createElementNS(namespaceURI, 'animate');

    if (element) {
      return element.toString().includes("SVGAnimate");
    }
  }

  return false;
}();exports.hasSvgAnimationSupport = hasSvgAnimationSupport;

var isExternalLink = function isExternalLink(link) {
  return new RegExp("^(https?:)?//(?!" + window.location.host + ")\\S+", 'gi').test(link);
};exports.isExternalLink = isExternalLink;
