// @skbkontur/react-ui/components/SidePage/SidePageFooter/SidePageFooter.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import cn from 'classnames';
import * as LayoutEvents from "../../../lib/LayoutEvents";
import { ThemeContext } from "../../../lib/theming/ThemeContext";
import { jsStyles } from "../SidePage.styles";
import { SidePageContext } from "../SidePageContext";
/**
                                                      * Футер сайдпейджа.
                                                      *
                                                      * @visibleName SidePage.Footer
                                                      */

export var SidePageFooter =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(SidePageFooter, _React$Component);

  function SidePageFooter() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.context = _this.context;
    _this.state = {
      fixed: false
    };
    _this.theme = void 0;
    _this.content = null;
    _this.wrapper = null;
    _this.layoutSub = null;

    _this.update = function () {
      _this.setProperStyles();
    };

    _this.refContent = function (node) {
      _this.content = node;
    };

    _this.refWrapper = function (node) {
      _this.wrapper = node;
    };

    _this.setProperStyles = function () {
      if (_this.wrapper && _this.content) {
        var wrapperRect = _this.wrapper.getBoundingClientRect();

        var contentRect = _this.content.getBoundingClientRect();

        var _fixed = wrapperRect.top > contentRect.top;

        _this.setState({
          fixed: _fixed
        });
      }
    };

    return _this;
  }

  var _proto = SidePageFooter.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.context.footerRef(this);
    this.update();
    this.layoutSub = LayoutEvents.addListener(this.update);
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    this.context.footerRef(null);

    if (this.layoutSub) {
      this.layoutSub.remove();
    }
  };

  _proto.render = function render() {
    var _this2 = this;

    return React.createElement(ThemeContext.Consumer, null, function (theme) {
      _this2.theme = theme;
      return _this2.renderMain();
    });
  };

  _proto.renderMain = function renderMain() {
    var _this3 = this;

    return React.createElement("div", {
      style: {
        height: this.getContentHeight()
      },
      ref: this.refWrapper
    }, React.createElement(SidePageContext.Consumer, null, function (_ref) {
      var _cn;

      var getWidth = _ref.getWidth;
      return React.createElement("div", {
        className: jsStyles.footer(),
        style: {
          width: getWidth()
        }
      }, React.createElement("div", {
        className: cn(jsStyles.footerContent(), (_cn = {}, _cn[jsStyles.panel(_this3.theme)] = !!_this3.props.panel, _cn[jsStyles.footerFixed(_this3.theme)] = _this3.state.fixed, _cn)),
        ref: _this3.refContent
      }, _this3.props.children));
    }));
  };

  _proto.getContentHeight = function getContentHeight() {
    if (!this.content) {
      return 'auto';
    }

    return this.content.getBoundingClientRect().height;
  };

  return SidePageFooter;
}(React.Component);
SidePageFooter.__KONTUR_REACT_UI__ = 'SidePageFooter';
SidePageFooter.contextType = SidePageContext;
