// @skbkontur/react-ui/cjs/components/Select/Select.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.Select = void 0;var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _reactDom = _interopRequireDefault(require("react-dom"));
var _invariant = _interopRequireDefault(require("invariant"));
var _warning = _interopRequireDefault(require("warning"));
var _classnames = _interopRequireDefault(require("classnames"));

var _identifiers = require("../../lib/events/keyboard/identifiers");







var _decorators = require("../../lib/locale/decorators");
var _reactGetTextContent = require("../../lib/reactGetTextContent");
var _Button = require("../Button");
var _DropdownContainer = require("../../internal/DropdownContainer");
var _filterProps = require("../../lib/filterProps");
var _Input = require("../Input");
var _Link = require("../Link");
var _Menu = require("../../internal/Menu");
var _MenuItem = require("../MenuItem");
var _MenuSeparator = require("../MenuSeparator");
var _RenderLayer = require("../../internal/RenderLayer");
var _createPropsGetter = require("../../lib/createPropsGetter");

var _utils = require("../../lib/utils");
var _ThemeContext = require("../../lib/theming/ThemeContext");


var _Item = require("./Item");
var _locale = require("./locale");
var _Select = require("./Select.styles");var _dec, _class, _class2, _temp;










var PASS_BUTTON_PROPS = {
  disabled: true,
  error: true,
  use: true,
  size: true,
  warning: true,

  onMouseEnter: true,
  onMouseLeave: true,
  onMouseOver: true };var






























































































Select = (_dec = (0, _decorators.locale)('Select', _locale.SelectLocaleHelper), _dec(_class = (_temp = _class2 = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(Select, _React$Component);function Select() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.














































    state = {
      opened: false,
      value: _this.props.defaultValue };_this.


    theme = void 0;_this.
    locale = void 0;_this.
    menu = void 0;_this.
    buttonElement = null;_this.
    getProps = (0, _createPropsGetter.createPropsGetter)(Select.defaultProps);_this.
























    open = function () {
      if (!_this.state.opened) {
        _this.setState({ opened: true });

        if (_this.props.onOpen) {
          _this.props.onOpen();
        }
      }
    };_this.




    close = function () {
      if (_this.state.opened) {
        _this.setState({ opened: false });

        if (_this.props.onClose) {
          _this.props.onClose();
        }
      }
    };_this.




    focus = function () {
      if (_this.buttonElement && _this.buttonElement.focus) {
        _this.buttonElement.focus();
      }
    };_this.







































































































































































    dropdownContainerGetParent = function () {
      return _reactDom.default.findDOMNode((0, _assertThisInitialized2.default)(_this));
    };_this.

    focusInput = function (input) {
      // fix cases when an Input is rendered in portal
      setTimeout(function () {return input == null ? void 0 : input.focus();}, 0);
    };_this.

    refMenu = function (menu) {
      _this.menu = menu;
    };_this.

    toggle = function () {
      if (_this.state.opened) {
        _this.close();
      } else {
        _this.open();
      }
    };_this.

    handleKey = function (e) {
      if (!_this.state.opened) {
        if ((0, _identifiers.isKeySpace)(e) || (0, _identifiers.isKeyArrowVertical)(e)) {
          e.preventDefault();
          _this.open();
        }
      } else {
        switch (true) {
          case (0, _identifiers.isKeyEscape)(e):
            _this.focus();
            _this.close();
            break;
          case (0, _identifiers.isKeyArrowUp)(e):
            e.preventDefault();
            if (_this.menu) {
              _this.menu.up();
            }
            break;
          case (0, _identifiers.isKeyArrowDown)(e):
            e.preventDefault();
            if (_this.menu) {
              _this.menu.down();
            }
            break;
          case (0, _identifiers.isKeyEnter)(e):
            e.preventDefault(); // To prevent form submission.
            if (_this.menu) {
              _this.menu.enter(e);
            }
            break;}

      }
      if (_this.props.onKeyDown) {
        _this.props.onKeyDown(e);
      }
    };_this.

    handleSearch = function (value) {
      _this.setState({ searchPattern: value });
    };_this.























































    buttonRef = function (element) {
      _this.buttonElement = element;
    };_this.

    getButton = function (buttonParams) {
      var button = _this.props._renderButton ?
      _this.props._renderButton(buttonParams) :
      _this.renderDefaultButton(buttonParams);

      var buttonElement = _react.default.Children.only(button);

      return _react.default.isValidElement(buttonElement) ?
      _react.default.cloneElement(buttonElement, {
        ref: _this.buttonRef,
        onFocus: _this.props.onFocus,
        onBlur: _this.props.onBlur }) :

      buttonElement;
    };return _this;}var _proto = Select.prototype;_proto.componentDidUpdate = function componentDidUpdate(_prevProps, prevState) {if (!prevState.opened && this.state.opened) {window.addEventListener('popstate', this.close);}if (prevState.opened && !this.state.opened) {window.removeEventListener('popstate', this.close);}};_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});} /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               * @public
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               */;_proto.renderMain = function renderMain() {var _this$renderLabel = this.renderLabel(),label = _this$renderLabel.label,isPlaceholder = _this$renderLabel.isPlaceholder;var buttonParams = { opened: this.state.opened, label: label, isPlaceholder: isPlaceholder, onClick: this.toggle, onKeyDown: this.handleKey };var style = { width: this.props.width, maxWidth: this.props.maxWidth || undefined };var button = this.getButton(buttonParams);return _react.default.createElement(_RenderLayer.RenderLayer, { onClickOutside: this.close, onFocusOutside: this.close, active: this.state.opened }, _react.default.createElement("span", { className: _Select.jsStyles.root(this.theme), style: style }, button, !this.props.disabled && this.state.opened && this.renderMenu()));};_proto.renderLabel = function renderLabel() {var value = this.getValue();var item = this.getItemByValue(value);if (item != null || value != null) {return { label: this.getProps().renderValue(value, item), isPlaceholder: false };}return { label: _react.default.createElement("span", null, this.props.placeholder || this.locale.placeholder), isPlaceholder: true };};_proto.renderDefaultButton = function renderDefaultButton(params) {var _cn;if (this.props.diadocLinkIcon) {(0, _warning.default)(false, "diadocLinkIcon has been deprecated");return this.renderLinkButton(params);}var buttonProps = (0, _extends2.default)({}, (0, _filterProps.filterProps)(this.props, PASS_BUTTON_PROPS), { align: 'left', disabled: this.props.disabled, _noPadding: true, width: '100%', onClick: params.onClick, onKeyDown: params.onKeyDown, active: params.opened });if (this.props._icon) {Object.assign(buttonProps, { _noPadding: false, _noRightPadding: true, icon: this.props._icon });}var labelProps = { className: (0, _classnames.default)((_cn = {}, _cn[_Select.jsStyles.label()] = this.props.use !== 'link', _cn[_Select.jsStyles.labelWithLeftIcon()] = !!this.props._icon, _cn[_Select.jsStyles.placeholder(this.theme)] = params.isPlaceholder, _cn[_Select.jsStyles.customUsePlaceholder()] = params.isPlaceholder && this.props.use !== 'default', _cn)), style: { paddingRight: (buttonProps.size === 'large' ? 31 : 28) + (this.props._icon ? 10 : 0) } };var useIsCustom = this.props.use !== 'default';return _react.default.createElement(_Button.Button, buttonProps, _react.default.createElement("span", labelProps, _react.default.createElement("span", { className: _Select.jsStyles.labelText() }, params.label)), _react.default.createElement("div", { className: _Select.jsStyles.arrowWrap() }, _react.default.createElement("div", { className: (0, _classnames.default)(_Select.jsStyles.arrow(this.theme), useIsCustom && _Select.jsStyles.customUseArrow()) })));};_proto.renderLinkButton = function renderLinkButton(params) {var linkProps = { disabled: params.disabled, icon: this.props.diadocLinkIcon, _button: true, _buttonOpened: params.opened, onClick: params.onClick, onKeyDown: params.onKeyDown };return _react.default.createElement(_Link.Link, linkProps, params.label);};_proto.renderMenu = function renderMenu() {var _this3 = this;var search = this.props.search ? _react.default.createElement("div", { className: _Select.jsStyles.search() }, _react.default.createElement(_Input.Input, { ref: this.focusInput, onValueChange: this.handleSearch, width: "100%" })) : null;var value = this.getValue();return _react.default.createElement(_DropdownContainer.DropdownContainer, { getParent: this.dropdownContainerGetParent, offsetY: -1, align: this.props.menuAlign, disablePortal: this.props.disablePortal }, _react.default.createElement(_Menu.Menu, { ref: this.refMenu, width: this.props.menuWidth, onItemClick: this.close, maxHeight: this.props.maxMenuHeight }, search, this.mapItems(function (iValue, item, i, comment) {if ((0, _utils.isFunction)(item)) {var element = item();if (_react.default.isValidElement(element)) {return _react.default.cloneElement(element, { key: i });}return null;}if (_react.default.isValidElement(item)) {return _react.default.cloneElement(item, { key: i });}return _react.default.createElement(_MenuItem.MenuItem, { key: i, state: _this3.getProps().areValuesEqual(iValue, value) ? 'selected' : null, onClick: _this3.select.bind(_this3, iValue), comment: comment }, _this3.getProps().renderItem(iValue, item));})));};_proto.select = function select(value) {this.focus();this.setState({ opened: false, value: value });if (this.props.onValueChange && !this.getProps().areValuesEqual(this.getValue(), value)) {this.props.onValueChange(value);}};_proto.getValue = function getValue() {if (this.props.value !== undefined) {return this.props.value;}return this.state.value;};_proto.mapItems = function mapItems(fn) {var items = this.props.items;if (!items) {return [];}var pattern = this.state.searchPattern && this.state.searchPattern.toLowerCase();var result = [];var index = 0;for (var _iterator = items, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {var _ref;if (_isArray) {if (_i >= _iterator.length) break;_ref = _iterator[_i++];} else {_i = _iterator.next();if (_i.done) break;_ref = _i.value;}var entry = _ref;var _normalizeEntry = normalizeEntry(entry),_value = _normalizeEntry[0],_item = _normalizeEntry[1],_comment = _normalizeEntry[2];if (!pattern || this.getProps().filterItem(_value, _item, pattern)) {result.push(fn(_value, _item, index, _comment));++index;}}return result;};_proto.getItemByValue = function getItemByValue(value) {if (value === null || value === undefined) {return null;}var items = this.props.items || [];for (var _iterator2 = items, _isArray2 = Array.isArray(_iterator2), _i2 = 0, _iterator2 = _isArray2 ? _iterator2 : _iterator2[Symbol.iterator]();;) {var _ref2;if (_isArray2) {if (_i2 >= _iterator2.length) break;_ref2 = _iterator2[_i2++];} else {_i2 = _iterator2.next();if (_i2.done) break;_ref2 = _i2.value;}var entry = _ref2;var _normalizeEntry2 = normalizeEntry(entry),itemValue = _normalizeEntry2[0],_item2 = _normalizeEntry2[1];if (this.getProps().areValuesEqual(itemValue, value)) {return _item2;}}return null;};return Select;}(_react.default.Component), _class2.__KONTUR_REACT_UI__ = 'Select', _class2.propTypes = { areValuesEqual: _propTypes.default.func, defaultValue: _propTypes.default.any, disablePortal: _propTypes.default.bool, disabled: _propTypes.default.bool, error: _propTypes.default.bool, filterItem: _propTypes.default.func, items: _propTypes.default.oneOfType([_propTypes.default.array, _propTypes.default.object]), maxMenuHeight: _propTypes.default.number, maxWidth: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]), placeholder: _propTypes.default.node, renderItem: _propTypes.default.func, renderValue: _propTypes.default.func, search: _propTypes.default.bool, value: _propTypes.default.any, width: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]), onValueChange: _propTypes.default.func, onMouseEnter: _propTypes.default.func, onMouseLeave: _propTypes.default.func, onMouseOver: _propTypes.default.func, onKeyDown: _propTypes.default.func }, _class2.defaultProps = { renderValue: renderValue, renderItem: renderItem, areValuesEqual: areValuesEqual, filterItem: filterItem, use: 'default' }, _class2.Item = _Item.Item, _class2.SEP = function () {return _react.default.createElement(_MenuSeparator.MenuSeparator, null);}, _class2.static = function (element) {(0, _invariant.default)(_react.default.isValidElement(element) || typeof element === 'function', 'Select.static(element) expects element to be a valid react element.');return element;}, _temp)) || _class);exports.Select = Select;
function renderValue(value, item) {
  return item;
}

function renderItem(value, item) {
  return item;
}

function areValuesEqual(value1, value2) {
  return value1 === value2;
}

function normalizeEntry(entry) {
  if (Array.isArray(entry)) {
    return entry;
  } else {
    return [entry, entry, undefined];
  }
}

function filterItem(value, item, pattern) {
  if (item === Select.SEP) {
    return false;
  }
  if (_react.default.isValidElement(item) || (0, _utils.isFunction)(item) && _react.default.isValidElement(item = item())) {
    item = (0, _reactGetTextContent.reactGetTextContent)(item);
  }
  if (typeof item === 'number') {
    item = item.toString(10);
  }
  if (typeof item !== 'string') {
    return false;
  }
  return item.toLowerCase().indexOf(pattern) !== -1;
}
