// @skbkontur/react-ui/cjs/components/Select/Select.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      margin: 1px 4px 4px;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 0;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      max-width: 100%;\n      overflow: hidden;\n      text-overflow: ellipsis;\n      vertical-align: top;\n      white-space: nowrap;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-sizing: border-box;\n      display: inline-block;\n      padding: 0 28px 0 9px;\n      max-width: 100%;\n      width: 100%;\n      position: relative;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border-top-color: inherit;\n      color: inherit;\n      opacity: 0.7;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      right: 0;\n      padding: 0 10px;\n\n      &::before {\n        content: '';\n        display: inline-block;\n        height: 100%;\n        vertical-align: middle;\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      border: 4px solid transparent;\n      border-bottom-width: 0;\n      border-top-color: ", ";\n\n      ", " & {\n        display: inline-block;\n        margin-bottom: 3px;\n        vertical-align: middle;\n      }\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: inherit;\n      opacity: 0.7;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n      text-overflow: ellipsis;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: relative;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject());



  },

  placeholder: function placeholder(t) {
    return (0, _Emotion.css)(_templateObject2(),
    t.sltPlaceholderColor);


  },

  customUsePlaceholder: function customUsePlaceholder() {
    return (0, _Emotion.css)(_templateObject3());



  },

  arrow: function arrow(t) {
    return (0, _Emotion.css)(_templateObject4(),


    t.btnMenuArrowColor,

    (0, _Emotion.cssName)(styles.arrowWrap()));





  },

  arrowWrap: function arrowWrap() {
    return (0, _Emotion.css)(_templateObject5());













  },

  customUseArrow: function customUseArrow() {
    return (0, _Emotion.css)(_templateObject6());




  },

  label: function label() {
    return (0, _Emotion.css)(_templateObject7());







  },

  labelText: function labelText() {
    return (0, _Emotion.css)(_templateObject8());







  },

  labelWithLeftIcon: function labelWithLeftIcon() {
    return (0, _Emotion.css)(_templateObject9());


  },

  search: function search() {
    return (0, _Emotion.css)(_templateObject10());


  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
