// @skbkontur/react-ui/internal/ThemePlayground/RadioPlayground/RadioPlayground.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { Gapped } from "../../../components/Gapped";
import { Radio } from "../../../components/Radio";
import { getComponentsFromPropsList } from "../helpers";
var propsList = [{
  value: '',
  children: 'Default',
  checked: false
}, {
  value: '',
  children: 'Checked'
}, {
  value: '',
  children: 'Disabled',
  disabled: true
}, {
  value: '',
  children: 'Focused',
  focused: true
}, {
  value: '',
  children: 'Error',
  error: true
}, {
  value: '',
  children: 'Warning',
  warning: true
}];
export var RadioPlayground =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(RadioPlayground, _React$Component);

  function RadioPlayground() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = RadioPlayground.prototype;

  _proto.render = function render() {
    return React.createElement(Gapped, {
      vertical: true
    }, getComponentsFromPropsList(React.createElement(Radio, {
      value: '',
      checked: true
    }), propsList));
  };

  return RadioPlayground;
}(React.Component);
