// @skbkontur/react-ui/cjs/internal/InputLikeText/InputLikeText.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.InputLikeText = void 0;var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));

var _identifiers = require("../../lib/events/keyboard/identifiers");
var _MouseDrag = require("../../lib/events/MouseDrag");
var _utils = require("../../lib/utils");

var _SelectionHelpers = require("../../components/DateInput/helpers/SelectionHelpers");

var _Input = require("../../components/Input/Input.styles");
var _ThemeContext = require("../../lib/theming/ThemeContext");


var _InputLikeText = require("./InputLikeText.styles");
var _HiddenInput = require("./HiddenInput");var












InputLikeText = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(InputLikeText, _React$Component);function InputLikeText() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.




    state = { blinking: false, focused: false };_this.

    theme = void 0;_this.
    node = null;_this.
    hiddenInput = null;_this.
    lastSelectedInnerNode = null;_this.
    frozen = false;_this.
    frozenBlur = false;_this.
    dragging = false;_this.
    focusTimeout = void 0;_this.
    blinkTimeout = void 0;_this.



































    selectInnerNode = function (node, start, end) {if (start === void 0) {start = 0;}if (end === void 0) {end = 1;}
      if (_this.dragging || !node) {
        return;
      }
      _this.frozen = true;
      _this.frozenBlur = true;

      _this.lastSelectedInnerNode = [node, start, end];
      setTimeout(function () {return (0, _SelectionHelpers.selectNodeContents)(node, start, end);}, 0);
      if (_this.focusTimeout) {
        clearInterval(_this.focusTimeout);
      }
      _this.focusTimeout = window.setTimeout(function () {return (_utils.isIE11 || _utils.isEdge) && _this.node && _this.node.focus();}, 0);
    };_this.






































































































    renderLeftIcon = function () {
      return _this.renderIcon(_this.props.leftIcon, _Input.jsStyles.leftIcon());
    };_this.

    renderRightIcon = function () {
      return _this.renderIcon(_this.props.rightIcon, _Input.jsStyles.rightIcon());
    };_this.

    renderIcon = function (icon, className) {
      if (!icon) {
        return null;
      }

      if (icon instanceof Function) {
        return _react.default.createElement("span", { className: className }, icon());
      }

      return _react.default.createElement("span", { className: (0, _classnames.default)(className, _Input.jsStyles.useDefaultColor(_this.theme)) }, icon);
    };_this.

    renderPrefix = function () {var
      prefix = _this.props.prefix;

      if (!prefix) {
        return null;
      }

      return _react.default.createElement("span", { className: _Input.jsStyles.prefix(_this.theme) }, prefix);
    };_this.

    renderSuffix = function () {var
      suffix = _this.props.suffix;

      if (!suffix) {
        return null;
      }

      return _react.default.createElement("span", { className: _Input.jsStyles.suffix(_this.theme) }, suffix);
    };_this.

    renderLeftSide = function () {
      var leftIcon = _this.renderLeftIcon();
      var prefix = _this.renderPrefix();

      if (!leftIcon && !prefix) {
        return null;
      }

      return (
        _react.default.createElement("span", { className: _Input.jsStyles.sideContainer() },
        leftIcon,
        prefix));


    };_this.

    renderRightSide = function () {
      var rightIcon = _this.renderRightIcon();
      var suffix = _this.renderSuffix();

      if (!rightIcon && !suffix) {
        return null;
      }

      return (
        _react.default.createElement("span", { className: _Input.jsStyles.sideContainer() },
        rightIcon,
        suffix));


    };_this.

    renderPlaceholder = function () {var _this$props =
      _this.props,children = _this$props.children,placeholder = _this$props.placeholder;

      if (!children && placeholder) {
        return _react.default.createElement("span", { className: _Input.jsStyles.placeholder(_this.theme) }, placeholder);
      }
      return null;
    };_this.

    handleDocumentMouseDown = function (e) {
      if (_this.state.focused && _this.node && e.target instanceof Node && !_this.node.contains(e.target)) {
        _this.defrost();
      }
    };_this.

    handleDocumentKeyDown = function (e) {
      if (_this.state.focused && (0, _identifiers.isKeyTab)(e)) {
        _this.defrost();
      }
    };_this.

    handleMouseDown = function (e) {
      _this.frozen = true;
    };_this.

    handleKeyDown = function (e) {
      if (_this.props.disabled) {
        return;
      }

      if (_utils.isIE11 && (0, _identifiers.isShortcutPaste)(e) && _this.hiddenInput) {
        _this.frozen = true;
        setTimeout(function () {
          if (_this.lastSelectedInnerNode) {var _this2;
            (_this2 = _this).selectInnerNode.apply(_this2, _this.lastSelectedInnerNode);
          }
          if (_this.node) {
            _this.node.focus();
          }
        }, 0);

        _this.hiddenInput.focus();
      }

      if (_this.props.onKeyDown) {
        _this.props.onKeyDown(e);
      }
    };_this.

    handleMouseDragStart = function (e) {
      _this.dragging = true;
      document.documentElement.classList.add(_InputLikeText.jsStyles.userSelectNone());

      if (_this.props.onMouseDragStart) {
        _this.props.onMouseDragStart(e);
      }
    };_this.

    handleMouseDragEnd = function (e) {
      // Дожидаемся onMouseUp
      setTimeout(function () {
        _this.dragging = false;

        if (_this.props.onMouseDragEnd) {
          _this.props.onMouseDragEnd(e);
        }
      }, 0);

      document.documentElement.classList.remove(_InputLikeText.jsStyles.userSelectNone());
    };_this.

    handleFocus = function (e) {
      if (_this.props.disabled) {
        if (_utils.isIE11) {
          (0, _SelectionHelpers.selectNodeContents)(document.body, 0, 0);
        }
        return;
      }

      if ((_utils.isIE11 || _utils.isEdge) && _this.frozen) {
        _this.frozen = false;
        if (_this.state.focused) {
          return;
        }
      }

      _this.setState({ focused: true });

      if (_this.props.onFocus) {
        _this.props.onFocus(e);
      }
    };_this.

    handleBlur = function (e) {
      if (_this.props.disabled) {
        e.stopPropagation();
        return;
      }

      if ((_utils.isIE11 || _utils.isEdge) && _this.frozenBlur) {
        _this.frozenBlur = false;
        return;
      }
      if ((_utils.isIE11 || _utils.isEdge) && _this.frozen) {
        return;
      }

      (0, _SelectionHelpers.removeAllSelections)();

      _this.setState({ focused: false });

      if (_this.props.onBlur) {
        _this.props.onBlur(e);
      }
    };_this.

    hiddenInputRef = function (el) {
      _this.hiddenInput = el;
    };_this.

    innerRef = function (el) {
      if (_this.props.innerRef) {
        _this.props.innerRef(el);
      }
      _this.node = el;
    };_this.

    defrost = function () {
      _this.frozen = false;
      _this.frozenBlur = false;
    };_this.

    getSizeClassName = function () {var _ref, _ref2, _ref3;
      switch (_this.props.size) {
        case 'large':
          return _ref = {}, _ref[
          _Input.jsStyles.sizeLarge(_this.theme)] = true, _ref[
          _Input.jsStyles.sizeLargeFallback(_this.theme)] = _utils.isIE11 || _utils.isEdge, _ref;

        case 'medium':
          return _ref2 = {}, _ref2[
          _Input.jsStyles.sizeMedium(_this.theme)] = true, _ref2[
          _Input.jsStyles.sizeMediumFallback(_this.theme)] = _utils.isIE11 || _utils.isEdge, _ref2;

        case 'small':
        default:
          return _ref3 = {}, _ref3[
          _Input.jsStyles.sizeSmall(_this.theme)] = true, _ref3[
          _Input.jsStyles.sizeSmallFallback(_this.theme)] = _utils.isIE11 || _utils.isEdge, _ref3;}


    };return _this;}var _proto = InputLikeText.prototype; /**
                                                           * @public
                                                           */_proto.focus = function focus() {if (this.node) {this.node.focus();}} /**
                                                                                                                                    * @public
                                                                                                                                    */;_proto.blur = function blur() {if (this.node) {this.node.blur();}} /**
                                                                                                                                                                                                           * @public
                                                                                                                                                                                                           */;_proto.blink = function blink() {var _this3 = this;if (this.props.disabled) {return;}this.setState({ blinking: true }, function () {_this3.blinkTimeout = window.setTimeout(function () {return _this3.setState({ blinking: false });}, 150);});};_proto.getNode = function getNode() {return this.node;};_proto.componentDidMount = function componentDidMount() {if (this.node) {_MouseDrag.MouseDrag.listen(this.node).onMouseDragStart(this.handleMouseDragStart).onMouseDragEnd(this.handleMouseDragEnd);}document.addEventListener('mousedown', this.handleDocumentMouseDown);document.addEventListener('keydown', this.handleDocumentKeyDown);};_proto.componentWillUnmount = function componentWillUnmount() {if (this.blinkTimeout) {clearTimeout(this.blinkTimeout);}_MouseDrag.MouseDrag.stop(this.node);document.removeEventListener('mousedown', this.handleDocumentMouseDown);document.removeEventListener('keydown', this.handleDocumentKeyDown);};_proto.render = function render() {var _this4 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this4.theme = theme;return _this4.renderMain();});};_proto.renderMain = function renderMain() {var _cn, _cn2;var _this$props2 = this.props,innerRef = _this$props2.innerRef,tabIndex = _this$props2.tabIndex,placeholder = _this$props2.placeholder,align = _this$props2.align,borderless = _this$props2.borderless,width = _this$props2.width,children = _this$props2.children,error = _this$props2.error,warning = _this$props2.warning,onValueChange = _this$props2.onValueChange,disabled = _this$props2.disabled,prefix = _this$props2.prefix,suffix = _this$props2.suffix,leftIcon = _this$props2.leftIcon,rightIcon = _this$props2.rightIcon,value = _this$props2.value,onMouseDragStart = _this$props2.onMouseDragStart,onMouseDragEnd = _this$props2.onMouseDragEnd,rest = (0, _objectWithoutPropertiesLoose2.default)(_this$props2, ["innerRef", "tabIndex", "placeholder", "align", "borderless", "width", "children", "error", "warning", "onValueChange", "disabled", "prefix", "suffix", "leftIcon", "rightIcon", "value", "onMouseDragStart", "onMouseDragEnd"]);var _this$state = this.state,focused = _this$state.focused,blinking = _this$state.blinking;var leftSide = this.renderLeftSide();var rightSide = this.renderRightSide();var className = (0, _classnames.default)(_InputLikeText.jsStyles.root(), _Input.jsStyles.root(this.theme), this.getSizeClassName(), (_cn = {}, _cn[_Input.jsStyles.borderless()] = !!borderless, _cn[_InputLikeText.jsStyles.withoutLeftSide()] = !leftSide, _cn[_Input.jsStyles.focus(this.theme)] = focused, _cn[_Input.jsStyles.blink(this.theme)] = blinking, _cn[_Input.jsStyles.warning(this.theme)] = !!warning, _cn[_Input.jsStyles.error(this.theme)] = !!error, _cn[_Input.jsStyles.disabled(this.theme)] = !!disabled, _cn[_Input.jsStyles.focusFallback(this.theme)] = focused && (_utils.isIE11 || _utils.isEdge), _cn[_Input.jsStyles.warningFallback(this.theme)] = !!warning && (_utils.isIE11 || _utils.isEdge), _cn[_Input.jsStyles.errorFallback(this.theme)] = !!error && (_utils.isIE11 || _utils.isEdge), _cn));var wrapperClass = (0, _classnames.default)(_Input.jsStyles.wrapper(), (_cn2 = {}, _cn2[_InputLikeText.jsStyles.userSelectContain()] = focused, _cn2));return _react.default.createElement("span", (0, _extends2.default)({}, rest, { className: className, style: { width: width, textAlign: align }, tabIndex: disabled ? undefined : 0, onFocus: this.handleFocus, onBlur: this.handleBlur, ref: this.innerRef, onKeyDown: this.handleKeyDown, onMouseDown: this.handleMouseDown }), _react.default.createElement("input", { type: "hidden", value: value }), leftSide, _react.default.createElement("span", { className: wrapperClass }, _react.default.createElement("span", { "data-tid": "InputLikeText__input", className: (0, _classnames.default)(_InputLikeText.jsStyles.input(), _Input.jsStyles.input(this.theme)) }, children), this.renderPlaceholder()), rightSide, _utils.isIE11 && focused && _react.default.createElement(_HiddenInput.HiddenInput, { nodeRef: this.hiddenInputRef }));};return InputLikeText;}(_react.default.Component);exports.InputLikeText = InputLikeText;InputLikeText.__KONTUR_REACT_UI__ = 'InputLikeText';InputLikeText.defaultProps = { size: 'small' };
