// @skbkontur/react-ui/cjs/components/Logotype/Logotype.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: flex;\n      align-items: center;\n      margin: 0;\n      cursor: pointer;\n      outline: 0;\n      padding: 0 10px;\n      position: relative;\n      background: none;\n      border: none;\n\n      &:hover,\n      &:focus {\n        background: rgba(0, 0, 0, 0.06);\n      }\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: flex;\n      align-items: center;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        display: inline-flex;\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      display: flex;\n      flex-wrap: nowrap;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: absolute;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      position: relative;\n      line-height: 0;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 60%;\n      margin: 0 0 2px 15px;\n      width: 1px;\n      background-color: ", ";\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-weight: 400;\n      text-decoration: none;\n      position: relative;\n      bottom: 2px;\n      outline: none;\n      user-select: none;\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root(t) {
    return (0, _Emotion.css)(_templateObject(),






    t.logoColor,


    t.logoHoverColor);


  },

  divider: function divider(t) {
    return (0, _Emotion.css)(_templateObject2(),



    t.tdDividerBg);

  },

  cloud: function cloud() {
    return (0, _Emotion.css)(_templateObject3());




  },

  inner: function inner() {
    return (0, _Emotion.css)(_templateObject4());






  },

  dropdown: function dropdown() {
    return (0, _Emotion.css)(_templateObject5());




  },

  inline: function inline() {
    return (0, _Emotion.css)(_templateObject6(),
    (0, _Emotion.cssName)(styles.dropdown()));



  },

  widgetWrapper: function widgetWrapper() {
    return (0, _Emotion.css)(_templateObject7());



  },

  button: function button() {
    return (0, _Emotion.css)(_templateObject8());















  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
