// @skbkontur/react-ui/components/Fias/types/types.js
export var FiasEstateStatuses;

(function (FiasEstateStatuses) {
  FiasEstateStatuses["Hold"] = "Hold";
  FiasEstateStatuses["House"] = "House";
  FiasEstateStatuses["HouseHold"] = "HouseHold";
  FiasEstateStatuses["None"] = "None";
})(FiasEstateStatuses || (FiasEstateStatuses = {}));

export var FiasStructureStatuses;

(function (FiasStructureStatuses) {
  FiasStructureStatuses["Structure"] = "Structure";
  FiasStructureStatuses["Construction"] = "Construction";
  FiasStructureStatuses["Liter"] = "Liter";
  FiasStructureStatuses["None"] = "None";
})(FiasStructureStatuses || (FiasStructureStatuses = {}));

export var FiasLiveStatuses;

(function (FiasLiveStatuses) {
  FiasLiveStatuses["active"] = "active";
  FiasLiveStatuses["inactive"] = "inactive";
})(FiasLiveStatuses || (FiasLiveStatuses = {}));

export var FiasFields;

(function (FiasFields) {
  FiasFields["region"] = "region";
  FiasFields["district"] = "district";
  FiasFields["city"] = "city";
  FiasFields["intracityarea"] = "intracityarea";
  FiasFields["settlement"] = "settlement";
  FiasFields["planningstructure"] = "planningstructure";
  FiasFields["street"] = "street";
  FiasFields["stead"] = "stead";
  FiasFields["house"] = "house";
  FiasFields["room"] = "room";
})(FiasFields || (FiasFields = {}));

export var FiasExtraFields;

(function (FiasExtraFields) {
  FiasExtraFields["postalcode"] = "postalcode";
})(FiasExtraFields || (FiasExtraFields = {}));

export var FiasFormValidation;

(function (FiasFormValidation) {
  FiasFormValidation["Error"] = "Error";
  FiasFormValidation["Warning"] = "Warning";
  FiasFormValidation["None"] = "None";
})(FiasFormValidation || (FiasFormValidation = {}));
