// @skbkontur/react-ui/components/Logotype/Logotype.styles/Logotype.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      display: flex;\n      align-items: center;\n      margin: 0;\n      cursor: pointer;\n      outline: 0;\n      padding: 0 10px;\n      position: relative;\n      background: none;\n      border: none;\n\n      &:hover,\n      &:focus {\n        background: rgba(0, 0, 0, 0.06);\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      display: flex;\n      align-items: center;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        display: inline-flex;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      display: flex;\n      flex-wrap: nowrap;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      position: relative;\n      line-height: 0;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 60%;\n      margin: 0 0 2px 15px;\n      width: 1px;\n      background-color: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      font-weight: 400;\n      text-decoration: none;\n      position: relative;\n      bottom: 2px;\n      outline: none;\n      user-select: none;\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.logoColor, t.logoHoverColor);
  },
  divider: function divider(t) {
    return css(_templateObject2(), t.tdDividerBg);
  },
  cloud: function cloud() {
    return css(_templateObject3());
  },
  inner: function inner() {
    return css(_templateObject4());
  },
  dropdown: function dropdown() {
    return css(_templateObject5());
  },
  inline: function inline() {
    return css(_templateObject6(), cssName(styles.dropdown()));
  },
  widgetWrapper: function widgetWrapper() {
    return css(_templateObject7());
  },
  button: function button() {
    return css(_templateObject8());
  }
};
export var jsStyles = memoizeStyle(styles);
