// @skbkontur/react-ui/components/TokenInput/TextWidthHelper/TextWidthHelper.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { RenderContainer } from "../../../internal/RenderContainer";
var styles = {
  position: 'absolute',
  left: -1000,
  top: -1000,
  fontSize: 14
};
export var TextWidthHelper =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(TextWidthHelper, _React$Component);

  function TextWidthHelper() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.element = null;

    _this.elementRef = function (node) {
      return _this.element = node;
    };

    return _this;
  }

  var _proto = TextWidthHelper.prototype;

  _proto.render = function render() {
    return React.createElement(RenderContainer, null, React.createElement("div", {
      style: styles,
      ref: this.elementRef
    }, this.props.text));
  };

  _proto.getTextWidth = function getTextWidth() {
    return this.element.getBoundingClientRect().width;
  };

  return TextWidthHelper;
}(React.Component);
