// @skbkontur/react-ui/components/SidePage/SidePage.styles/SidePage.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject26() {
  var data = _taggedTemplateLiteralLoose(["\n      opacity: 0.01;\n      transition: opacity 0.15s ease-out;\n    "]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _taggedTemplateLiteralLoose(["\n      opacity: 1;\n    "]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _taggedTemplateLiteralLoose(["\n      transition: transform 0.18s cubic-bezier(0.22, 0.61, 0.36, 1), opacity 0.18s cubic-bezier(0.22, 0.61, 0.36, 1);\n      opacity: 1;\n      transform: translate(0, 0) !important;\n    "]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _taggedTemplateLiteralLoose(["\n      transform: translateX(-100px);\n    "]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _taggedTemplateLiteralLoose(["\n      transform: translateX(100px);\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      border-top: 1px solid ", ";\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: 20px 35px 20px 30px;\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n      bottom: 0;\n      position: fixed;\n      z-index: 10;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      line-height: 24px;\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      height: 12px;\n      line-height: 0;\n      width: 12px;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", " !important;\n      border-top: 0 !important;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      color: ", ";\n      cursor: pointer;\n      position: absolute;\n      right: 30px;\n      text-align: center;\n      text-decoration: none;\n      width: 24px;\n\n      ", "& {\n        line-height: 24px;\n      }\n\n      &:hover {\n        color: ", ";\n      }\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      left: 0;\n      right: auto;\n\n      & ", " {\n        float: left;\n      }\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "& {\n        box-shadow: ", ";\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      background: #333;\n      opacity: 0.6;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      position: absolute;\n      width: 100%;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      overflow: hidden;\n      text-overflow: ellipsis;\n      white-space: nowrap;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-right: 85px;\n      padding-left: 30px;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      box-shadow: 0 1px ", ";\n      font-size: 18px;\n      line-height: 24px;\n      padding: 13px 0;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      font-size: 22px;\n      line-height: 30px;\n      padding: 25px 0;\n      position: relative;\n      width: 100%;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      align-items: stretch;\n      display: flex;\n      flex-direction: column;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      float: right;\n      height: 100%;\n      width: 100%;\n      overflow-y: auto;\n      position: relative;\n      white-space: normal;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 30px;\n      padding-right: 35px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      z-index: 0;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      position: fixed;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      height: 100%;\n      position: fixed;\n      right: 0;\n      top: 0;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root() {
    return css(_templateObject());
  },
  overlay: function overlay() {
    return css(_templateObject2());
  },
  body: function body() {
    return css(_templateObject3());
  },
  bodyContainer: function bodyContainer() {
    return css(_templateObject4());
  },
  container: function container(t) {
    return css(_templateObject5(), t.bgDefault);
  },
  layout: function layout() {
    return css(_templateObject6());
  },
  header: function header() {
    return css(_templateObject7());
  },
  headerFixed: function headerFixed(t) {
    return css(_templateObject8(), t.bgDefault, t.borderColorGrayLight);
  },
  title: function title() {
    return css(_templateObject9());
  },
  titleFixed: function titleFixed() {
    return css(_templateObject10());
  },
  background: function background() {
    return css(_templateObject11());
  },
  backgroundGray: function backgroundGray() {
    return css(_templateObject12());
  },
  shadow: function shadow(t) {
    return css(_templateObject13(), cssName(styles.container(t)), t.sidePageContainerShadow);
  },
  leftSide: function leftSide(t) {
    return css(_templateObject14(), cssName(styles.container(t)));
  },
  close: function close(t) {
    return css(_templateObject15(), t.sidePageCloseButtonColor, cssName(styles.fixed()), t.sidePageCloseButtonHoverColor);
  },
  panel: function panel(t) {
    return css(_templateObject16(), t.sidePageFooterPanelBg);
  },
  closeIcon: function closeIcon() {
    return css(_templateObject17());
  },
  fixed: function fixed() {
    return css(_templateObject18());
  },
  footer: function footer() {
    return css(_templateObject19());
  },
  footerContent: function footerContent() {
    return css(_templateObject20());
  },
  footerFixed: function footerFixed(t) {
    return css(_templateObject21(), t.bgDefault, t.borderColorGrayLight);
  },
  transitionLeft: function transitionLeft() {
    return css(_templateObject22());
  },
  transitionRight: function transitionRight() {
    return css(_templateObject23());
  },
  transitionActive: function transitionActive() {
    return css(_templateObject24());
  },
  transitionLeave: function transitionLeave() {
    return css(_templateObject25());
  },
  transitionLeaveActive: function transitionLeaveActive() {
    return css(_templateObject26());
  }
};
export var jsStyles = memoizeStyle(styles);
