// @skbkontur/react-ui/cjs/components/Loader/Loader.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n      ", ";\n\n      &::before {\n        content: ' ';\n        display: inline-block;\n        height: 100%;\n        min-height: 100%;\n        vertical-align: middle;\n      }\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      position: fixed;\n\n      &::before {\n        content: ' ';\n        display: inline-block;\n        height: 100%;\n        min-height: 100%;\n        vertical-align: middle;\n      }\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: block;\n      margin: auto;\n      text-align: center;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      box-sizing: border-box;\n      display: inline-block;\n      position: relative;\n      width: 100%;\n      z-index: inherit;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      bottom: 0;\n      left: 0;\n      position: absolute;\n      right: 0;\n      top: 0;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", ";\n\n      background: ", ";\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  active: function active(t) {
    return (0, _Emotion.css)(_templateObject(),
    styles.fillContainerPosition(),

    t.loaderBg);

  },

  fillContainerPosition: function fillContainerPosition() {
    return (0, _Emotion.css)(_templateObject2());






  },

  loader: function loader() {
    return (0, _Emotion.css)(_templateObject3());






  },

  spinnerContainer: function spinnerContainer() {
    return (0, _Emotion.css)(_templateObject4());




  },

  spinnerContainerSticky: function spinnerContainerSticky() {
    return (0, _Emotion.css)(_templateObject5(),
    styles.spinnerContainer());











  },

  spinnerContainerCenter: function spinnerContainerCenter() {
    return (0, _Emotion.css)(_templateObject6(),
    styles.spinnerContainer(),
    styles.fillContainerPosition());









  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
