// @skbkontur/react-ui/components/Button/Button.styles/Button.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject39() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      border-radius: inherit;\n      overflow: hidden;\n\n      &::before {\n        content: '';\n        height: 100%;\n        position: absolute;\n        opacity: 0.2;\n        background: linear-gradient(-110deg, #ccc 30%, transparent 0, transparent 60%, #ccc 0);\n        background-size: 30px 100%;\n        top: 0;\n        left: 0;\n        right: -30px;\n        animation: ", " 1s linear infinite;\n        transform: rotateY(180deg);\n      }\n    "]);

  _templateObject39 = function _templateObject39() {
    return data;
  };

  return data;
}

function _templateObject38() {
  var data = _taggedTemplateLiteralLoose(["\n    0% {\n      transform: translateX(0) rotateY(180deg);\n    }\n\n    100% {\n      transform: translateX(-30px) rotateY(180deg);\n    }\n  "]);

  _templateObject38 = function _templateObject38() {
    return data;
  };

  return data;
}

function _templateObject37() {
  var data = _taggedTemplateLiteralLoose(["\n      &:not(", "):not(", "):not(", "):not(", ") {\n        &,\n        &:hover,\n        &:active {\n          box-shadow: none !important;\n        }\n      }\n    "]);

  _templateObject37 = function _templateObject37() {
    return data;
  };

  return data;
}

function _templateObject36() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-right: 15px;\n      padding-left: 15px;\n    "]);

  _templateObject36 = function _templateObject36() {
    return data;
  };

  return data;
}

function _templateObject35() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      padding-right: 7px;\n    "]);

  _templateObject35 = function _templateObject35() {
    return data;
  };

  return data;
}

function _templateObject34() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      border-radius: 2px 2px 2px 16px;\n      box-sizing: border-box;\n      z-index: 1;\n    "]);

  _templateObject34 = function _templateObject34() {
    return data;
  };

  return data;
}

function _templateObject33() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-right: 0;\n      margin-left: 10px;\n    "]);

  _templateObject33 = function _templateObject33() {
    return data;
  };

  return data;
}

function _templateObject32() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-right: 10px;\n    "]);

  _templateObject32 = function _templateObject32() {
    return data;
  };

  return data;
}

function _templateObject31() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      padding: 0;\n    "]);

  _templateObject31 = function _templateObject31() {
    return data;
  };

  return data;
}

function _templateObject30() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-right: 0 !important;\n    "]);

  _templateObject30 = function _templateObject30() {
    return data;
  };

  return data;
}

function _templateObject29() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 0 !important;\n      padding-right: 0 !important;\n    "]);

  _templateObject29 = function _templateObject29() {
    return data;
  };

  return data;
}

function _templateObject28() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 5px !important;\n      padding-right: 5px !important;\n    "]);

  _templateObject28 = function _templateObject28() {
    return data;
  };

  return data;
}

function _templateObject27() {
  var data = _taggedTemplateLiteralLoose(["\n      padding: ", ";\n      box-sizing: border-box;\n      display: inline-block;\n    "]);

  _templateObject27 = function _templateObject27() {
    return data;
  };

  return data;
}

function _templateObject26() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      white-space: nowrap;\n      display: inline-block;\n      width: 100%;\n      vertical-align: top;\n    "]);

  _templateObject26 = function _templateObject26() {
    return data;
  };

  return data;
}

function _templateObject25() {
  var data = _taggedTemplateLiteralLoose(["\n      &:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n    "]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: ", " !important;\n      background: ", " !important;\n      color: ", " !important;\n      border: ", " !important;\n\n      &:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n\n      &,\n      &:not(", ") {\n        ", " {\n          background: ", " !important;\n          box-shadow: ", " !important;\n        }\n      }\n    "]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n\n      ", ";\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      visibility: visible;\n\n      ", ":not(", ") & {\n        box-shadow: ", " !important;\n      }\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: 2px -2px 0 0 ", " !important;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: 2px -2px 0 0 ", " !important;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      ", " {\n        box-shadow: inset 0 0 0 1px ", ";\n        border-color: transparent !important;\n      }\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      &", " {\n        outline-color: transparent;\n      }\n      &:not(", ") {\n        line-height: normal !important;\n      }\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: default;\n      pointer-events: none;\n      border-color: transparent !important;\n\n      &:not(", ") {\n        background: ", " !important;\n        color: ", " !important;\n        box-shadow: ", " !important;\n\n        ", " {\n          background: ", " !important;\n          box-shadow: ", " !important;\n        }\n      }\n\n      &", " {\n        color: ", " !important;\n      }\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      position: relative;\n      z-index: 2;\n\n      &", " {\n        color: ", ";\n        text-decoration: ", ";\n      }\n\n      &:not(", "):not(", ") {\n        border: ", ";\n\n        &,\n        &:hover,\n        &:active,\n        ", " {\n          box-shadow: inset 0 0 0 1px ", ", 0 0 0 ", " ", ";\n\n          &", ", &", " {\n            box-shadow: inset 0 0 0 1px ", " !important;\n            border-color: transparent !important;\n          }\n          ", " {\n            box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", ";\n\n            &", " {\n              box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", " !important;\n            }\n\n            &", " {\n              box-shadow: inset -1px 1px 0 0 ", ", 2px -2px 0 0 ", " !important;\n            }\n          }\n        }\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      background: none;\n      border-radius: ", " !important;\n      border: none;\n      box-shadow: none;\n      color: ", " !important;\n      display: inline;\n      line-height: inherit !important;\n      margin: 0;\n      padding: 0;\n\n      &:hover {\n        color: ", ";\n        text-decoration: ", ";\n      }\n      &:active {\n        color: ", " !important;\n      }\n      ", " {\n        display: inline;\n      }\n      ", " {\n        padding-right: ", ";\n      }\n      ", " ,\n      ", "  {\n        box-shadow: none;\n        left: -2px !important;\n        right: -2px !important;\n        bottom: -2px !important;\n      }\n      ", "  {\n        background: ", " !important;\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      border-radius: ", ";\n\n      ", ";\n\n      ", " {\n        border-radius: ", ";\n      }\n\n      ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      border-radius: inherit;\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      box-shadow: 0 0 0 2px ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      border-radius: inherit;\n      position: absolute;\n      top: 0;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      box-shadow: 0 0 0 2px ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n      ", ";\n\n      background-clip: border-box;\n      background-position: center;\n      background-repeat: no-repeat;\n      background-size: contain;\n      color: ", ";\n      cursor: pointer;\n      display: inline-block;\n      position: relative;\n      text-align: center;\n      width: 100%;\n\n      &::-moz-focus-inner {\n        border: 0;\n        padding: 0;\n      }\n      &::after {\n        content: '';\n        display: inline-block;\n        vertical-align: baseline;\n        width: 0;\n      }\n      &:not(", ") {\n        border-radius: ", ";\n      }\n      ", "& {\n        padding: 0;\n      }\n      &:active:not(", "):not(", ") {\n        ", " {\n          transform: translateY(1px) !important;\n        }\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n0% {\n  transform: translateX(50px) rotate(-44.3deg);\n}\n\n100% {\n  transform: translateX(21px) translateY(30px) rotate(-44.3deg);\n}\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, keyframes, memoizeStyle } from "../../../lib/theming/Emotion";
import { resetButton, resetText } from "../../../lib/styles/Mixins";
import { buttonUseMixin, buttonHoverMixin, buttonActiveMixin, buttonSizeMixin, buttonArrowMixin, buttonLoadingArrowMixin } from "../Button.mixins";
var btn_loading_arrow = keyframes(_templateObject());
var styles = {
  root: function root(t) {
    return css(_templateObject2(), resetButton(), resetText(), t.textColorDefault, cssName(styles.sizeSmall(t)), t.btnBorderRadius, cssName(styles.link(t)), cssName(styles.link(t)), cssName(styles.disabled(t)), cssName(styles.caption()));
  },
  warning: function warning(t) {
    return css(_templateObject3(), t.borderColorWarning);
  },
  error: function error(t) {
    return css(_templateObject4(), t.borderColorError);
  },
  sizeSmall: function sizeSmall(t) {
    return css(_templateObject5(), t.btnSmallBorderRadius, buttonSizeMixin(t.btnFontSizeSmall, t.controlHeightSmall, t.btnHeightShift, t.controlLineHeightSmall, t.btnPaddingXSmall, t.btnPaddingYSmall, cssName(styles.link(t)), cssName(styles.fallback(t))), cssName(styles.arrow()), t.btnSmallArrowBorderRadius, buttonArrowMixin(t.btnSmallArrowTop, t.btnSmallArrowLeft, t.btnSmallArrowRight, t.btnSmallArrowLength, 'rotate(53deg) skewX(24deg) skewY(10deg)', cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  sizeMedium: function sizeMedium(t) {
    return css(_templateObject6(), buttonSizeMixin(t.btnFontSizeMedium, t.controlHeightMedium, t.btnHeightShift, t.controlLineHeightMedium, t.btnPaddingXMedium, t.btnPaddingYMedium, cssName(styles.link(t)), cssName(styles.fallback(t))), buttonArrowMixin('9px', t.btnMediumArrowLeft, t.btnMediumArrowRight, '20.2px', t.btnMediumArrowTransform, cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  sizeLarge: function sizeLarge(t) {
    return css(_templateObject7(), buttonSizeMixin(t.btnFontSizeLarge, t.controlHeightLarge, t.btnHeightShift, t.controlLineHeightLarge, t.btnPaddingXLarge, t.btnPaddingYLarge, cssName(styles.link(t)), cssName(styles.fallback(t))), buttonArrowMixin('10.2px', t.btnLargeArrowLeft, '-10.8px', '22.2px', t.btnLargeArrowTransform, cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  sizeSmallLoading: function sizeSmallLoading(t) {
    return css(_templateObject8(), buttonLoadingArrowMixin(t.btnSmallArrowTop, t.btnSmallArrowTop, '-207px', '441%', t.btnSmallArrowBg, t.btnSmallArrowLeftLoadingDelay, btn_loading_arrow, cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  sizeMediumLoading: function sizeMediumLoading(t) {
    return css(_templateObject9(), buttonLoadingArrowMixin('0', '0', t.btnMediumArrowLeftLoadingLeft, '441%', t.btnMediumArrowBg, t.btnMediumArrowLeftLoadingDelay, btn_loading_arrow, cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  sizeLargeLoading: function sizeLargeLoading(t) {
    return css(_templateObject10(), buttonLoadingArrowMixin('-32px', '-36px', ' -198px', '700%', t.btnLargeArrowBg, t.btnLargeArrowLeftLoadingDelay, btn_loading_arrow, cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  link: function link(t) {
    return css(_templateObject11(), t.btnLinkBorderRadius, t.linkColor, t.linkHoverColor, t.linkHoverTextDecoration, t.linkActiveColor, cssName(styles.caption()), cssName(styles.icon()), t.linkIconMarginRight, cssName(styles.warning(t)), cssName(styles.error(t)), cssName(styles.error(t)), t.errorSecondary);
  },
  focus: function focus(t) {
    return css(_templateObject12(), cssName(styles.link(t)), t.linkColor, t.linkHoverTextDecoration, cssName(styles.disabled(t)), cssName(styles.link(t)), t.btnFocusBorder, cssName(styles.active(t)), t.outlineColorFocus, t.btnFocusShadowWidth, t.borderColorFocus, cssName(styles.warning(t)), cssName(styles.error(t)), t.outlineColorFocus, cssName(styles.arrow()), t.outlineColorFocus, t.borderColorFocus, cssName(styles.arrowWarning(t)), t.outlineColorFocus, t.borderColorWarning, cssName(styles.arrowError(t)), t.outlineColorFocus, t.borderColorError);
  },
  disabled: function disabled(t) {
    return css(_templateObject13(), cssName(styles.link(t)), t.btnDisabledBg, t.btnDisabledTextColor, t.btnDisabledShadow, cssName(styles.arrow()), t.btnDisabledBg, t.btnDisabledShadowArrow, cssName(styles.link(t)), t.linkDisabledColor);
  },
  fallback: function fallback(t) {
    return css(_templateObject14(), cssName(styles.disabled(t)), cssName(styles.link(t)));
  },
  validationRoot: function validationRoot(t) {
    return css(_templateObject15(), cssName(styles.focus(t)), t.outlineColorFocus);
  },
  arrowWarning: function arrowWarning(t) {
    return css(_templateObject16(), t.borderColorWarning);
  },
  arrowError: function arrowError(t) {
    return css(_templateObject17(), t.borderColorError);
  },
  arrowLeft: function arrowLeft(t) {
    return css(_templateObject18(), cssName(styles.checked(t)), cssName(styles.focus(t)), t.btnCheckedShadowArrowLeft);
  },
  "default": function _default(t) {
    return css(_templateObject19(), buttonUseMixin(t.btnDefaultBg, t.btnDefaultBgStart, t.btnDefaultBgEnd, t.btnDefaultBgArrowStart, t.btnDefaultBgArrowEnd, t.btnDefaultShadow, t.btnDefaultShadowArrow, t.btnDefaultShadowArrowLeft, t.btnDefaultTextColor, t.btnDefaultBorder, cssName(styles.checked(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonHoverMixin(t.btnDefaultHoverBg, t.btnDefaultHoverBgStart, t.btnDefaultHoverBgEnd, t.btnDefaultHoverBgStart, t.btnDefaultHoverBgEnd, t.btnDefaultHoverShadow, t.btnDefaultHoverShadowArrow, t.btnDefaultHoverShadowArrowLeft, t.btnDefaultHoverBorderColor, cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonActiveMixin(t.btnDefaultActiveBg, t.btnDefaultActiveBg, t.btnDefaultActiveBg, t.btnDefaultActiveShadow, t.btnDefaultActiveShadowArrow, t.btnDefaultActiveShadowArrowLeft, cssName(styles.active(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  primary: function primary(t) {
    return css(_templateObject20(), buttonUseMixin(t.btnPrimaryBg, t.btnPrimaryBgStart, t.btnPrimaryBgEnd, t.btnPrimaryBgArrowStart, t.btnPrimaryBgArrowEnd, t.btnPrimaryShadow, t.btnPrimaryShadowArrow, t.btnPrimaryShadowArrowLeft, t.btnPrimaryTextColor, t.btnPrimaryBorder, cssName(styles.checked(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonHoverMixin(t.btnPrimaryHoverBg, t.btnPrimaryHoverBgStart, t.btnPrimaryHoverBgEnd, t.btnPrimaryHoverBgStart, t.btnPrimaryHoverBgEnd, t.btnPrimaryHoverShadow, t.btnPrimaryHoverShadowArrow, t.btnPrimaryHoverShadowArrowLeft, t.btnPrimaryHoverBorderColor, cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonActiveMixin(t.btnPrimaryActiveBg, t.btnPrimaryActiveBg, t.btnPrimaryActiveBg, t.btnPrimaryActiveShadow, t.btnPrimaryActiveShadowArrow, t.btnPrimaryActiveShadowArrowLeft, cssName(styles.active(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  success: function success(t) {
    return css(_templateObject21(), buttonUseMixin(t.btnSuccessBg, t.btnSuccessBgStart, t.btnSuccessBgEnd, t.btnSuccessBgArrowStart, t.btnSuccessBgArrowEnd, t.btnSuccessShadow, t.btnSuccessShadowArrow, t.btnSuccessShadowArrowLeft, t.btnSuccessTextColor, t.btnSuccessBorder, cssName(styles.checked(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonHoverMixin(t.btnSuccessHoverBg, t.btnSuccessHoverBgStart, t.btnSuccessHoverBgEnd, t.btnSuccessHoverBgStart, t.btnSuccessHoverBgEnd, t.btnSuccessHoverShadow, t.btnSuccessHoverShadowArrow, t.btnSuccessHoverShadowArrowLeft, t.btnSuccessHoverBorderColor, cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonActiveMixin(t.btnSuccessActiveBg, t.btnSuccessActiveBg, t.btnSuccessActiveBg, t.btnSuccessActiveShadow, t.btnSuccessActiveShadowArrow, t.btnSuccessActiveShadowArrowLeft, cssName(styles.active(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  danger: function danger(t) {
    return css(_templateObject22(), buttonUseMixin(t.btnDangerBg, t.btnDangerBgStart, t.btnDangerBgEnd, t.btnDangerBgArrowStart, t.btnDangerBgArrowEnd, t.btnDangerShadow, t.btnDangerShadowArrow, t.btnDangerShadowArrowLeft, t.btnDangerTextColor, t.btnDangerBorder, cssName(styles.checked(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonHoverMixin(t.btnDangerHoverBg, t.btnDangerHoverBgStart, t.btnDangerHoverBgEnd, t.btnDangerHoverBgStart, t.btnDangerHoverBgEnd, t.btnDangerHoverShadow, t.btnDangerHoverShadowArrow, t.btnDangerHoverShadowArrowLeft, t.btnDangerHoverBorderColor, cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonActiveMixin(t.btnDangerActiveBg, t.btnDangerActiveBg, t.btnDangerActiveBg, t.btnDangerActiveShadow, t.btnDangerActiveShadowArrow, t.btnDangerActiveShadowArrowLeft, cssName(styles.active(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  pay: function pay(t) {
    return css(_templateObject23(), buttonUseMixin(t.btnPayBg, t.btnPayBgStart, t.btnPayBgEnd, t.btnPayBgArrowStart, t.btnPayBgArrowEnd, t.btnPayShadow, t.btnPayShadowArrow, t.btnPayShadowArrowLeft, t.btnPayTextColor, t.btnPayBorder, cssName(styles.checked(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonHoverMixin(t.btnPayHoverBg, t.btnPayHoverBgStart, t.btnPayHoverBgEnd, t.btnPayHoverBgStart, t.btnPayHoverBgEnd, t.btnPayHoverShadow, t.btnPayHoverShadowArrow, t.btnPayHoverShadowArrowLeft, t.btnPayHoverBorderColor, cssName(styles.arrow()), cssName(styles.arrowLeft(t))), buttonActiveMixin(t.btnPayActiveBg, t.btnPayActiveBg, t.btnPayActiveBg, t.btnPayActiveShadow, t.btnPayActiveShadowArrow, t.btnPayActiveShadowArrowLeft, cssName(styles.active(t)), cssName(styles.arrow()), cssName(styles.arrowLeft(t))));
  },
  checked: function checked(t) {
    return css(_templateObject24(), t.btnCheckedShadow, t.btnCheckedBg, t.btnCheckedTextColor, t.btnDefaultCheckedBorder, cssName(styles.link(t)), cssName(styles.disabled(t)), cssName(styles.caption()), cssName(styles.focus(t)), cssName(styles.arrow()), t.btnCheckedBg, t.btnCheckedShadowArrow);
  },
  active: function active(t) {
    return css(_templateObject25(), cssName(styles.link(t)), cssName(styles.disabled(t)), cssName(styles.caption()));
  },
  caption: function caption() {
    return css(_templateObject26());
  },
  wrap: function wrap(t) {
    return css(_templateObject27(), t.btnWrapPadding);
  },
  narrow: function narrow() {
    return css(_templateObject28());
  },
  noPadding: function noPadding() {
    return css(_templateObject29());
  },
  noRightPadding: function noRightPadding() {
    return css(_templateObject30());
  },
  wrapLink: function wrapLink(t) {
    return css(_templateObject31(), styles.wrap(t));
  },
  wrapArrow: function wrapArrow() {
    return css(_templateObject32());
  },
  wrapArrowLeft: function wrapArrowLeft() {
    return css(_templateObject33());
  },
  arrow: function arrow() {
    return css(_templateObject34());
  },
  icon: function icon() {
    return css(_templateObject35());
  },
  buttonWithIcon: function buttonWithIcon() {
    return css(_templateObject36());
  },
  borderless: function borderless(t) {
    var focus = cssName(styles.focus(t));
    var disabled = cssName(styles.disabled(t));
    var checked = cssName(styles.checked(t));
    var active = cssName(styles.active(t));
    return css(_templateObject37(), focus, disabled, active, checked);
  },
  loading: function loading() {
    var btn_loading = keyframes(_templateObject38());
    return css(_templateObject39(), btn_loading);
  }
};
export var jsStyles = memoizeStyle(styles);
