// @skbkontur/react-ui/components/TopBar/TopBar.styles/TopBar.styles.js
import _taggedTemplateLiteralLoose from "@babel/runtime/helpers/esm/taggedTemplateLiteralLoose";

function _templateObject22() {
  var data = _taggedTemplateLiteralLoose(["\n      padding-left: 10px;\n    "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = _taggedTemplateLiteralLoose(["\n      box-sizing: border-box;\n      height: 0;\n      overflow: hidden;\n      padding-right: 10px;\n      white-space: normal;\n    "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n      position: absolute;\n      right: 0;\n      text-align: center;\n      width: 20px;\n    "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n      color: #aaa;\n      padding-left: 10px;\n      position: absolute;\n      right: 20px;\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n      box-sizing: border-box;\n      left: 0;\n      overflow: hidden;\n      padding: 0 20px 0 10px;\n      position: absolute;\n      text-overflow: ellipsis;\n      width: 100%;\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n      display: table-cell;\n      height: 50px;\n      white-space: nowrap;\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n      display: inline-block;\n      box-sizing: border-box;\n      height: 50px;\n      line-height: 50px;\n      padding: 0 10px;\n      vertical-align: middle;\n      white-space: nowrap;\n    "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n      display: flex;\n      flex-wrap: nowrap;\n      justify-content: space-between;\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: 0 -10px;\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n      margin: auto;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-bottom: 0;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n      &,\n      &:hover,\n      ", " {\n        background: #ffe3e3;\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n      &,\n      &:hover,\n      ", " {\n        background: #ffeca9;\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-right: 0;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n      margin-right: 4px;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n      box-shadow: none;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      cursor: pointer;\n      height: 50px;\n      line-height: 50px;\n      outline: 0;\n      padding: 0 10px;\n      position: relative;\n      vertical-align: middle;\n\n      &:hover,\n      &:focus {\n        background: rgba(0, 0, 0, 0.06);\n      }\n\n      .kontur-projects_opened & {\n        background: rgba(0, 0, 0, 0.06);\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n      background: rgba(0, 0, 0, 0.06);\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      ", ";\n\n      padding-right: 60px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n      align-items: stretch;\n      display: flex;\n      flex-wrap: nowrap;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      align-self: center;\n      background-color: ", ";\n      height: 30px;\n      width: 1px;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n      background: ", ";\n      box-shadow: ", ";\n      height: 50px;\n      margin-bottom: 20px;\n      padding: 0 15px;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { css, cssName, memoizeStyle } from "../../../lib/theming/Emotion";
var styles = {
  root: function root(t) {
    return css(_templateObject(), t.tbBg, t.tbShadow);
  },
  divider: function divider(t) {
    return css(_templateObject2(), t.tdDividerBg);
  },
  endItems: function endItems() {
    return css(_templateObject3());
  },
  startItems: function startItems() {
    return css(_templateObject4(), styles.endItems());
  },
  buttonActive: function buttonActive() {
    return css(_templateObject5());
  },
  button: function button() {
    return css(_templateObject6());
  },
  noShadow: function noShadow() {
    return css(_templateObject7());
  },
  icon: function icon() {
    return css(_templateObject8());
  },
  iconOnly: function iconOnly() {
    return css(_templateObject9());
  },
  usePay: function usePay() {
    return css(_templateObject10(), cssName(styles.buttonActive()));
  },
  useDanger: function useDanger() {
    return css(_templateObject11(), cssName(styles.buttonActive()));
  },
  noMargin: function noMargin() {
    return css(_templateObject12());
  },
  center: function center() {
    return css(_templateObject13());
  },
  containerWrap: function containerWrap() {
    return css(_templateObject14());
  },
  container: function container() {
    return css(_templateObject15());
  },
  item: function item() {
    return css(_templateObject16());
  },
  spwDropdown: function spwDropdown() {
    return css(_templateObject17());
  },
  organizationsTitle: function organizationsTitle() {
    return css(_templateObject18());
  },
  organizationsComment: function organizationsComment() {
    return css(_templateObject19());
  },
  organizationsArrow: function organizationsArrow() {
    return css(_templateObject20());
  },
  organizationsTitleDummy: function organizationsTitleDummy() {
    return css(_templateObject21());
  },
  organizationsCommentDummy: function organizationsCommentDummy() {
    return css(_templateObject22());
  }
};
export var jsStyles = memoizeStyle(styles);
