// @skbkontur/react-ui/internal/ThemePlayground/PagingPlayground/PagingPlayground.js
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import React from 'react';
import { Paging } from "../../../components/Paging";
export var PagingPlayground =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(PagingPlayground, _React$Component);

  function PagingPlayground() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.state = {
      active: 1
    };

    _this.handlePageChange = function (pageNumber) {
      _this.setState({
        active: pageNumber
      });
    };

    return _this;
  }

  var _proto = PagingPlayground.prototype;

  _proto.render = function render() {
    return React.createElement(Paging, {
      activePage: this.state.active,
      onPageChange: this.handlePageChange,
      pagesCount: 12
    });
  };

  return PagingPlayground;
}(React.Component);
