// @skbkontur/react-ui/cjs/components/SidePage/SidePageHeader.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.SidePageHeader = void 0;var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));

var _Sticky = require("../Sticky");
var _CrossIcon = require("../../internal/icons/CrossIcon");
var _utils = require("../../lib/utils");
var _ThemeContext = require("../../lib/theming/ThemeContext");


var _SidePage = require("./SidePage.styles");
var _SidePageContext = require("./SidePageContext");

var REGULAR_HEADER_PADDING_TOP = 25;
var FIXED_HEADER_PADDING_TOP = 13;
var FONT_FAMILY_CORRECTION = 1;
var CLOSE_ELEMENT_OFFSET = REGULAR_HEADER_PADDING_TOP - FIXED_HEADER_PADDING_TOP - FONT_FAMILY_CORRECTION;
var FIXED_HEADER_HEIGHT = 50;









/**
                               * Шапка сайдпейджа
                               *
                               * @visibleName SidePage.Header
                               */var
SidePageHeader = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(SidePageHeader, _React$Component);function SidePageHeader() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.


    state = {
      isReadyToFix: false };_this.


    theme = void 0;_this.
    wrapper = null;_this.
    lastRegularHeight = 0;_this.












    componentDidMount = function () {
      window.addEventListener('scroll', _this.update, true);
    };_this.

    componentWillUnmount = function () {
      window.removeEventListener('scroll', _this.update, true);
    };_this.

    update = function () {
      _this.updateReadyToFix();
    };_this.





















    renderHeader = function (fixed) {var _cn, _cn2;if (fixed === void 0) {fixed = false;}
      return (
        _react.default.createElement("div", { className: (0, _classnames.default)(_SidePage.jsStyles.header(), (_cn = {}, _cn[_SidePage.jsStyles.headerFixed(_this.theme)] = fixed, _cn)) },
        _this.renderClose(),
        _react.default.createElement("div", { className: (0, _classnames.default)(_SidePage.jsStyles.title(), (_cn2 = {}, _cn2[_SidePage.jsStyles.titleFixed()] = fixed, _cn2)) },
        (0, _utils.isFunction)(_this.props.children) ? _this.props.children(fixed) : _this.props.children)));



    };_this.

    renderCloseContent = function (fixed) {return (
        _react.default.createElement(_SidePageContext.SidePageContext.Consumer, null,
        function (_ref) {var _cn3;var requestClose = _ref.requestClose;return (
            _react.default.createElement("a", {
              className: (0, _classnames.default)(_SidePage.jsStyles.close(_this.theme), (_cn3 = {}, _cn3[
              _SidePage.jsStyles.fixed()] = fixed, _cn3)),

              onClick: requestClose,
              "data-tid": "SidePage__close" },

            _react.default.createElement("span", { className: _SidePage.jsStyles.closeIcon() },
            _react.default.createElement(_CrossIcon.CrossIcon, null))));}));};_this.






    renderClose = function () {return (
        _react.default.createElement(_Sticky.Sticky, { side: "top", offset: CLOSE_ELEMENT_OFFSET },
        _this.renderCloseContent));};_this.



    updateReadyToFix = function () {
      if (_this.wrapper) {
        var wrapperScrolledUp = _this.wrapper.getBoundingClientRect().top;
        var isReadyToFix = _this.regularHeight + wrapperScrolledUp <= FIXED_HEADER_HEIGHT;
        _this.setState(function (state) {return state.isReadyToFix !== isReadyToFix ? { isReadyToFix: isReadyToFix } : state;});
      }
    };_this.

    wrapperRef = function (el) {
      _this.wrapper = el;
    };return _this;}var _proto = SidePageHeader.prototype;_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var isReadyToFix = this.state.isReadyToFix;return _react.default.createElement("div", { ref: this.wrapperRef }, isReadyToFix ? _react.default.createElement(_Sticky.Sticky, { side: "top" }, this.renderHeader) : this.renderHeader());};(0, _createClass2.default)(SidePageHeader, [{ key: "regularHeight", get: function get() {var isReadyToFix = this.state.isReadyToFix;if (!this.wrapper) {return 0;}if (!isReadyToFix) {this.lastRegularHeight = this.wrapper.getBoundingClientRect().height;}return this.lastRegularHeight;} }]);return SidePageHeader;}(_react.default.Component);exports.SidePageHeader = SidePageHeader;SidePageHeader.__KONTUR_REACT_UI__ = 'SidePageHeader';
