// @skbkontur/react-ui/cjs/components/SidePage/SidePageFooter.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.SidePageFooter = void 0;var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));

var LayoutEvents = _interopRequireWildcard(require("../../lib/LayoutEvents"));
var _ThemeContext = require("../../lib/theming/ThemeContext");


var _SidePage = require("./SidePage.styles");
var _SidePageContext = require("./SidePageContext");









/**
                                                      * Футер сайдпейджа.
                                                      *
                                                      * @visibleName SidePage.Footer
                                                      */var

SidePageFooter = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(SidePageFooter, _React$Component);function SidePageFooter() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.



    context = _this.context;_this.

    state = {
      fixed: false };_this.


    theme = void 0;_this.
    content = null;_this.
    wrapper = null;_this.
    layoutSub = null;_this.

























    update = function () {
      _this.setProperStyles();
    };_this.




























    refContent = function (node) {
      _this.content = node;
    };_this.

    refWrapper = function (node) {
      _this.wrapper = node;
    };_this.

    setProperStyles = function () {
      if (_this.wrapper && _this.content) {
        var wrapperRect = _this.wrapper.getBoundingClientRect();
        var contentRect = _this.content.getBoundingClientRect();
        var _fixed = wrapperRect.top > contentRect.top;
        _this.setState({ fixed: _fixed });
      }
    };return _this;}var _proto = SidePageFooter.prototype;_proto.componentDidMount = function componentDidMount() {this.context.footerRef(this);this.update();this.layoutSub = LayoutEvents.addListener(this.update);};_proto.componentWillUnmount = function componentWillUnmount() {this.context.footerRef(null);if (this.layoutSub) {this.layoutSub.remove();}};_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var _this3 = this;return _react.default.createElement("div", { style: { height: this.getContentHeight() }, ref: this.refWrapper }, _react.default.createElement(_SidePageContext.SidePageContext.Consumer, null, function (_ref) {var _cn;var getWidth = _ref.getWidth;return _react.default.createElement("div", { className: _SidePage.jsStyles.footer(), style: { width: getWidth() } }, _react.default.createElement("div", { className: (0, _classnames.default)(_SidePage.jsStyles.footerContent(), (_cn = {}, _cn[_SidePage.jsStyles.panel(_this3.theme)] = !!_this3.props.panel, _cn[_SidePage.jsStyles.footerFixed(_this3.theme)] = _this3.state.fixed, _cn)), ref: _this3.refContent }, _this3.props.children));}));};_proto.

  getContentHeight = function getContentHeight() {
    if (!this.content) {
      return 'auto';
    }
    return this.content.getBoundingClientRect().height;
  };return SidePageFooter;}(_react.default.Component);exports.SidePageFooter = SidePageFooter;SidePageFooter.__KONTUR_REACT_UI__ = 'SidePageFooter';SidePageFooter.contextType = _SidePageContext.SidePageContext;
