// @skbkontur/react-ui/cjs/components/SidePage/SidePage.js
"use strict";var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.SidePage = void 0;var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));var _react = _interopRequireDefault(require("react"));
var _reactTransitionGroup = require("react-transition-group");
var _classnames = _interopRequireDefault(require("classnames"));

var _identifiers = require("../../lib/events/keyboard/identifiers");
var LayoutEvents = _interopRequireWildcard(require("../../lib/LayoutEvents"));
var _stopPropagation = require("../../lib/events/stopPropagation");
var _HideBodyVerticalScroll = require("../../internal/HideBodyVerticalScroll");
var _ModalStack = require("../../lib/ModalStack");
var _RenderContainer = require("../../internal/RenderContainer");
var _RenderLayer = require("../../internal/RenderLayer");
var _ZIndex = require("../../internal/ZIndex");
var _ThemeContext = require("../../lib/theming/ThemeContext");


var _SidePageBody = require("./SidePageBody");
var _SidePageContainer = require("./SidePageContainer");
var _SidePageContext = require("./SidePageContext");
var _SidePageFooter = require("./SidePageFooter");
var _SidePageHeader = require("./SidePageHeader");
var _SidePage = require("./SidePage.styles");
















































var TRANSITION_TIMEOUT = 200;

/**
                               * Сайдпейдж
                               *
                               * Содержит в себе три компоненты: **SidePage.Header**,
                               * **SidePage.Body** и **SidePage.Footer**
                               *
                               * Для отображения серой плашки в футере в компонент
                               * **Footer** необходимо передать пропс **panel**
                               */var
SidePage = /*#__PURE__*/function (_React$Component) {(0, _inheritsLoose2.default)(SidePage, _React$Component);function SidePage() {var _this;for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}_this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;_this.






    state = {};_this.
    theme = void 0;_this.
    stackSubscription = null;_this.
    layoutRef = null;_this.
    footer = null;_this.


















    updateLayout = function () {
      if (_this.footer) {
        _this.footer.update();
      }
    };_this.












































































    getWidth = function () {
      if (!_this.layoutRef) {
        return 'auto';
      }
      return _this.layoutRef.getBoundingClientRect().width;
    };_this.











































    handleStackChange = function (stack) {
      var sidePages = stack.filter(function (x) {return x instanceof SidePage;});
      var currentSidePagePosition = sidePages.indexOf((0, _assertThisInitialized2.default)(_this));
      var isSidePageOnStackTop = stack[0] instanceof SidePage;

      var hasMargin = sidePages.length > 1 && currentSidePagePosition === sidePages.length - 1;
      var hasShadow = sidePages.length < 3 || currentSidePagePosition > sidePages.length - 3;
      var hasBackground = currentSidePagePosition === sidePages.length - 1 && isSidePageOnStackTop;

      _this.setState({
        stackPosition: stack.indexOf((0, _assertThisInitialized2.default)(_this)),
        hasMargin: hasMargin,
        hasShadow: hasShadow,
        hasBackground: hasBackground });

    };_this.

    handleClickOutside = function (e) {
      if (_this.state.stackPosition === 0 && !_this.props.ignoreBackgroundClick) {
        // ignore mousedown on window scrollbar
        if (e instanceof MouseEvent && e.clientX > document.documentElement.clientWidth) {
          return;
        }
        _this.requestClose();
      }
    };_this.

    handleKeyDown = function (e) {
      if (_this.state.stackPosition !== 0) {
        return;
      }
      if ((0, _identifiers.isKeyEscape)(e)) {
        (0, _stopPropagation.stopPropagation)(e);
        _this.requestClose();
      }
    };_this.

    requestClose = function () {
      if (_this.props.disableClose) {
        return;
      }
      if (_this.props.onClose) {
        _this.props.onClose();
      }
    };_this.

    footerRef = function (ref) {
      _this.footer = ref;
    };return _this;}var _proto = SidePage.prototype;_proto.componentDidMount = function componentDidMount() {window.addEventListener('keydown', this.handleKeyDown);this.stackSubscription = _ModalStack.ModalStack.add(this, this.handleStackChange);};_proto.componentWillUnmount = function componentWillUnmount() {window.removeEventListener('keydown', this.handleKeyDown);if (this.stackSubscription != null) {this.stackSubscription.remove();}_ModalStack.ModalStack.remove(this);} /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              * Обновляет разметку компонента.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              * @public
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              */;_proto.render = function render() {var _this2 = this;return _react.default.createElement(_ThemeContext.ThemeContext.Consumer, null, function (theme) {_this2.theme = theme;return _this2.renderMain();});};_proto.renderMain = function renderMain() {var _this$props = this.props,blockBackground = _this$props.blockBackground,disableAnimations = _this$props.disableAnimations;return _react.default.createElement(_RenderContainer.RenderContainer, null, _react.default.createElement("div", null, blockBackground && this.renderShadow(), _react.default.createElement(_reactTransitionGroup.CSSTransition, { in: true, classNames: this.getTransitionNames(), appear: !disableAnimations, enter: !disableAnimations, exit: false, timeout: { enter: TRANSITION_TIMEOUT, exit: TRANSITION_TIMEOUT } }, this.renderContainer())));};_proto.renderContainer = function renderContainer() {var _cn,_this3 = this;var _this$props2 = this.props,width = _this$props2.width,blockBackground = _this$props2.blockBackground,fromLeft = _this$props2.fromLeft;return _react.default.createElement(_ZIndex.ZIndex, { priority: 'Sidepage', "data-tid": "SidePage__root", className: (0, _classnames.default)((_cn = {}, _cn[_SidePage.jsStyles.root()] = true, _cn[_SidePage.jsStyles.leftSide(this.theme)] = Boolean(fromLeft), _cn)), onScroll: LayoutEvents.emit, createStackingContext: true, style: { width: width || (blockBackground ? 800 : 500) } }, _react.default.createElement(_RenderLayer.RenderLayer, { onClickOutside: this.handleClickOutside, active: true }, _react.default.createElement("div", { "data-tid": "SidePage__container", className: (0, _classnames.default)(_SidePage.jsStyles.container(this.theme), this.state.hasShadow && _SidePage.jsStyles.shadow(this.theme)), style: this.getSidebarStyle() }, _react.default.createElement("div", { ref: function ref(_) {return _this3.layoutRef = _;}, className: _SidePage.jsStyles.layout() }, _react.default.createElement(_SidePageContext.SidePageContext.Provider, { value: { requestClose: this.requestClose, getWidth: this.getWidth, updateLayout: this.updateLayout, footerRef: this.footerRef } }, this.props.children)))));};_proto.renderShadow = function renderShadow() {var _cn2;return _react.default.createElement(_ZIndex.ZIndex, { priority: 'Sidepage', className: _SidePage.jsStyles.overlay(), onScroll: LayoutEvents.emit }, _react.default.createElement(_HideBodyVerticalScroll.HideBodyVerticalScroll, { key: "hbvs" }), _react.default.createElement("div", { key: "overlay", className: (0, _classnames.default)((_cn2 = {}, _cn2[_SidePage.jsStyles.background()] = true, _cn2[_SidePage.jsStyles.backgroundGray()] = this.state.hasBackground, _cn2)) }));};_proto.getSidebarStyle = function getSidebarStyle() {var sidePageStyle = {};if (this.state.hasMargin) {if (this.props.fromLeft) {sidePageStyle.marginLeft = 20;} else {sidePageStyle.marginRight = 20;}}return sidePageStyle;};_proto.getTransitionNames = function getTransitionNames() {var transition = this.props.fromLeft ? _SidePage.jsStyles.transitionRight : _SidePage.jsStyles.transitionLeft;return { enter: transition(), enterActive: _SidePage.jsStyles.transitionActive(), exit: _SidePage.jsStyles.transitionLeave(), exitActive: _SidePage.jsStyles.transitionLeaveActive(), appear: transition(), appearActive: _SidePage.jsStyles.transitionActive() };};return SidePage;}(_react.default.Component);exports.SidePage = SidePage;SidePage.__KONTUR_REACT_UI__ = 'SidePage';SidePage.Header = _SidePageHeader.SidePageHeader;SidePage.Body = _SidePageBody.SidePageBody;SidePage.Footer = _SidePageFooter.SidePageFooter;SidePage.Container = _SidePageContainer.SidePageContainer;
