// @skbkontur/react-ui/cjs/components/SidePage/SidePage.styles.js
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");exports.__esModule = true;exports.jsStyles = void 0;var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));var _Emotion = require("../../lib/theming/Emotion");function _templateObject26() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      opacity: 0.01;\n      transition: opacity 0.15s ease-out;\n    "]);_templateObject26 = function _templateObject26() {return data;};return data;}function _templateObject25() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      opacity: 1;\n    "]);_templateObject25 = function _templateObject25() {return data;};return data;}function _templateObject24() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transition: transform 0.18s cubic-bezier(0.22, 0.61, 0.36, 1), opacity 0.18s cubic-bezier(0.22, 0.61, 0.36, 1);\n      opacity: 1;\n      transform: translate(0, 0) !important;\n    "]);_templateObject24 = function _templateObject24() {return data;};return data;}function _templateObject23() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateX(-100px);\n    "]);_templateObject23 = function _templateObject23() {return data;};return data;}function _templateObject22() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      transform: translateX(100px);\n    "]);_templateObject22 = function _templateObject22() {return data;};return data;}function _templateObject21() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      border-top: 1px solid ", ";\n    "]);_templateObject21 = function _templateObject21() {return data;};return data;}function _templateObject20() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding: 20px 35px 20px 30px;\n    "]);_templateObject20 = function _templateObject20() {return data;};return data;}function _templateObject19() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      bottom: 0;\n      position: fixed;\n      z-index: 10;\n    "]);_templateObject19 = function _templateObject19() {return data;};return data;}function _templateObject18() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      line-height: 24px;\n    "]);_templateObject18 = function _templateObject18() {return data;};return data;}function _templateObject17() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      display: inline-block;\n      height: 12px;\n      line-height: 0;\n      width: 12px;\n    "]);_templateObject17 = function _templateObject17() {return data;};return data;}function _templateObject16() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", " !important;\n      border-top: 0 !important;\n    "]);_templateObject16 = function _templateObject16() {return data;};return data;}function _templateObject15() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      color: ", ";\n      cursor: pointer;\n      position: absolute;\n      right: 30px;\n      text-align: center;\n      text-decoration: none;\n      width: 24px;\n\n      ", "& {\n        line-height: 24px;\n      }\n\n      &:hover {\n        color: ", ";\n      }\n    "]);_templateObject15 = function _templateObject15() {return data;};return data;}function _templateObject14() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      left: 0;\n      right: auto;\n\n      & ", " {\n        float: left;\n      }\n    "]);_templateObject14 = function _templateObject14() {return data;};return data;}function _templateObject13() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      ", "& {\n        box-shadow: ", ";\n      }\n    "]);_templateObject13 = function _templateObject13() {return data;};return data;}function _templateObject12() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: #333;\n      opacity: 0.6;\n    "]);_templateObject12 = function _templateObject12() {return data;};return data;}function _templateObject11() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      position: absolute;\n      width: 100%;\n    "]);_templateObject11 = function _templateObject11() {return data;};return data;}function _templateObject10() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      overflow: hidden;\n      text-overflow: ellipsis;\n      white-space: nowrap;\n    "]);_templateObject10 = function _templateObject10() {return data;};return data;}function _templateObject9() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-right: 85px;\n      padding-left: 30px;\n    "]);_templateObject9 = function _templateObject9() {return data;};return data;}function _templateObject8() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      box-shadow: 0 1px ", ";\n      font-size: 18px;\n      line-height: 24px;\n      padding: 13px 0;\n    "]);_templateObject8 = function _templateObject8() {return data;};return data;}function _templateObject7() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      font-size: 22px;\n      line-height: 30px;\n      padding: 25px 0;\n      position: relative;\n      width: 100%;\n    "]);_templateObject7 = function _templateObject7() {return data;};return data;}function _templateObject6() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      align-items: stretch;\n      display: flex;\n      flex-direction: column;\n    "]);_templateObject6 = function _templateObject6() {return data;};return data;}function _templateObject5() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      background: ", ";\n      float: right;\n      height: 100%;\n      width: 100%;\n      overflow-y: auto;\n      position: relative;\n      white-space: normal;\n    "]);_templateObject5 = function _templateObject5() {return data;};return data;}function _templateObject4() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      padding-left: 30px;\n      padding-right: 35px;\n    "]);_templateObject4 = function _templateObject4() {return data;};return data;}function _templateObject3() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      z-index: 0;\n    "]);_templateObject3 = function _templateObject3() {return data;};return data;}function _templateObject2() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      position: fixed;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n    "]);_templateObject2 = function _templateObject2() {return data;};return data;}function _templateObject() {var data = (0, _taggedTemplateLiteralLoose2.default)(["\n      height: 100%;\n      position: fixed;\n      right: 0;\n      top: 0;\n    "]);_templateObject = function _templateObject() {return data;};return data;}


var styles = {
  root: function root() {
    return (0, _Emotion.css)(_templateObject());





  },

  overlay: function overlay() {
    return (0, _Emotion.css)(_templateObject2());






  },

  body: function body() {
    return (0, _Emotion.css)(_templateObject3());



  },

  bodyContainer: function bodyContainer() {
    return (0, _Emotion.css)(_templateObject4());



  },

  container: function container(t) {
    return (0, _Emotion.css)(_templateObject5(),
    t.bgDefault);







  },

  layout: function layout() {
    return (0, _Emotion.css)(_templateObject6());




  },

  header: function header() {
    return (0, _Emotion.css)(_templateObject7());






  },

  headerFixed: function headerFixed(t) {
    return (0, _Emotion.css)(_templateObject8(),
    t.bgDefault,
    t.borderColorGrayLight);




  },

  title: function title() {
    return (0, _Emotion.css)(_templateObject9());



  },

  titleFixed: function titleFixed() {
    return (0, _Emotion.css)(_templateObject10());




  },

  background: function background() {
    return (0, _Emotion.css)(_templateObject11());




  },

  backgroundGray: function backgroundGray() {
    return (0, _Emotion.css)(_templateObject12());



  },

  shadow: function shadow(t) {
    return (0, _Emotion.css)(_templateObject13(),
    (0, _Emotion.cssName)(styles.container(t)),
    t.sidePageContainerShadow);


  },

  leftSide: function leftSide(t) {
    return (0, _Emotion.css)(_templateObject14(),



    (0, _Emotion.cssName)(styles.container(t)));



  },

  close: function close(t) {
    return (0, _Emotion.css)(_templateObject15(),
    t.sidePageCloseButtonColor,







    (0, _Emotion.cssName)(styles.fixed()),




    t.sidePageCloseButtonHoverColor);


  },

  panel: function panel(t) {
    return (0, _Emotion.css)(_templateObject16(),
    t.sidePageFooterPanelBg);


  },

  closeIcon: function closeIcon() {
    return (0, _Emotion.css)(_templateObject17());





  },

  fixed: function fixed() {
    return (0, _Emotion.css)(_templateObject18());


  },

  footer: function footer() {
    return (0, _Emotion.css)(_templateObject19());




  },

  footerContent: function footerContent() {
    return (0, _Emotion.css)(_templateObject20());


  },

  footerFixed: function footerFixed(t) {
    return (0, _Emotion.css)(_templateObject21(),
    t.bgDefault,
    t.borderColorGrayLight);

  },

  transitionLeft: function transitionLeft() {
    return (0, _Emotion.css)(_templateObject22());


  },

  transitionRight: function transitionRight() {
    return (0, _Emotion.css)(_templateObject23());


  },

  transitionActive: function transitionActive() {
    return (0, _Emotion.css)(_templateObject24());




  },

  transitionLeave: function transitionLeave() {
    return (0, _Emotion.css)(_templateObject25());


  },

  transitionLeaveActive: function transitionLeaveActive() {
    return (0, _Emotion.css)(_templateObject26());



  } };


var jsStyles = (0, _Emotion.memoizeStyle)(styles);exports.jsStyles = jsStyles;
